{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9278ec8a-aa99-4547-b7f4-33f8e9f6042b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(edgeR)\n",
    "library(sva)\n",
    "library(fgsea)\n",
    "library(tidyr)\n",
    "source('./ABCA7lof2//degs.r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "20b7f287-4a15-4660-aff9-9378eff2523f",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_data = readRDS('./processed_data/single_cell/stats_input_data_0825.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "16f02950-bee9-46f9-ba6f-a5353b4f6998",
   "metadata": {},
   "outputs": [],
   "source": [
    "summed_counts = all_data$summed_counts_by_ind # does it make sense to be summing on the counts vs taking the mean of the counts? --> compare to nebula? --> ok because we are normalizing?\n",
    "meta = all_data$summary\n",
    "meta$seq_batch = ifelse(meta$seq_batch=='JBM',1,0)\n",
    "expressed = all_data$expressed10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "85c0bbcf-5eac-4ee3-820d-82c267fe1247",
   "metadata": {},
   "outputs": [],
   "source": [
    "# filter out individuals with too few cells per celltype\n",
    "keep = summed_counts$ncells>=10\n",
    "summed_counts_indexed = summed_counts$summed_counts[,keep]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "a208f24a-0b62-4c91-9ab2-8b62597ecadc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  "
     ]
    }
   ],
   "source": [
    "# compute degs (all samples) without gene cutoff\n",
    "exp = lapply(names(expressed), function(x) rownames(summed_counts$summed_counts))\n",
    "names(exp) = names(expressed)\n",
    "             \n",
    "vars = c('LOF', 'amyloid', 'nft',  'msex', 'age_death', 'pmi', 'seq_batch', 'APOE4')\n",
    "limma_inputs = get_limma_inputs(summed_counts_indexed, exp, meta, vars)\n",
    "aggs = limma_inputs$aggs\n",
    "metadata = limma_inputs$metadata\n",
    "\n",
    "\n",
    "degs_all = lapply(names(aggs)[!names(aggs)%in%c('Vascular')], function(x) RunDiffExprAnalysisLimma(aggs[[x]], metadata[[x]], n.sv=NULL))\n",
    "names(degs_all) = names(aggs)[!names(aggs)%in%c('Vascular')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "7478bae0-27a3-4f44-8e71-9f5025f93333",
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(degs_all,'./processed_data/for_plotting/degs_no_exp.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "48f38d7b-3b10-4976-902f-c9e2b9533164",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant surrogate variables is:  10 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  9 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  7 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  9 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  7 \n",
      "Iteration (out of 5 ):1  2  3  4  5  "
     ]
    }
   ],
   "source": [
    "# compute degs (all samples)\n",
    "vars = c('LOF', 'amyloid', 'nft',  'msex', 'age_death', 'pmi', 'seq_batch', 'APOE4')\n",
    "limma_inputs = get_limma_inputs(summed_counts_indexed, expressed, meta, vars)\n",
    "aggs = limma_inputs$aggs\n",
    "metadata = limma_inputs$metadata\n",
    "\n",
    "\n",
    "degs_all = lapply(names(aggs)[!names(aggs)%in%c('Vascular')], function(x) RunDiffExprAnalysisLimma(aggs[[x]], metadata[[x]], n.sv=NULL))\n",
    "names(degs_all) = names(aggs)[!names(aggs)%in%c('Vascular')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3dfb9905-66b4-4922-a81a-700aa421fa81",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the ids\n",
    "ids = strsplit(colnames(summed_counts_indexed), '[.]')\n",
    "ids = unlist(lapply(1:length(ids), function(x) ids[[x]][2]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "262fbf0a-02c3-417d-85b9-b6e0f5ef554a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant surrogate variables is:  6 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  6 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  "
     ]
    }
   ],
   "source": [
    "# compute degs (no APOE4 samples)\n",
    "vars = c('LOF', 'amyloid', 'nft',  'msex', 'age_death', 'pmi', 'seq_batch')\n",
    "sele = rownames(meta)[meta$APOE4==0]\n",
    "sc_noAPOE4 = summed_counts_indexed[,ids%in%sele]\n",
    "limma_inputs = get_limma_inputs(sc_noAPOE4, expressed, meta[sele,], vars)\n",
    "aggs = limma_inputs$aggs\n",
    "metadata = limma_inputs$metadata\n",
    "degs_noAPOE4 = lapply(names(aggs)[!names(aggs)%in%c('Vascular')], function(x) RunDiffExprAnalysisLimma(aggs[[x]], metadata[[x]], n.sv=NULL, exclude_apoe=TRUE))\n",
    "names(degs_noAPOE4) = names(aggs)[!names(aggs)%in%c('Vascular')]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9fb2f2a5-2706-4811-b79d-2f3927497fd2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  5 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  "
     ]
    }
   ],
   "source": [
    "# compute degs (no APOE4 samples)\n",
    "vars = c('LOF', 'amyloid', 'nft',  'msex', 'age_death', 'pmi', 'APOE4')\n",
    "sele = rownames(meta)[meta$seq_batch==1]\n",
    "sc_JBM = summed_counts_indexed[,ids%in%sele]\n",
    "limma_inputs = get_limma_inputs(sc_JBM, expressed, meta[sele,], vars)\n",
    "aggs = limma_inputs$aggs\n",
    "metadata = limma_inputs$metadata\n",
    "degs_JBM = lapply(names(aggs)[!names(aggs)%in%c('Vascular')], function(x) RunDiffExprAnalysisLimma(aggs[[x]], metadata[[x]], n.sv=NULL, exclude_batch=TRUE))\n",
    "names(degs_JBM) = names(aggs)[!names(aggs)%in%c('Vascular')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5a562439-f63a-425a-b830-d3ccab407148",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  2 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  4 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  3 \n",
      "Iteration (out of 5 ):1  2  3  4  5  Number of significant surrogate variables is:  2 \n",
      "Iteration (out of 5 ):1  2  3  4  5  "
     ]
    }
   ],
   "source": [
    "# exclude both\n",
    "vars = c('LOF', 'amyloid', 'nft',  'msex', 'age_death', 'pmi')\n",
    "sele = rownames(meta)[(meta$seq_batch==1) & (meta$APOE4==0)]\n",
    "sc_exclude_both = summed_counts_indexed[,ids%in%sele]\n",
    "limma_inputs = get_limma_inputs(sc_exclude_both, expressed, meta[sele,], vars)\n",
    "aggs = limma_inputs$aggs\n",
    "metadata = limma_inputs$metadata\n",
    "degs_exclude_both = lapply(names(aggs)[!names(aggs)%in%c('Vascular')], function(x) RunDiffExprAnalysisLimma(aggs[[x]], metadata[[x]], n.sv=NULL, exclude_both=TRUE))\n",
    "names(degs_exclude_both) = names(aggs)[!names(aggs)%in%c('Vascular')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "239c315f-3f48-4b88-98eb-8626c798b002",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the results\n",
    "degs = list()\n",
    "degs[['degs_all']] = degs_all\n",
    "degs[['degs_exclude_both']] = degs_exclude_both\n",
    "degs[['degs_JBM']] = degs_JBM\n",
    "degs[['degs_noAPOE4']] = degs_noAPOE4\n",
    "\n",
    "saveRDS(degs, './processed_data/single_cell/pseudobulk_degs_0825.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a9a5f6d4-41e8-43d4-a5c1-106c8e8e9a13",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save deg scores as matrix\n",
    "out = list()\n",
    "scores = get_deg_scores(degs$degs_all)\n",
    "for(i in names(scores)){\n",
    "    df = as.data.frame(scores[[i]]$scores)\n",
    "    colnames(df) = c('score')\n",
    "    df$gene = rownames(df)\n",
    "    df$celltype = i\n",
    "    out[[i]] = df\n",
    "}\n",
    "all_scores = as.data.frame(do.call('rbind', out)%>%pivot_wider(., values_from='score', names_from='celltype'))\n",
    "rownames(all_scores) = all_scores$gene\n",
    "all_scores$gene = NULL\n",
    "all_scores[is.na(all_scores)] = 0\n",
    "write.csv(all_scores, './processed_data/single_cell/all_scores_0825.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "f65d1366-d18b-4e59-9d23-03d16c9c909e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load degs\n",
    "degs = readRDS('./processed_data/single_cell/pseudobulk_degs_0825.rds')$degs_all\n",
    "scores = get_deg_scores(degs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9b4d7dbc-23bd-47ae-88cb-13845559fed1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# compute fgsea results\n",
    "all_paths = read.csv('./processed_data/genesets/all_paths.csv', row.names = 'X')\n",
    "pathways = as.list(as.data.frame(t(all_paths)))\n",
    "temp = lapply(names(pathways), function(x) pathways[[x]][!(pathways[[x]]=='')])\n",
    "names(temp) = names(pathways)\n",
    "              \n",
    "out = lapply(names(scores), function(x) cbind(x, fgsea(pathways = temp, stats = scores[[x]]$scores, minSize = 5, maxSize = 1000)))\n",
    "names(out) = names(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "946c9c96-7747-4a79-939c-5492e5e81a36",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the fgsea results\n",
    "saveRDS(out, './processed_data/for_plotting/fgsea_out_0825.rds')\n",
    "\n",
    "# save all LE results\n",
    "df = as.data.frame(do.call('rbind', out))\n",
    "df = df[,!colnames(df)=='leadingEdge']\n",
    "write.csv(df, './processed_data/for_plotting/fgsea_out_0825.csv')\n",
    "\n",
    "# save all fgsea results as csv\n",
    "df = as.data.frame(do.call('rbind', out))\n",
    "df = df[,!colnames(df)=='leadingEdge']\n",
    "write.csv(df, './processed_data/for_plotting/fgsea_0825.csv')\n",
    "\n",
    "# save LE genes per celltype\n",
    "for(i in names(out)){\n",
    "    res = out[[i]]\n",
    "    res = res[res$pval<0.05,]\n",
    "\n",
    "    df = as.data.frame(unique(unname(unlist(res$leadingEdge))))\n",
    "    colnames(df) = 'gene'\n",
    "    write.csv(df, paste0(paste0('./processed_data/for_plotting/leading_edge_0825', i), '.csv'))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7b5fa2c4-1520-46b1-b35c-f7c799e6231f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save all LEs\n",
    "out = readRDS('./processed_data/for_plotting/fgsea_out_0825.rds')\n",
    "res = do.call('rbind', out)\n",
    "res$score = sign(res$NES) * -log10(res$pval)\n",
    "res = res[order(res$pval,decreasing=FALSE),]\n",
    "res = res[res$pval<0.05,]\n",
    "# save all leading edge genes\n",
    "df = as.data.frame(unique(unname(unlist(res$leadingEdge))))\n",
    "colnames(df) = 'gene'\n",
    "write.csv(df, './processed_data/for_plotting/leading_edge_0825.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "4336db22-b44d-4876-8e02-5b0b05382502",
   "metadata": {},
   "outputs": [],
   "source": [
    "# enrichment for lipid genesets\n",
    "get_gset_names_by_category = function(cat, gsets){\n",
    "  gset = unlist(lapply(gsets, function(x) unlist(sum(sapply(cat, grepl, x))>0)))\n",
    "  gset = (gsets[gset])\n",
    "  return(gset)\n",
    "}\n",
    "\n",
    "pathways = readRDS('../ABCA7lof/processed_data/pathway_databases/pathways.rds')$all_paths\n",
    "\n",
    "                       \n",
    "o = get_gset_names_by_category(c('sterol', 'lipid', 'glycer', 'fatt', 'ceramide', 'phosphatidyl'), names(pathways))\n",
    "temp2 = pathways[o]\n",
    "out = lapply(names(scores), function(x) cbind(x, fgsea(pathways = temp2, stats = scores[[x]]$scores, minSize = 5, maxSize = 1000)))\n",
    "names(out) = names(scores)\n",
    "                       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "dcfb4da5-aa19-4a2e-b1a7-a74a1aabff6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save lipid paths\n",
    "res = out$Ex\n",
    "res = res[order(res$pval,decreasing=FALSE),]\n",
    "res = res[res$pval<0.05,]\n",
    "saveRDS(res, './processed_data/for_plotting/fgsea_out_ex_lipid.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "711f4ee3-1b11-491d-8781-e28e1546b650",
   "metadata": {},
   "outputs": [],
   "source": [
    "pathways = readRDS('../ABCA7lof/processed_data/pathway_databases/pathways.rds')$all_paths\n",
    "o = get_gset_names_by_category(c('kappa'), names(pathways))\n",
    "temp2 = pathways[o]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b5264b23-e4c0-467a-af60-58584055af99",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load degs\n",
    "degs = readRDS('./processed_data/single_cell/pseudobulk_degs_0825.rds')$degs_all\n",
    "scores = get_deg_scores(degs)\n",
    "out = lapply(names(scores), function(x) cbind(x, fgsea(pathways = temp2, stats = scores[[x]]$scores, minSize = 5, maxSize = 1000)))\n",
    "names(out) = names(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a0f56c3d-9561-41d8-9b60-0d51ebd945a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "p = readRDS('../ABCA7lof/processed_data/pathway_databases/pathways.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f9e69548-af63-49b0-b2ed-ddeb9a4c483f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>$`NF-kappa B signaling pathway`</strong> = <style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'UBE2I'</li><li>'IL1R1'</li><li>'IGH'</li><li>'VCAM1'</li><li>'TICAM2'</li><li>'TICAM1'</li><li>'EDAR'</li><li>'CXCL12'</li><li>'RELA'</li><li>'TNF'</li><li>'RELB'</li><li>'MAP3K7'</li><li>'CARD10'</li><li>'CARD11'</li><li>'PIAS4'</li><li>'EDARADD'</li><li>'DDX58'</li><li>'GADD45B'</li><li>'GADD45A'</li><li>'TNFSF14'</li><li>'TRAF1'</li><li>'GADD45G'</li><li>'TIRAP'</li><li>'PRKCB'</li><li>'TRAF3'</li><li>'TRAF2'</li><li>'NFKB1'</li><li>'NFKB2'</li><li>'MYD88'</li><li>'TRAF5'</li><li>'CSNK2B'</li><li>'TRAF6'</li><li>'BCL2L1'</li><li>'BTK'</li><li>'TAB3'</li><li>'PRKCQ'</li><li>'TAB2'</li><li>'MAP3K14'</li><li>'CARD14'</li><li>'TAB1'</li><li>'BIRC2'</li><li>'BIRC3'</li><li>'CCL4L1'</li><li>'CCL4L2'</li><li>'BCL10'</li><li>'PLAU'</li><li>'EDA2R'</li><li>'ICAM1'</li><li>'TNFSF11'</li><li>'PLCG1'</li><li>'PLCG2'</li><li>'CD14'</li><li>'LYN'</li><li>'CCL21'</li><li>'CFLAR'</li><li>'IRAK4'</li><li>'TNFRSF1A'</li><li>'IL1B'</li><li>'NFKBIA'</li><li>'CYLD'</li><li>'BCL2'</li><li>'ATM'</li><li>'TLR4'</li><li>'LAT'</li><li>'EDA'</li><li>'CXCL8'</li><li>'CD40'</li><li>'TNFRSF11A'</li><li>'BCL2A1'</li><li>'TRADD'</li><li>'TNFAIP3'</li><li>'LY96'</li><li>'CXCL2'</li><li>'CXCL1'</li><li>'CXCL3'</li><li>'IRAK1'</li><li>'LBP'</li><li>'BLNK'</li><li>'RIPK1'</li><li>'PIDD1'</li><li>'CSNK2A2'</li><li>'SYK'</li><li>'CSNK2A1'</li><li>'CHUK'</li><li>'PARP1'</li><li>'CSNK2A3'</li><li>'ZAP70'</li><li>'LCK'</li><li>'LTBR'</li><li>'XIAP'</li><li>'LTB'</li><li>'LTA'</li><li>'CD40LG'</li><li>'CCL13'</li><li>'TNFRSF13C'</li><li>'TNFSF13B'</li><li>'PTGS2'</li><li>'MALT1'</li><li>'IKBKB'</li><li>'TRIM25'</li><li>'CCL4'</li><li>'ERC1'</li><li>'CCL19'</li><li>'IKBKG'</li></ol>\n"
      ],
      "text/latex": [
       "\\textbf{\\$`NF-kappa B signaling pathway`} = \\begin{enumerate*}\n",
       "\\item 'UBE2I'\n",
       "\\item 'IL1R1'\n",
       "\\item 'IGH'\n",
       "\\item 'VCAM1'\n",
       "\\item 'TICAM2'\n",
       "\\item 'TICAM1'\n",
       "\\item 'EDAR'\n",
       "\\item 'CXCL12'\n",
       "\\item 'RELA'\n",
       "\\item 'TNF'\n",
       "\\item 'RELB'\n",
       "\\item 'MAP3K7'\n",
       "\\item 'CARD10'\n",
       "\\item 'CARD11'\n",
       "\\item 'PIAS4'\n",
       "\\item 'EDARADD'\n",
       "\\item 'DDX58'\n",
       "\\item 'GADD45B'\n",
       "\\item 'GADD45A'\n",
       "\\item 'TNFSF14'\n",
       "\\item 'TRAF1'\n",
       "\\item 'GADD45G'\n",
       "\\item 'TIRAP'\n",
       "\\item 'PRKCB'\n",
       "\\item 'TRAF3'\n",
       "\\item 'TRAF2'\n",
       "\\item 'NFKB1'\n",
       "\\item 'NFKB2'\n",
       "\\item 'MYD88'\n",
       "\\item 'TRAF5'\n",
       "\\item 'CSNK2B'\n",
       "\\item 'TRAF6'\n",
       "\\item 'BCL2L1'\n",
       "\\item 'BTK'\n",
       "\\item 'TAB3'\n",
       "\\item 'PRKCQ'\n",
       "\\item 'TAB2'\n",
       "\\item 'MAP3K14'\n",
       "\\item 'CARD14'\n",
       "\\item 'TAB1'\n",
       "\\item 'BIRC2'\n",
       "\\item 'BIRC3'\n",
       "\\item 'CCL4L1'\n",
       "\\item 'CCL4L2'\n",
       "\\item 'BCL10'\n",
       "\\item 'PLAU'\n",
       "\\item 'EDA2R'\n",
       "\\item 'ICAM1'\n",
       "\\item 'TNFSF11'\n",
       "\\item 'PLCG1'\n",
       "\\item 'PLCG2'\n",
       "\\item 'CD14'\n",
       "\\item 'LYN'\n",
       "\\item 'CCL21'\n",
       "\\item 'CFLAR'\n",
       "\\item 'IRAK4'\n",
       "\\item 'TNFRSF1A'\n",
       "\\item 'IL1B'\n",
       "\\item 'NFKBIA'\n",
       "\\item 'CYLD'\n",
       "\\item 'BCL2'\n",
       "\\item 'ATM'\n",
       "\\item 'TLR4'\n",
       "\\item 'LAT'\n",
       "\\item 'EDA'\n",
       "\\item 'CXCL8'\n",
       "\\item 'CD40'\n",
       "\\item 'TNFRSF11A'\n",
       "\\item 'BCL2A1'\n",
       "\\item 'TRADD'\n",
       "\\item 'TNFAIP3'\n",
       "\\item 'LY96'\n",
       "\\item 'CXCL2'\n",
       "\\item 'CXCL1'\n",
       "\\item 'CXCL3'\n",
       "\\item 'IRAK1'\n",
       "\\item 'LBP'\n",
       "\\item 'BLNK'\n",
       "\\item 'RIPK1'\n",
       "\\item 'PIDD1'\n",
       "\\item 'CSNK2A2'\n",
       "\\item 'SYK'\n",
       "\\item 'CSNK2A1'\n",
       "\\item 'CHUK'\n",
       "\\item 'PARP1'\n",
       "\\item 'CSNK2A3'\n",
       "\\item 'ZAP70'\n",
       "\\item 'LCK'\n",
       "\\item 'LTBR'\n",
       "\\item 'XIAP'\n",
       "\\item 'LTB'\n",
       "\\item 'LTA'\n",
       "\\item 'CD40LG'\n",
       "\\item 'CCL13'\n",
       "\\item 'TNFRSF13C'\n",
       "\\item 'TNFSF13B'\n",
       "\\item 'PTGS2'\n",
       "\\item 'MALT1'\n",
       "\\item 'IKBKB'\n",
       "\\item 'TRIM25'\n",
       "\\item 'CCL4'\n",
       "\\item 'ERC1'\n",
       "\\item 'CCL19'\n",
       "\\item 'IKBKG'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "**$`NF-kappa B signaling pathway`** = 1. 'UBE2I'\n",
       "2. 'IL1R1'\n",
       "3. 'IGH'\n",
       "4. 'VCAM1'\n",
       "5. 'TICAM2'\n",
       "6. 'TICAM1'\n",
       "7. 'EDAR'\n",
       "8. 'CXCL12'\n",
       "9. 'RELA'\n",
       "10. 'TNF'\n",
       "11. 'RELB'\n",
       "12. 'MAP3K7'\n",
       "13. 'CARD10'\n",
       "14. 'CARD11'\n",
       "15. 'PIAS4'\n",
       "16. 'EDARADD'\n",
       "17. 'DDX58'\n",
       "18. 'GADD45B'\n",
       "19. 'GADD45A'\n",
       "20. 'TNFSF14'\n",
       "21. 'TRAF1'\n",
       "22. 'GADD45G'\n",
       "23. 'TIRAP'\n",
       "24. 'PRKCB'\n",
       "25. 'TRAF3'\n",
       "26. 'TRAF2'\n",
       "27. 'NFKB1'\n",
       "28. 'NFKB2'\n",
       "29. 'MYD88'\n",
       "30. 'TRAF5'\n",
       "31. 'CSNK2B'\n",
       "32. 'TRAF6'\n",
       "33. 'BCL2L1'\n",
       "34. 'BTK'\n",
       "35. 'TAB3'\n",
       "36. 'PRKCQ'\n",
       "37. 'TAB2'\n",
       "38. 'MAP3K14'\n",
       "39. 'CARD14'\n",
       "40. 'TAB1'\n",
       "41. 'BIRC2'\n",
       "42. 'BIRC3'\n",
       "43. 'CCL4L1'\n",
       "44. 'CCL4L2'\n",
       "45. 'BCL10'\n",
       "46. 'PLAU'\n",
       "47. 'EDA2R'\n",
       "48. 'ICAM1'\n",
       "49. 'TNFSF11'\n",
       "50. 'PLCG1'\n",
       "51. 'PLCG2'\n",
       "52. 'CD14'\n",
       "53. 'LYN'\n",
       "54. 'CCL21'\n",
       "55. 'CFLAR'\n",
       "56. 'IRAK4'\n",
       "57. 'TNFRSF1A'\n",
       "58. 'IL1B'\n",
       "59. 'NFKBIA'\n",
       "60. 'CYLD'\n",
       "61. 'BCL2'\n",
       "62. 'ATM'\n",
       "63. 'TLR4'\n",
       "64. 'LAT'\n",
       "65. 'EDA'\n",
       "66. 'CXCL8'\n",
       "67. 'CD40'\n",
       "68. 'TNFRSF11A'\n",
       "69. 'BCL2A1'\n",
       "70. 'TRADD'\n",
       "71. 'TNFAIP3'\n",
       "72. 'LY96'\n",
       "73. 'CXCL2'\n",
       "74. 'CXCL1'\n",
       "75. 'CXCL3'\n",
       "76. 'IRAK1'\n",
       "77. 'LBP'\n",
       "78. 'BLNK'\n",
       "79. 'RIPK1'\n",
       "80. 'PIDD1'\n",
       "81. 'CSNK2A2'\n",
       "82. 'SYK'\n",
       "83. 'CSNK2A1'\n",
       "84. 'CHUK'\n",
       "85. 'PARP1'\n",
       "86. 'CSNK2A3'\n",
       "87. 'ZAP70'\n",
       "88. 'LCK'\n",
       "89. 'LTBR'\n",
       "90. 'XIAP'\n",
       "91. 'LTB'\n",
       "92. 'LTA'\n",
       "93. 'CD40LG'\n",
       "94. 'CCL13'\n",
       "95. 'TNFRSF13C'\n",
       "96. 'TNFSF13B'\n",
       "97. 'PTGS2'\n",
       "98. 'MALT1'\n",
       "99. 'IKBKB'\n",
       "100. 'TRIM25'\n",
       "101. 'CCL4'\n",
       "102. 'ERC1'\n",
       "103. 'CCL19'\n",
       "104. 'IKBKG'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`NF-kappa B signaling pathway`\n",
       "  [1] \"UBE2I\"     \"IL1R1\"     \"IGH\"       \"VCAM1\"     \"TICAM2\"    \"TICAM1\"   \n",
       "  [7] \"EDAR\"      \"CXCL12\"    \"RELA\"      \"TNF\"       \"RELB\"      \"MAP3K7\"   \n",
       " [13] \"CARD10\"    \"CARD11\"    \"PIAS4\"     \"EDARADD\"   \"DDX58\"     \"GADD45B\"  \n",
       " [19] \"GADD45A\"   \"TNFSF14\"   \"TRAF1\"     \"GADD45G\"   \"TIRAP\"     \"PRKCB\"    \n",
       " [25] \"TRAF3\"     \"TRAF2\"     \"NFKB1\"     \"NFKB2\"     \"MYD88\"     \"TRAF5\"    \n",
       " [31] \"CSNK2B\"    \"TRAF6\"     \"BCL2L1\"    \"BTK\"       \"TAB3\"      \"PRKCQ\"    \n",
       " [37] \"TAB2\"      \"MAP3K14\"   \"CARD14\"    \"TAB1\"      \"BIRC2\"     \"BIRC3\"    \n",
       " [43] \"CCL4L1\"    \"CCL4L2\"    \"BCL10\"     \"PLAU\"      \"EDA2R\"     \"ICAM1\"    \n",
       " [49] \"TNFSF11\"   \"PLCG1\"     \"PLCG2\"     \"CD14\"      \"LYN\"       \"CCL21\"    \n",
       " [55] \"CFLAR\"     \"IRAK4\"     \"TNFRSF1A\"  \"IL1B\"      \"NFKBIA\"    \"CYLD\"     \n",
       " [61] \"BCL2\"      \"ATM\"       \"TLR4\"      \"LAT\"       \"EDA\"       \"CXCL8\"    \n",
       " [67] \"CD40\"      \"TNFRSF11A\" \"BCL2A1\"    \"TRADD\"     \"TNFAIP3\"   \"LY96\"     \n",
       " [73] \"CXCL2\"     \"CXCL1\"     \"CXCL3\"     \"IRAK1\"     \"LBP\"       \"BLNK\"     \n",
       " [79] \"RIPK1\"     \"PIDD1\"     \"CSNK2A2\"   \"SYK\"       \"CSNK2A1\"   \"CHUK\"     \n",
       " [85] \"PARP1\"     \"CSNK2A3\"   \"ZAP70\"     \"LCK\"       \"LTBR\"      \"XIAP\"     \n",
       " [91] \"LTB\"       \"LTA\"       \"CD40LG\"    \"CCL13\"     \"TNFRSF13C\" \"TNFSF13B\" \n",
       " [97] \"PTGS2\"     \"MALT1\"     \"IKBKB\"     \"TRIM25\"    \"CCL4\"      \"ERC1\"     \n",
       "[103] \"CCL19\"     \"IKBKG\"    \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "p$kegg['NF-kappa B signaling pathway']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "deee48dd-8524-4a5e-97e3-ce3dc6c8daf9",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in `[.data.table`(out$Ex, \"leadingEdge\"): When i is a data.table (or character vector), the columns to join by must be specified using 'on=' argument (see ?data.table), by keying x (i.e. sorted, and, marked as sorted, see ?setkey), or by sharing column names between x and i (i.e., a natural join). Keyed joins might have further speed benefits on very large data due to x being sorted in RAM.\n",
     "output_type": "error",
     "traceback": [
      "Error in `[.data.table`(out$Ex, \"leadingEdge\"): When i is a data.table (or character vector), the columns to join by must be specified using 'on=' argument (see ?data.table), by keying x (i.e. sorted, and, marked as sorted, see ?setkey), or by sharing column names between x and i (i.e., a natural join). Keyed joins might have further speed benefits on very large data due to x being sorted in RAM.\nTraceback:\n",
      "1. out$Ex[\"leadingEdge\"]",
      "2. `[.data.table`(out$Ex, \"leadingEdge\")",
      "3. stop(\"When i is a data.table (or character vector), the columns to join by must be specified using 'on=' argument (see ?data.table), by keying x (i.e. sorted, and, marked as sorted, see ?setkey), or by sharing column names between x and i (i.e., a natural join). Keyed joins might have further speed benefits on very large data due to x being sorted in RAM.\")"
     ]
    }
   ],
   "source": [
    "out$Ex['leadingEdge']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36963b2f-ba68-481a-9c78-0bfbcebafca5",
   "metadata": {},
   "source": [
    "# below is scrap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ecaea937-b04b-46d2-90bc-d163170f24c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "pathways = readRDS('../ABCA7lof/processed_data/pathway_databases/pathways.rds')$all_paths\n",
    "o = get_gset_names_by_category(c('sterol', 'lipid', 'glycer', 'fatt', 'ceramide', 'phosphatidyl'), names(pathways))\n",
    "temp2 = pathways[o]\n",
    "out = lapply(names(scores), function(x) cbind(x, fgsea(pathways = temp2, stats = scores[[x]]$scores, minSize = 5, maxSize = 1000)))\n",
    "names(out) = names(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "d84b7b88-c0a5-4606-bbcf-63f1b9441e27",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = out$Ex\n",
    "res = res[order(res$pval,decreasing=FALSE),]\n",
    "res = res[res$pval<0.05,]\n",
    "saveRDS(res, './processed_data/for_plotting/fgsea_out_ex_lipid.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "6ab9b803-2634-45af-86bc-b7886b9e5f68",
   "metadata": {},
   "outputs": [],
   "source": [
    "# pathways = readRDS('../ABCA7lof/processed_data/pathway_databases/pathways.rds')\n",
    "# temp2 = pathways$kegg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5b025d68-efa5-4ab7-ab80-859e5ec92bc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = lapply(names(scores), function(x) cbind(x, fgsea(pathways = temp, stats = scores[[x]]$scores, minSize = 5, maxSize = 1000)))\n",
    "names(out) = names(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "852e1ba0-c922-45c4-872f-9c062a04c45b",
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(out, './processed_data/for_plotting/fgsea_out.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "62978586-fcfb-421b-be70-1255a79b9d45",
   "metadata": {},
   "outputs": [],
   "source": [
    "# res = do.call('rbind', out)\n",
    "# res$score = sign(res$NES) * -log10(res$pval)\n",
    "# res = res[res$x=='Ex',]\n",
    "# res = res[order(res$pval,decreasing=FALSE),]\n",
    "# res = res[res$pval<0.05,]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "548da68c-73ef-408a-80b1-11913384eccb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save all fgsea results\n",
    "df = as.data.frame(do.call('rbind', out))\n",
    "df = df[,!colnames(df)=='leadingEdge']\n",
    "write.csv(df, './processed_data/for_plotting/fgsea_out.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "fa17c9ca-e7da-49db-9937-05b507adad03",
   "metadata": {},
   "outputs": [],
   "source": [
    "d= read.csv('./processed_data/for_plotting/fgsea_out.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "32806f45-813e-4d6d-bf48-9bb4b01b8bcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 28 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X</th><th scope=col>x</th><th scope=col>pathway</th><th scope=col>pval</th><th scope=col>padj</th><th scope=col>log2err</th><th scope=col>ES</th><th scope=col>NES</th><th scope=col>size</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1070</th><td>1070</td><td>Ex</td><td>ATM Signaling Network in Development and Disease  WP3878                         </td><td>2.642628e-02</td><td>5.768063e-01</td><td>0.3524879</td><td> 0.4661167</td><td> 1.486559</td><td> 38</td></tr>\n",
       "\t<tr><th scope=row>1071</th><td>1071</td><td>Ex</td><td>ATM Signaling Pathway WP2516                                                     </td><td>4.708067e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.5113852</td><td> 1.489119</td><td> 25</td></tr>\n",
       "\t<tr><th scope=row>1107</th><td>1107</td><td>Ex</td><td>Cell Cycle WP179                                                                 </td><td>7.740895e-03</td><td>2.814871e-01</td><td>0.4070179</td><td> 0.4440157</td><td> 1.575000</td><td> 68</td></tr>\n",
       "\t<tr><th scope=row>1125</th><td>1125</td><td>Ex</td><td>Cytosine methylation WP3585                                                      </td><td>1.421194e-02</td><td>4.162774e-01</td><td>0.3807304</td><td>-0.7201311</td><td>-1.667474</td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>1127</th><td>1127</td><td>Ex</td><td>DNA Damage Response (only ATM dependent) WP710                                   </td><td>4.118616e-02</td><td>5.768063e-01</td><td>0.2878571</td><td> 0.3813560</td><td> 1.374809</td><td> 72</td></tr>\n",
       "\t<tr><th scope=row>1128</th><td>1128</td><td>Ex</td><td>DNA Damage Response WP707                                                        </td><td>2.540257e-02</td><td>5.768063e-01</td><td>0.3524879</td><td> 0.4703205</td><td> 1.499966</td><td> 38</td></tr>\n",
       "\t<tr><th scope=row>1130</th><td>1130</td><td>Ex</td><td>DNA IR-damage and cellular response via ATR WP4016                               </td><td>1.460523e-02</td><td>4.162774e-01</td><td>0.3807304</td><td> 0.4356262</td><td> 1.496880</td><td> 58</td></tr>\n",
       "\t<tr><th scope=row>1141</th><td>1141</td><td>Ex</td><td>Dual hijack model of Vif in HIV infection WP3300                                 </td><td>7.636293e-03</td><td>2.814871e-01</td><td>0.4070179</td><td> 0.7909029</td><td> 1.678167</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>1147</th><td>1147</td><td>Ex</td><td>Ebola Virus Pathway on Host WP4217                                               </td><td>3.839155e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.3879516</td><td> 1.398586</td><td> 72</td></tr>\n",
       "\t<tr><th scope=row>1150</th><td>1150</td><td>Ex</td><td>Electron Transport Chain (OXPHOS system in mitochondria) WP111                   </td><td>4.482974e-13</td><td>1.793189e-10</td><td>0.9325952</td><td> 0.6721279</td><td> 2.494630</td><td> 86</td></tr>\n",
       "\t<tr><th scope=row>1252</th><td>1252</td><td>Ex</td><td>Lipid Metabolism Pathway WP3965                                                  </td><td>1.561040e-02</td><td>4.162774e-01</td><td>0.3807304</td><td>-0.5368216</td><td>-1.636260</td><td> 24</td></tr>\n",
       "\t<tr><th scope=row>1260</th><td>1260</td><td>Ex</td><td>MET in type 1 papillary renal cell carcinoma WP4205                              </td><td>3.179925e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.4233929</td><td> 1.426231</td><td> 52</td></tr>\n",
       "\t<tr><th scope=row>1273</th><td>1273</td><td>Ex</td><td>Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292</td><td>3.594578e-02</td><td>5.768063e-01</td><td>0.3217759</td><td>-0.7617559</td><td>-1.534233</td><td>  6</td></tr>\n",
       "\t<tr><th scope=row>1280</th><td>1280</td><td>Ex</td><td>Mitochondrial complex I assembly model OXPHOS system WP4324                      </td><td>1.111494e-04</td><td>7.409963e-03</td><td>0.5384341</td><td> 0.5718491</td><td> 1.900620</td><td> 49</td></tr>\n",
       "\t<tr><th scope=row>1282</th><td>1282</td><td>Ex</td><td>Monoamine Transport WP727                                                        </td><td>2.923424e-02</td><td>5.768063e-01</td><td>0.3524879</td><td> 0.6466599</td><td> 1.610163</td><td> 13</td></tr>\n",
       "\t<tr><th scope=row>1302</th><td>1302</td><td>Ex</td><td>Nonalcoholic fatty liver disease WP4396                                          </td><td>2.701125e-06</td><td>2.701125e-04</td><td>0.6272567</td><td> 0.4950600</td><td> 1.928410</td><td>117</td></tr>\n",
       "\t<tr><th scope=row>1321</th><td>1321</td><td>Ex</td><td>Oxidative phosphorylation WP623                                                  </td><td>1.979099e-07</td><td>2.638799e-05</td><td>0.6901325</td><td> 0.6648928</td><td> 2.218668</td><td> 50</td></tr>\n",
       "\t<tr><th scope=row>1331</th><td>1331</td><td>Ex</td><td>Parkin-Ubiquitin Proteasomal System pathway WP2359                               </td><td>5.178647e-03</td><td>2.589323e-01</td><td>0.4070179</td><td> 0.4779280</td><td> 1.614828</td><td> 53</td></tr>\n",
       "\t<tr><th scope=row>1338</th><td>1338</td><td>Ex</td><td>Peptide GPCRs WP24                                                               </td><td>3.806502e-04</td><td>2.175144e-02</td><td>0.4984931</td><td>-0.8342118</td><td>-1.931630</td><td>  9</td></tr>\n",
       "\t<tr><th scope=row>1353</th><td>1353</td><td>Ex</td><td>Prostaglandin Synthesis and Regulation WP98                                      </td><td>6.442356e-03</td><td>2.814871e-01</td><td>0.4070179</td><td> 0.6764142</td><td> 1.718576</td><td> 14</td></tr>\n",
       "\t<tr><th scope=row>1354</th><td>1354</td><td>Ex</td><td>Proteasome Degradation WP183                                                     </td><td>1.098764e-07</td><td>2.197527e-05</td><td>0.7049757</td><td> 0.6655312</td><td> 2.218301</td><td> 48</td></tr>\n",
       "\t<tr><th scope=row>1369</th><td>1369</td><td>Ex</td><td>Retinoblastoma Gene in Cancer WP2446                                             </td><td>8.847954e-05</td><td>7.078363e-03</td><td>0.5384341</td><td> 0.5845668</td><td> 1.950630</td><td> 50</td></tr>\n",
       "\t<tr><th scope=row>1370</th><td>1370</td><td>Ex</td><td>Rett syndrome causing genes WP4312                                               </td><td>4.814160e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.4406628</td><td> 1.433175</td><td> 42</td></tr>\n",
       "\t<tr><th scope=row>1384</th><td>1384</td><td>Ex</td><td>Signaling of Hepatocyte Growth Factor Receptor WP313                             </td><td>3.533015e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.4882262</td><td> 1.455229</td><td> 28</td></tr>\n",
       "\t<tr><th scope=row>1404</th><td>1404</td><td>Ex</td><td>TCA Cycle (aka Krebs or citric acid cycle) WP78                                  </td><td>1.147722e-02</td><td>3.825740e-01</td><td>0.3807304</td><td> 0.6318732</td><td> 1.684476</td><td> 17</td></tr>\n",
       "\t<tr><th scope=row>1423</th><td>1423</td><td>Ex</td><td>Trans-sulfuration and one carbon metabolism WP2525                               </td><td>3.689719e-02</td><td>5.768063e-01</td><td>0.3217759</td><td>-0.5297412</td><td>-1.536371</td><td> 20</td></tr>\n",
       "\t<tr><th scope=row>1424</th><td>1424</td><td>Ex</td><td>Trans-sulfuration pathway WP2333                                                 </td><td>2.807165e-02</td><td>5.768063e-01</td><td>0.3524879</td><td>-0.7331603</td><td>-1.540983</td><td>  7</td></tr>\n",
       "\t<tr><th scope=row>1457</th><td>1457</td><td>Ex</td><td>miRNA Regulation of DNA Damage Response WP1530                                   </td><td>4.820451e-02</td><td>5.768063e-01</td><td>0.3217759</td><td> 0.4510785</td><td> 1.443481</td><td> 39</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 28 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & X & x & pathway & pval & padj & log2err & ES & NES & size\\\\\n",
       "  & <int> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int>\\\\\n",
       "\\hline\n",
       "\t1070 & 1070 & Ex & ATM Signaling Network in Development and Disease  WP3878                          & 2.642628e-02 & 5.768063e-01 & 0.3524879 &  0.4661167 &  1.486559 &  38\\\\\n",
       "\t1071 & 1071 & Ex & ATM Signaling Pathway WP2516                                                      & 4.708067e-02 & 5.768063e-01 & 0.3217759 &  0.5113852 &  1.489119 &  25\\\\\n",
       "\t1107 & 1107 & Ex & Cell Cycle WP179                                                                  & 7.740895e-03 & 2.814871e-01 & 0.4070179 &  0.4440157 &  1.575000 &  68\\\\\n",
       "\t1125 & 1125 & Ex & Cytosine methylation WP3585                                                       & 1.421194e-02 & 4.162774e-01 & 0.3807304 & -0.7201311 & -1.667474 &   9\\\\\n",
       "\t1127 & 1127 & Ex & DNA Damage Response (only ATM dependent) WP710                                    & 4.118616e-02 & 5.768063e-01 & 0.2878571 &  0.3813560 &  1.374809 &  72\\\\\n",
       "\t1128 & 1128 & Ex & DNA Damage Response WP707                                                         & 2.540257e-02 & 5.768063e-01 & 0.3524879 &  0.4703205 &  1.499966 &  38\\\\\n",
       "\t1130 & 1130 & Ex & DNA IR-damage and cellular response via ATR WP4016                                & 1.460523e-02 & 4.162774e-01 & 0.3807304 &  0.4356262 &  1.496880 &  58\\\\\n",
       "\t1141 & 1141 & Ex & Dual hijack model of Vif in HIV infection WP3300                                  & 7.636293e-03 & 2.814871e-01 & 0.4070179 &  0.7909029 &  1.678167 &   7\\\\\n",
       "\t1147 & 1147 & Ex & Ebola Virus Pathway on Host WP4217                                                & 3.839155e-02 & 5.768063e-01 & 0.3217759 &  0.3879516 &  1.398586 &  72\\\\\n",
       "\t1150 & 1150 & Ex & Electron Transport Chain (OXPHOS system in mitochondria) WP111                    & 4.482974e-13 & 1.793189e-10 & 0.9325952 &  0.6721279 &  2.494630 &  86\\\\\n",
       "\t1252 & 1252 & Ex & Lipid Metabolism Pathway WP3965                                                   & 1.561040e-02 & 4.162774e-01 & 0.3807304 & -0.5368216 & -1.636260 &  24\\\\\n",
       "\t1260 & 1260 & Ex & MET in type 1 papillary renal cell carcinoma WP4205                               & 3.179925e-02 & 5.768063e-01 & 0.3217759 &  0.4233929 &  1.426231 &  52\\\\\n",
       "\t1273 & 1273 & Ex & Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292 & 3.594578e-02 & 5.768063e-01 & 0.3217759 & -0.7617559 & -1.534233 &   6\\\\\n",
       "\t1280 & 1280 & Ex & Mitochondrial complex I assembly model OXPHOS system WP4324                       & 1.111494e-04 & 7.409963e-03 & 0.5384341 &  0.5718491 &  1.900620 &  49\\\\\n",
       "\t1282 & 1282 & Ex & Monoamine Transport WP727                                                         & 2.923424e-02 & 5.768063e-01 & 0.3524879 &  0.6466599 &  1.610163 &  13\\\\\n",
       "\t1302 & 1302 & Ex & Nonalcoholic fatty liver disease WP4396                                           & 2.701125e-06 & 2.701125e-04 & 0.6272567 &  0.4950600 &  1.928410 & 117\\\\\n",
       "\t1321 & 1321 & Ex & Oxidative phosphorylation WP623                                                   & 1.979099e-07 & 2.638799e-05 & 0.6901325 &  0.6648928 &  2.218668 &  50\\\\\n",
       "\t1331 & 1331 & Ex & Parkin-Ubiquitin Proteasomal System pathway WP2359                                & 5.178647e-03 & 2.589323e-01 & 0.4070179 &  0.4779280 &  1.614828 &  53\\\\\n",
       "\t1338 & 1338 & Ex & Peptide GPCRs WP24                                                                & 3.806502e-04 & 2.175144e-02 & 0.4984931 & -0.8342118 & -1.931630 &   9\\\\\n",
       "\t1353 & 1353 & Ex & Prostaglandin Synthesis and Regulation WP98                                       & 6.442356e-03 & 2.814871e-01 & 0.4070179 &  0.6764142 &  1.718576 &  14\\\\\n",
       "\t1354 & 1354 & Ex & Proteasome Degradation WP183                                                      & 1.098764e-07 & 2.197527e-05 & 0.7049757 &  0.6655312 &  2.218301 &  48\\\\\n",
       "\t1369 & 1369 & Ex & Retinoblastoma Gene in Cancer WP2446                                              & 8.847954e-05 & 7.078363e-03 & 0.5384341 &  0.5845668 &  1.950630 &  50\\\\\n",
       "\t1370 & 1370 & Ex & Rett syndrome causing genes WP4312                                                & 4.814160e-02 & 5.768063e-01 & 0.3217759 &  0.4406628 &  1.433175 &  42\\\\\n",
       "\t1384 & 1384 & Ex & Signaling of Hepatocyte Growth Factor Receptor WP313                              & 3.533015e-02 & 5.768063e-01 & 0.3217759 &  0.4882262 &  1.455229 &  28\\\\\n",
       "\t1404 & 1404 & Ex & TCA Cycle (aka Krebs or citric acid cycle) WP78                                   & 1.147722e-02 & 3.825740e-01 & 0.3807304 &  0.6318732 &  1.684476 &  17\\\\\n",
       "\t1423 & 1423 & Ex & Trans-sulfuration and one carbon metabolism WP2525                                & 3.689719e-02 & 5.768063e-01 & 0.3217759 & -0.5297412 & -1.536371 &  20\\\\\n",
       "\t1424 & 1424 & Ex & Trans-sulfuration pathway WP2333                                                  & 2.807165e-02 & 5.768063e-01 & 0.3524879 & -0.7331603 & -1.540983 &   7\\\\\n",
       "\t1457 & 1457 & Ex & miRNA Regulation of DNA Damage Response WP1530                                    & 4.820451e-02 & 5.768063e-01 & 0.3217759 &  0.4510785 &  1.443481 &  39\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 28 × 9\n",
       "\n",
       "| <!--/--> | X &lt;int&gt; | x &lt;chr&gt; | pathway &lt;chr&gt; | pval &lt;dbl&gt; | padj &lt;dbl&gt; | log2err &lt;dbl&gt; | ES &lt;dbl&gt; | NES &lt;dbl&gt; | size &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1070 | 1070 | Ex | ATM Signaling Network in Development and Disease  WP3878                          | 2.642628e-02 | 5.768063e-01 | 0.3524879 |  0.4661167 |  1.486559 |  38 |\n",
       "| 1071 | 1071 | Ex | ATM Signaling Pathway WP2516                                                      | 4.708067e-02 | 5.768063e-01 | 0.3217759 |  0.5113852 |  1.489119 |  25 |\n",
       "| 1107 | 1107 | Ex | Cell Cycle WP179                                                                  | 7.740895e-03 | 2.814871e-01 | 0.4070179 |  0.4440157 |  1.575000 |  68 |\n",
       "| 1125 | 1125 | Ex | Cytosine methylation WP3585                                                       | 1.421194e-02 | 4.162774e-01 | 0.3807304 | -0.7201311 | -1.667474 |   9 |\n",
       "| 1127 | 1127 | Ex | DNA Damage Response (only ATM dependent) WP710                                    | 4.118616e-02 | 5.768063e-01 | 0.2878571 |  0.3813560 |  1.374809 |  72 |\n",
       "| 1128 | 1128 | Ex | DNA Damage Response WP707                                                         | 2.540257e-02 | 5.768063e-01 | 0.3524879 |  0.4703205 |  1.499966 |  38 |\n",
       "| 1130 | 1130 | Ex | DNA IR-damage and cellular response via ATR WP4016                                | 1.460523e-02 | 4.162774e-01 | 0.3807304 |  0.4356262 |  1.496880 |  58 |\n",
       "| 1141 | 1141 | Ex | Dual hijack model of Vif in HIV infection WP3300                                  | 7.636293e-03 | 2.814871e-01 | 0.4070179 |  0.7909029 |  1.678167 |   7 |\n",
       "| 1147 | 1147 | Ex | Ebola Virus Pathway on Host WP4217                                                | 3.839155e-02 | 5.768063e-01 | 0.3217759 |  0.3879516 |  1.398586 |  72 |\n",
       "| 1150 | 1150 | Ex | Electron Transport Chain (OXPHOS system in mitochondria) WP111                    | 4.482974e-13 | 1.793189e-10 | 0.9325952 |  0.6721279 |  2.494630 |  86 |\n",
       "| 1252 | 1252 | Ex | Lipid Metabolism Pathway WP3965                                                   | 1.561040e-02 | 4.162774e-01 | 0.3807304 | -0.5368216 | -1.636260 |  24 |\n",
       "| 1260 | 1260 | Ex | MET in type 1 papillary renal cell carcinoma WP4205                               | 3.179925e-02 | 5.768063e-01 | 0.3217759 |  0.4233929 |  1.426231 |  52 |\n",
       "| 1273 | 1273 | Ex | Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292 | 3.594578e-02 | 5.768063e-01 | 0.3217759 | -0.7617559 | -1.534233 |   6 |\n",
       "| 1280 | 1280 | Ex | Mitochondrial complex I assembly model OXPHOS system WP4324                       | 1.111494e-04 | 7.409963e-03 | 0.5384341 |  0.5718491 |  1.900620 |  49 |\n",
       "| 1282 | 1282 | Ex | Monoamine Transport WP727                                                         | 2.923424e-02 | 5.768063e-01 | 0.3524879 |  0.6466599 |  1.610163 |  13 |\n",
       "| 1302 | 1302 | Ex | Nonalcoholic fatty liver disease WP4396                                           | 2.701125e-06 | 2.701125e-04 | 0.6272567 |  0.4950600 |  1.928410 | 117 |\n",
       "| 1321 | 1321 | Ex | Oxidative phosphorylation WP623                                                   | 1.979099e-07 | 2.638799e-05 | 0.6901325 |  0.6648928 |  2.218668 |  50 |\n",
       "| 1331 | 1331 | Ex | Parkin-Ubiquitin Proteasomal System pathway WP2359                                | 5.178647e-03 | 2.589323e-01 | 0.4070179 |  0.4779280 |  1.614828 |  53 |\n",
       "| 1338 | 1338 | Ex | Peptide GPCRs WP24                                                                | 3.806502e-04 | 2.175144e-02 | 0.4984931 | -0.8342118 | -1.931630 |   9 |\n",
       "| 1353 | 1353 | Ex | Prostaglandin Synthesis and Regulation WP98                                       | 6.442356e-03 | 2.814871e-01 | 0.4070179 |  0.6764142 |  1.718576 |  14 |\n",
       "| 1354 | 1354 | Ex | Proteasome Degradation WP183                                                      | 1.098764e-07 | 2.197527e-05 | 0.7049757 |  0.6655312 |  2.218301 |  48 |\n",
       "| 1369 | 1369 | Ex | Retinoblastoma Gene in Cancer WP2446                                              | 8.847954e-05 | 7.078363e-03 | 0.5384341 |  0.5845668 |  1.950630 |  50 |\n",
       "| 1370 | 1370 | Ex | Rett syndrome causing genes WP4312                                                | 4.814160e-02 | 5.768063e-01 | 0.3217759 |  0.4406628 |  1.433175 |  42 |\n",
       "| 1384 | 1384 | Ex | Signaling of Hepatocyte Growth Factor Receptor WP313                              | 3.533015e-02 | 5.768063e-01 | 0.3217759 |  0.4882262 |  1.455229 |  28 |\n",
       "| 1404 | 1404 | Ex | TCA Cycle (aka Krebs or citric acid cycle) WP78                                   | 1.147722e-02 | 3.825740e-01 | 0.3807304 |  0.6318732 |  1.684476 |  17 |\n",
       "| 1423 | 1423 | Ex | Trans-sulfuration and one carbon metabolism WP2525                                | 3.689719e-02 | 5.768063e-01 | 0.3217759 | -0.5297412 | -1.536371 |  20 |\n",
       "| 1424 | 1424 | Ex | Trans-sulfuration pathway WP2333                                                  | 2.807165e-02 | 5.768063e-01 | 0.3524879 | -0.7331603 | -1.540983 |   7 |\n",
       "| 1457 | 1457 | Ex | miRNA Regulation of DNA Damage Response WP1530                                    | 4.820451e-02 | 5.768063e-01 | 0.3217759 |  0.4510785 |  1.443481 |  39 |\n",
       "\n"
      ],
      "text/plain": [
       "     X    x \n",
       "1070 1070 Ex\n",
       "1071 1071 Ex\n",
       "1107 1107 Ex\n",
       "1125 1125 Ex\n",
       "1127 1127 Ex\n",
       "1128 1128 Ex\n",
       "1130 1130 Ex\n",
       "1141 1141 Ex\n",
       "1147 1147 Ex\n",
       "1150 1150 Ex\n",
       "1252 1252 Ex\n",
       "1260 1260 Ex\n",
       "1273 1273 Ex\n",
       "1280 1280 Ex\n",
       "1282 1282 Ex\n",
       "1302 1302 Ex\n",
       "1321 1321 Ex\n",
       "1331 1331 Ex\n",
       "1338 1338 Ex\n",
       "1353 1353 Ex\n",
       "1354 1354 Ex\n",
       "1369 1369 Ex\n",
       "1370 1370 Ex\n",
       "1384 1384 Ex\n",
       "1404 1404 Ex\n",
       "1423 1423 Ex\n",
       "1424 1424 Ex\n",
       "1457 1457 Ex\n",
       "     pathway                                                                          \n",
       "1070 ATM Signaling Network in Development and Disease  WP3878                         \n",
       "1071 ATM Signaling Pathway WP2516                                                     \n",
       "1107 Cell Cycle WP179                                                                 \n",
       "1125 Cytosine methylation WP3585                                                      \n",
       "1127 DNA Damage Response (only ATM dependent) WP710                                   \n",
       "1128 DNA Damage Response WP707                                                        \n",
       "1130 DNA IR-damage and cellular response via ATR WP4016                               \n",
       "1141 Dual hijack model of Vif in HIV infection WP3300                                 \n",
       "1147 Ebola Virus Pathway on Host WP4217                                               \n",
       "1150 Electron Transport Chain (OXPHOS system in mitochondria) WP111                   \n",
       "1252 Lipid Metabolism Pathway WP3965                                                  \n",
       "1260 MET in type 1 papillary renal cell carcinoma WP4205                              \n",
       "1273 Methionine metabolism leading to Sulphur Amino Acids and related disorders WP4292\n",
       "1280 Mitochondrial complex I assembly model OXPHOS system WP4324                      \n",
       "1282 Monoamine Transport WP727                                                        \n",
       "1302 Nonalcoholic fatty liver disease WP4396                                          \n",
       "1321 Oxidative phosphorylation WP623                                                  \n",
       "1331 Parkin-Ubiquitin Proteasomal System pathway WP2359                               \n",
       "1338 Peptide GPCRs WP24                                                               \n",
       "1353 Prostaglandin Synthesis and Regulation WP98                                      \n",
       "1354 Proteasome Degradation WP183                                                     \n",
       "1369 Retinoblastoma Gene in Cancer WP2446                                             \n",
       "1370 Rett syndrome causing genes WP4312                                               \n",
       "1384 Signaling of Hepatocyte Growth Factor Receptor WP313                             \n",
       "1404 TCA Cycle (aka Krebs or citric acid cycle) WP78                                  \n",
       "1423 Trans-sulfuration and one carbon metabolism WP2525                               \n",
       "1424 Trans-sulfuration pathway WP2333                                                 \n",
       "1457 miRNA Regulation of DNA Damage Response WP1530                                   \n",
       "     pval         padj         log2err   ES         NES       size\n",
       "1070 2.642628e-02 5.768063e-01 0.3524879  0.4661167  1.486559  38 \n",
       "1071 4.708067e-02 5.768063e-01 0.3217759  0.5113852  1.489119  25 \n",
       "1107 7.740895e-03 2.814871e-01 0.4070179  0.4440157  1.575000  68 \n",
       "1125 1.421194e-02 4.162774e-01 0.3807304 -0.7201311 -1.667474   9 \n",
       "1127 4.118616e-02 5.768063e-01 0.2878571  0.3813560  1.374809  72 \n",
       "1128 2.540257e-02 5.768063e-01 0.3524879  0.4703205  1.499966  38 \n",
       "1130 1.460523e-02 4.162774e-01 0.3807304  0.4356262  1.496880  58 \n",
       "1141 7.636293e-03 2.814871e-01 0.4070179  0.7909029  1.678167   7 \n",
       "1147 3.839155e-02 5.768063e-01 0.3217759  0.3879516  1.398586  72 \n",
       "1150 4.482974e-13 1.793189e-10 0.9325952  0.6721279  2.494630  86 \n",
       "1252 1.561040e-02 4.162774e-01 0.3807304 -0.5368216 -1.636260  24 \n",
       "1260 3.179925e-02 5.768063e-01 0.3217759  0.4233929  1.426231  52 \n",
       "1273 3.594578e-02 5.768063e-01 0.3217759 -0.7617559 -1.534233   6 \n",
       "1280 1.111494e-04 7.409963e-03 0.5384341  0.5718491  1.900620  49 \n",
       "1282 2.923424e-02 5.768063e-01 0.3524879  0.6466599  1.610163  13 \n",
       "1302 2.701125e-06 2.701125e-04 0.6272567  0.4950600  1.928410 117 \n",
       "1321 1.979099e-07 2.638799e-05 0.6901325  0.6648928  2.218668  50 \n",
       "1331 5.178647e-03 2.589323e-01 0.4070179  0.4779280  1.614828  53 \n",
       "1338 3.806502e-04 2.175144e-02 0.4984931 -0.8342118 -1.931630   9 \n",
       "1353 6.442356e-03 2.814871e-01 0.4070179  0.6764142  1.718576  14 \n",
       "1354 1.098764e-07 2.197527e-05 0.7049757  0.6655312  2.218301  48 \n",
       "1369 8.847954e-05 7.078363e-03 0.5384341  0.5845668  1.950630  50 \n",
       "1370 4.814160e-02 5.768063e-01 0.3217759  0.4406628  1.433175  42 \n",
       "1384 3.533015e-02 5.768063e-01 0.3217759  0.4882262  1.455229  28 \n",
       "1404 1.147722e-02 3.825740e-01 0.3807304  0.6318732  1.684476  17 \n",
       "1423 3.689719e-02 5.768063e-01 0.3217759 -0.5297412 -1.536371  20 \n",
       "1424 2.807165e-02 5.768063e-01 0.3524879 -0.7331603 -1.540983   7 \n",
       "1457 4.820451e-02 5.768063e-01 0.3217759  0.4510785  1.443481  39 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "d[d$x=='Ex' & d$pval<0.05,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "926062b3-f7b3-43b2-b789-4c02fd7a1cd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = do.call('rbind', out)\n",
    "res$score = sign(res$NES) * -log10(res$pval)\n",
    "res = res[order(res$pval,decreasing=FALSE),]\n",
    "res = res[res$pval<0.05,]\n",
    "# save all leading edge genes\n",
    "df = as.data.frame(unique(unname(unlist(res$leadingEdge))))\n",
    "colnames(df) = 'gene'\n",
    "write.csv(df, './processed_data/for_plotting/leading_edge.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b06e50ed-ade6-4426-8a2f-f4f846700737",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save Ex leading edge genes\n",
    "for(i in names(out)){\n",
    "    res = out[[i]]\n",
    "    res = res[res$pval<0.05,]\n",
    "\n",
    "    df = as.data.frame(unique(unname(unlist(res$leadingEdge))))\n",
    "    colnames(df) = 'gene'\n",
    "    write.csv(df, paste0(paste0('./processed_data/for_plotting/leading_edge_', i), '.csv'))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cfbdca7e-063c-4991-8507-43dfb2c0bfd0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 17 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>x</th><th scope=col>pathway</th><th scope=col>pval</th><th scope=col>padj</th><th scope=col>log2err</th><th scope=col>ES</th><th scope=col>NES</th><th scope=col>size</th><th scope=col>leadingEdge</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;list&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Opc</td><td>Aryl Hydrocarbon Receptor Pathway WP2873                             </td><td>0.002572431</td><td>0.6554441</td><td>0.4317077</td><td>-0.7207746</td><td>-1.837274</td><td>14</td><td>PTGES3, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337</td><td>0.005053618</td><td>0.6554441</td><td>0.4070179</td><td> 0.8148782</td><td> 1.682967</td><td> 7</td><td>IL6ST, J....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Genes related to primary cilium development (based on CRISPR) WP4536 </td><td>0.005215735</td><td>0.6554441</td><td>0.4070179</td><td>-0.4228744</td><td>-1.574863</td><td>72</td><td>CEP162, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Calcium Regulation in the Cardiac Cell WP536                         </td><td>0.010014804</td><td>0.8365105</td><td>0.3807304</td><td>-0.4167157</td><td>-1.562717</td><td>77</td><td>SLC8A3, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>NLR Proteins WP288                                                   </td><td>0.011906788</td><td>0.8365105</td><td>0.3807304</td><td> 0.8512859</td><td> 1.605635</td><td> 5</td><td>MAP3K7, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>The human immune response to tuberculosis WP4197                     </td><td>0.013313164</td><td>0.8365105</td><td>0.3807304</td><td> 0.6736173</td><td> 1.713088</td><td>13</td><td>IFNAR1, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Iron metabolism in placenta WP2007                                   </td><td>0.017136413</td><td>0.9006333</td><td>0.3524879</td><td> 0.7922463</td><td> 1.579650</td><td> 6</td><td>    FTH1, TF</td></tr>\n",
       "\t<tr><td>Opc</td><td>Cholesterol Biosynthesis Pathway WP197                               </td><td>0.028558719</td><td>0.9006333</td><td>0.3524879</td><td> 0.6690772</td><td> 1.620472</td><td>11</td><td>IDI1, MS....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Type III interferon signaling WP2113                                 </td><td>0.028759138</td><td>0.9006333</td><td>0.3524879</td><td> 0.7700633</td><td> 1.535420</td><td> 6</td><td>IL10RB, JAK1</td></tr>\n",
       "\t<tr><td>Opc</td><td>Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203         </td><td>0.029701420</td><td>0.9006333</td><td>0.3524879</td><td> 0.5435175</td><td> 1.589700</td><td>22</td><td>NFKBIA, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Cytosine methylation WP3585                                          </td><td>0.030497993</td><td>0.9006333</td><td>0.3524879</td><td>-0.7210000</td><td>-1.603557</td><td> 9</td><td>IDH1, TE....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Disorders of the Krebs cycle WP4236                                  </td><td>0.031572976</td><td>0.9006333</td><td>0.3217759</td><td>-0.7642704</td><td>-1.505662</td><td> 6</td><td>SUCLG1, ....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Integrated Cancer Pathway WP1971                                     </td><td>0.033164890</td><td>0.9006333</td><td>0.3217759</td><td> 0.5312923</td><td> 1.553944</td><td>22</td><td>BARD1, J....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Notch Signaling Pathway WP61                                         </td><td>0.033445268</td><td>0.9006333</td><td>0.3217759</td><td> 0.4424579</td><td> 1.493911</td><td>42</td><td>NUMBL, C....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Fluoropyrimidine Activity WP1601                                     </td><td>0.038846975</td><td>0.9301887</td><td>0.3217759</td><td>-0.7054697</td><td>-1.539483</td><td> 8</td><td>ABCC5, M....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Pyrimidine metabolism WP4022                                         </td><td>0.039772712</td><td>0.9301887</td><td>0.3217759</td><td>-0.4916573</td><td>-1.473225</td><td>25</td><td>UCKL1, N....</td></tr>\n",
       "\t<tr><td>Opc</td><td>Selenium Micronutrient Network WP15                                  </td><td>0.049019608</td><td>0.9301887</td><td>0.2878571</td><td>-0.4764195</td><td>-1.448909</td><td>28</td><td>SELENOP,....</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 17 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " x & pathway & pval & padj & log2err & ES & NES & size & leadingEdge\\\\\n",
       " <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <list>\\\\\n",
       "\\hline\n",
       "\t Opc & Aryl Hydrocarbon Receptor Pathway WP2873                              & 0.002572431 & 0.6554441 & 0.4317077 & -0.7207746 & -1.837274 & 14 & PTGES3, ....\\\\\n",
       "\t Opc & ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337 & 0.005053618 & 0.6554441 & 0.4070179 &  0.8148782 &  1.682967 &  7 & IL6ST, J....\\\\\n",
       "\t Opc & Genes related to primary cilium development (based on CRISPR) WP4536  & 0.005215735 & 0.6554441 & 0.4070179 & -0.4228744 & -1.574863 & 72 & CEP162, ....\\\\\n",
       "\t Opc & Calcium Regulation in the Cardiac Cell WP536                          & 0.010014804 & 0.8365105 & 0.3807304 & -0.4167157 & -1.562717 & 77 & SLC8A3, ....\\\\\n",
       "\t Opc & NLR Proteins WP288                                                    & 0.011906788 & 0.8365105 & 0.3807304 &  0.8512859 &  1.605635 &  5 & MAP3K7, ....\\\\\n",
       "\t Opc & The human immune response to tuberculosis WP4197                      & 0.013313164 & 0.8365105 & 0.3807304 &  0.6736173 &  1.713088 & 13 & IFNAR1, ....\\\\\n",
       "\t Opc & Iron metabolism in placenta WP2007                                    & 0.017136413 & 0.9006333 & 0.3524879 &  0.7922463 &  1.579650 &  6 &     FTH1, TF\\\\\n",
       "\t Opc & Cholesterol Biosynthesis Pathway WP197                                & 0.028558719 & 0.9006333 & 0.3524879 &  0.6690772 &  1.620472 & 11 & IDI1, MS....\\\\\n",
       "\t Opc & Type III interferon signaling WP2113                                  & 0.028759138 & 0.9006333 & 0.3524879 &  0.7700633 &  1.535420 &  6 & IL10RB, JAK1\\\\\n",
       "\t Opc & Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203          & 0.029701420 & 0.9006333 & 0.3524879 &  0.5435175 &  1.589700 & 22 & NFKBIA, ....\\\\\n",
       "\t Opc & Cytosine methylation WP3585                                           & 0.030497993 & 0.9006333 & 0.3524879 & -0.7210000 & -1.603557 &  9 & IDH1, TE....\\\\\n",
       "\t Opc & Disorders of the Krebs cycle WP4236                                   & 0.031572976 & 0.9006333 & 0.3217759 & -0.7642704 & -1.505662 &  6 & SUCLG1, ....\\\\\n",
       "\t Opc & Integrated Cancer Pathway WP1971                                      & 0.033164890 & 0.9006333 & 0.3217759 &  0.5312923 &  1.553944 & 22 & BARD1, J....\\\\\n",
       "\t Opc & Notch Signaling Pathway WP61                                          & 0.033445268 & 0.9006333 & 0.3217759 &  0.4424579 &  1.493911 & 42 & NUMBL, C....\\\\\n",
       "\t Opc & Fluoropyrimidine Activity WP1601                                      & 0.038846975 & 0.9301887 & 0.3217759 & -0.7054697 & -1.539483 &  8 & ABCC5, M....\\\\\n",
       "\t Opc & Pyrimidine metabolism WP4022                                          & 0.039772712 & 0.9301887 & 0.3217759 & -0.4916573 & -1.473225 & 25 & UCKL1, N....\\\\\n",
       "\t Opc & Selenium Micronutrient Network WP15                                   & 0.049019608 & 0.9301887 & 0.2878571 & -0.4764195 & -1.448909 & 28 & SELENOP,....\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 17 × 9\n",
       "\n",
       "| x &lt;chr&gt; | pathway &lt;chr&gt; | pval &lt;dbl&gt; | padj &lt;dbl&gt; | log2err &lt;dbl&gt; | ES &lt;dbl&gt; | NES &lt;dbl&gt; | size &lt;int&gt; | leadingEdge &lt;list&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| Opc | Aryl Hydrocarbon Receptor Pathway WP2873                              | 0.002572431 | 0.6554441 | 0.4317077 | -0.7207746 | -1.837274 | 14 | PTGES3, .... |\n",
       "| Opc | ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337 | 0.005053618 | 0.6554441 | 0.4070179 |  0.8148782 |  1.682967 |  7 | IL6ST, J.... |\n",
       "| Opc | Genes related to primary cilium development (based on CRISPR) WP4536  | 0.005215735 | 0.6554441 | 0.4070179 | -0.4228744 | -1.574863 | 72 | CEP162, .... |\n",
       "| Opc | Calcium Regulation in the Cardiac Cell WP536                          | 0.010014804 | 0.8365105 | 0.3807304 | -0.4167157 | -1.562717 | 77 | SLC8A3, .... |\n",
       "| Opc | NLR Proteins WP288                                                    | 0.011906788 | 0.8365105 | 0.3807304 |  0.8512859 |  1.605635 |  5 | MAP3K7, .... |\n",
       "| Opc | The human immune response to tuberculosis WP4197                      | 0.013313164 | 0.8365105 | 0.3807304 |  0.6736173 |  1.713088 | 13 | IFNAR1, .... |\n",
       "| Opc | Iron metabolism in placenta WP2007                                    | 0.017136413 | 0.9006333 | 0.3524879 |  0.7922463 |  1.579650 |  6 |     FTH1, TF |\n",
       "| Opc | Cholesterol Biosynthesis Pathway WP197                                | 0.028558719 | 0.9006333 | 0.3524879 |  0.6690772 |  1.620472 | 11 | IDI1, MS.... |\n",
       "| Opc | Type III interferon signaling WP2113                                  | 0.028759138 | 0.9006333 | 0.3524879 |  0.7700633 |  1.535420 |  6 | IL10RB, JAK1 |\n",
       "| Opc | Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203          | 0.029701420 | 0.9006333 | 0.3524879 |  0.5435175 |  1.589700 | 22 | NFKBIA, .... |\n",
       "| Opc | Cytosine methylation WP3585                                           | 0.030497993 | 0.9006333 | 0.3524879 | -0.7210000 | -1.603557 |  9 | IDH1, TE.... |\n",
       "| Opc | Disorders of the Krebs cycle WP4236                                   | 0.031572976 | 0.9006333 | 0.3217759 | -0.7642704 | -1.505662 |  6 | SUCLG1, .... |\n",
       "| Opc | Integrated Cancer Pathway WP1971                                      | 0.033164890 | 0.9006333 | 0.3217759 |  0.5312923 |  1.553944 | 22 | BARD1, J.... |\n",
       "| Opc | Notch Signaling Pathway WP61                                          | 0.033445268 | 0.9006333 | 0.3217759 |  0.4424579 |  1.493911 | 42 | NUMBL, C.... |\n",
       "| Opc | Fluoropyrimidine Activity WP1601                                      | 0.038846975 | 0.9301887 | 0.3217759 | -0.7054697 | -1.539483 |  8 | ABCC5, M.... |\n",
       "| Opc | Pyrimidine metabolism WP4022                                          | 0.039772712 | 0.9301887 | 0.3217759 | -0.4916573 | -1.473225 | 25 | UCKL1, N.... |\n",
       "| Opc | Selenium Micronutrient Network WP15                                   | 0.049019608 | 0.9301887 | 0.2878571 | -0.4764195 | -1.448909 | 28 | SELENOP,.... |\n",
       "\n"
      ],
      "text/plain": [
       "   x   pathway                                                              \n",
       "1  Opc Aryl Hydrocarbon Receptor Pathway WP2873                             \n",
       "2  Opc ncRNAs involved in STAT3 signaling in hepatocellular carcinoma WP4337\n",
       "3  Opc Genes related to primary cilium development (based on CRISPR) WP4536 \n",
       "4  Opc Calcium Regulation in the Cardiac Cell WP536                         \n",
       "5  Opc NLR Proteins WP288                                                   \n",
       "6  Opc The human immune response to tuberculosis WP4197                     \n",
       "7  Opc Iron metabolism in placenta WP2007                                   \n",
       "8  Opc Cholesterol Biosynthesis Pathway WP197                               \n",
       "9  Opc Type III interferon signaling WP2113                                 \n",
       "10 Opc Thymic Stromal LymphoPoietin (TSLP) Signaling Pathway WP2203         \n",
       "11 Opc Cytosine methylation WP3585                                          \n",
       "12 Opc Disorders of the Krebs cycle WP4236                                  \n",
       "13 Opc Integrated Cancer Pathway WP1971                                     \n",
       "14 Opc Notch Signaling Pathway WP61                                         \n",
       "15 Opc Fluoropyrimidine Activity WP1601                                     \n",
       "16 Opc Pyrimidine metabolism WP4022                                         \n",
       "17 Opc Selenium Micronutrient Network WP15                                  \n",
       "   pval        padj      log2err   ES         NES       size leadingEdge \n",
       "1  0.002572431 0.6554441 0.4317077 -0.7207746 -1.837274 14   PTGES3, ....\n",
       "2  0.005053618 0.6554441 0.4070179  0.8148782  1.682967  7   IL6ST, J....\n",
       "3  0.005215735 0.6554441 0.4070179 -0.4228744 -1.574863 72   CEP162, ....\n",
       "4  0.010014804 0.8365105 0.3807304 -0.4167157 -1.562717 77   SLC8A3, ....\n",
       "5  0.011906788 0.8365105 0.3807304  0.8512859  1.605635  5   MAP3K7, ....\n",
       "6  0.013313164 0.8365105 0.3807304  0.6736173  1.713088 13   IFNAR1, ....\n",
       "7  0.017136413 0.9006333 0.3524879  0.7922463  1.579650  6       FTH1, TF\n",
       "8  0.028558719 0.9006333 0.3524879  0.6690772  1.620472 11   IDI1, MS....\n",
       "9  0.028759138 0.9006333 0.3524879  0.7700633  1.535420  6   IL10RB, JAK1\n",
       "10 0.029701420 0.9006333 0.3524879  0.5435175  1.589700 22   NFKBIA, ....\n",
       "11 0.030497993 0.9006333 0.3524879 -0.7210000 -1.603557  9   IDH1, TE....\n",
       "12 0.031572976 0.9006333 0.3217759 -0.7642704 -1.505662  6   SUCLG1, ....\n",
       "13 0.033164890 0.9006333 0.3217759  0.5312923  1.553944 22   BARD1, J....\n",
       "14 0.033445268 0.9006333 0.3217759  0.4424579  1.493911 42   NUMBL, C....\n",
       "15 0.038846975 0.9301887 0.3217759 -0.7054697 -1.539483  8   ABCC5, M....\n",
       "16 0.039772712 0.9301887 0.3217759 -0.4916573 -1.473225 25   UCKL1, N....\n",
       "17 0.049019608 0.9301887 0.2878571 -0.4764195 -1.448909 28   SELENOP,...."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "temp = out[['Opc']]\n",
    "temp = temp[order(temp$pval,decreasing=FALSE),]\n",
    "temp[temp$pval<0.05,]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r_env]",
   "language": "R",
   "name": "conda-env-.conda-r_env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
