{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5a3893bd-e8ff-4798-8a75-b623c0f4fd9d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(patchwork)\n",
    "library(ggplot2)\n",
    "library(magick)\n",
    "library(tidytext)\n",
    "library(SingleCellExperiment)\n",
    "library(tidyHeatmap)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(plotgardener)\n",
    "library(ggplot2)\n",
    "library(ggrepel)\n",
    "library(gridExtra)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ae30089-f85d-480d-9b29-ff1f6349bb5c",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 1**</u>\n",
    "##### Cohort metadata overview\n",
    "<a name=\"EF1\"></a>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7175ec8d-2d8c-456d-b193-a937202533c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# proteomics data\n",
    "proteomics = read.csv('.//processed_data//for_plotting/ABCA7_proteomics.csv')\n",
    "# metadata\n",
    "all_data = readRDS('./processed_data/single_cell_data/analysis_input_data.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df43bc42-7b1c-4adb-a32e-ac3f8d19abaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "proteomics$LOF = ifelse(proteomics$LOF=='LOF', 'LoF', 'Con')\n",
    "proteomics$LOF = factor(proteomics$LOF, levels = c('Con', 'LoF'))\n",
    "\n",
    "p000 = boxplot_w_stats(df = proteomics, x = 'LOF', y = 'ABCA7', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'ABCA7', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('All')\n",
    "p001 = boxplot_w_stats(df = proteomics, x = 'LOF', y = 'RBFOX3', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'RBFOX3', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('All')\n",
    "p002 = boxplot_w_stats(df = proteomics[proteomics$grp==TRUE,], x = 'LOF', y = 'ABCA7', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'ABCA7', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('Subset')\n",
    "p003 = boxplot_w_stats(df = proteomics[proteomics$grp==TRUE,], x = 'LOF', y = 'RBFOX3', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'RBFOX3', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('Subset')\n",
    "p004 = get_permutation_plot('RBFOX3', proteomics)\n",
    "p005 = get_permutation_plot('ABCA7', proteomics)\n",
    "\n",
    "proteomics$norm_ABCA7 = proteomics$ABCA7-proteomics$RBFOX3\n",
    "pi = boxplot_w_stats(df = proteomics, x = 'LOF', y = 'norm_ABCA7', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'ABCA7-RBFOX3', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('All')\n",
    "pii = boxplot_w_stats(df = proteomics[proteomics$grp==TRUE,], x = 'LOF', y = 'norm_ABCA7', palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = 'ABCA7-RBFOX3', width = .5, alpha = .5)+ theme(legend.position=\"none\")+ggtitle('Subset')\n",
    "\n",
    "plot_grp0 = (((p000|pi|p001)/(p002|pii|p003))|(p004/p005))+plot_layout(widths = c(3,1))\n",
    "\n",
    "# load the data\n",
    "features = c('cogn_global_lv', 'amyloid', 'tangles','age_death', 'pmi')\n",
    "\n",
    "# continuous variables by variant\n",
    "metadata = all_data$summary\n",
    "metadata$LOF = ifelse(metadata$LOF==1, 'LoF', 'Con')\n",
    "metadata$LOF = factor(metadata$LOF, levels = c('Con', 'LoF'))\n",
    "\n",
    "plt00 = boxplot_w_stats(df = as.data.frame(metadata), x = 'LOF', y = features[1], palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = features[1], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt01 = boxplot_w_stats(df = as.data.frame(metadata), x = 'LOF', y = features[2], palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = features[2], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt02 = boxplot_w_stats(df = as.data.frame(metadata), x = 'LOF', y = features[3], palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = features[3], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt03 = boxplot_w_stats(df = as.data.frame(metadata), x = 'LOF', y = features[4], palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = features[4], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt04 = boxplot_w_stats(df = as.data.frame(metadata), x = 'LOF', y = features[5], palette = cols, comparisons = list(c('Con', 'LoF')), xlab = '', ylab = features[5], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "\n",
    "# categorical variables by genotype\n",
    "features = c('braaksc', 'ceradsc', 'msex','apoe_genotype', 'cogdx')\n",
    "plt10 = get_barplot(as.data.frame(metadata), 'LOF', features[1])\n",
    "plt11 = get_barplot(as.data.frame(metadata), 'LOF', features[2])\n",
    "plt12 = get_barplot(as.data.frame(metadata), 'LOF', features[3])\n",
    "plt13 = get_barplot(as.data.frame(metadata), 'LOF', features[4])\n",
    "plt14 = get_barplot(as.data.frame(metadata), 'LOF', features[5])\n",
    "\n",
    "cont_plots = (plt00|plt01|plt02|plt03|plt04)\n",
    "disc_plots = (plt10|plt11|plt12|plt13|plt14)\n",
    "\n",
    "# get the images\n",
    "img = image_read('./processed_data/for_plotting/Asset 1.png')#, native = TRUE)\n",
    "p1 = image_ggplot(image_border((img), \"grey\", \"4x4\"))+ggtitle('rs547447016 (c.2126_2132delACGAGGA)')+theme(plot.title = element_text(color = \"grey\"))\n",
    "p2 = image_ggplot(image_border((img), \"red\", \"4x4\"))+ggtitle('rs547447016 (c.2126_2132delACGAGGA)')+theme(plot.title = element_text(color = \"red\"))\n",
    "p3 = image_ggplot(image_border((img), \"grey\", \"4x4\"))+ggtitle('rs201060968 (c.3641G>A)')+theme(plot.title = element_text(color = \"grey\"))\n",
    "p4 = image_ggplot(image_border((img), \"red\", \"4x4\"))+ggtitle('rs201060968 (c.3641G>A)')+theme(plot.title = element_text(color = \"red\"))\n",
    "p5 = image_ggplot(image_border((img), \"grey\", \"4x4\"))+ggtitle('chr19.1053362_G_A')+theme(plot.title = element_text(color = \"grey\"))\n",
    "p6 = image_ggplot(image_border((img), \"red\", \"4x4\"))+ggtitle('chr19.1053362_G_A')+theme(plot.title = element_text(color = \"red\"))\n",
    "pA = (p1/p2/p3/p4/p5/p6)\n",
    "\n",
    "# show all \n",
    "options(repr.plot.width=18, repr.plot.height=15)\n",
    "\n",
    "extended1 = (((pA|plot_grp0) + plot_layout(widths=c(1,2.1)))/(cont_plots/disc_plots + plot_layout(heights = c(1, 1)))) + plot_layout(heights=c(1.3,1)) + plot_annotation(tag_levels = 'A')\n",
    "extended1\n",
    "pdf('./pdf_figures/Extended_Figure_1.pdf', width=18, height=15)\n",
    "extended1\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d895e63f-bbc4-483f-8385-b088b6fa8bf5",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 2**</u>\n",
    "<a name=\"EF2\"></a>\n",
    "##### Sample swap results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ffa3c1f-f043-49ec-ae17-290425e4e533",
   "metadata": {},
   "outputs": [],
   "source": [
    "files = list.files(path = \"./pdf_figures/.\")\n",
    "files = files[unlist(lapply(files, function(x) endsWith(x,'swap_res.pdf')))]\n",
    "                            \n",
    "plots = list()\n",
    "for(i in 1:length(files)){\n",
    "    plots[[i]] = image_ggplot(image_trim(image_read_pdf(paste0('./pdf_figures/', files[i]))))\n",
    "}\n",
    "     \n",
    "options(repr.plot.width=18, repr.plot.height=14)\n",
    "\n",
    "pdf('./pdf_figures/Extended_Figure_X.pdf', width=18, height=14)\n",
    "do.call(\"grid.arrange\", c(plots, ncol=7))\n",
    "dev.off()\n",
    "                            \n",
    "do.call(\"grid.arrange\", c(plots, ncol=7))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a641ed5d-4fd4-45f0-9c9b-e298cfb7f2aa",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 3**</u>\n",
    "<a name=\"EF3\"></a>\n",
    "##### batch correction overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6d4de0d-a146-4413-844b-3b501309adf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# colors\n",
    "colors = readRDS('./processed_data/single_cell_data/Cell_group_colors.rds')\n",
    "\n",
    "# metadata\n",
    "all_data = readRDS('./processed_data/single_cell_data/analysis_input_data.rds')\n",
    "\n",
    "# cell coordinates\n",
    "coords = read.csv('./processed_data//for_plotting/umap_coords.csv', row.names=1)\n",
    "\n",
    "# DEGS (JBM, all..)\n",
    "degs = readRDS('./processed_data/differentially_expressed_genes_data/abca7lof_pseudobulk_degs_nov10.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db706fc4-8b59-477f-ac0d-8f7e06ef841b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load the data\n",
    "features = c('cogn_global_lv', 'amyloid', 'tangles','age_death', 'pmi')\n",
    "\n",
    "# continuous variables by variant\n",
    "metadata = all_data$summary\n",
    "metadata$seq_batch = ifelse(metadata$seq_batch=='JBM', 'batch_1', 'batch_2')\n",
    "metadata$seq_batch = factor(metadata$seq_batch, levels = c('batch_1', 'batch_2'))\n",
    "\n",
    "cols_batch = c('blue', 'pink')\n",
    "names(cols_batch) = c('batch_1', 'batch_2')\n",
    "\n",
    "cols = c('grey', 'red')\n",
    "names(cols) = c('Con', 'LoF')\n",
    "\n",
    "plt00 = boxplot_w_stats(df = as.data.frame(metadata), x = 'seq_batch', y = features[1], palette = cols_batch, comparisons = list(c('batch_1', 'batch_2')), xlab = '', ylab = features[1], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt01 = boxplot_w_stats(df = as.data.frame(metadata), x = 'seq_batch', y = features[2], palette = cols_batch, comparisons = list(c('batch_1', 'batch_2')), xlab = '', ylab = features[2], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt02 = boxplot_w_stats(df = as.data.frame(metadata), x = 'seq_batch', y = features[3], palette = cols_batch, comparisons = list(c('batch_1', 'batch_2')), xlab = '', ylab = features[3], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt03 = boxplot_w_stats(df = as.data.frame(metadata), x = 'seq_batch', y = features[4], palette = cols_batch, comparisons = list(c('batch_1', 'batch_2')), xlab = '', ylab = features[4], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "plt04 = boxplot_w_stats(df = as.data.frame(metadata), x = 'seq_batch', y = features[5], palette = cols_batch, comparisons = list(c('batch_1', 'batch_2')), xlab = '', ylab = features[5], width = .5, alpha = .5)+ theme(legend.position=\"none\")\n",
    "\n",
    "# categorical variables by genotype\n",
    "features = c('braaksc', 'ceradsc', 'msex','apoe_genotype', 'cogdx')\n",
    "plt10 = get_barplot(as.data.frame(metadata), 'seq_batch', features[1])\n",
    "plt11 = get_barplot(as.data.frame(metadata), 'seq_batch', features[2])\n",
    "plt12 = get_barplot(as.data.frame(metadata), 'seq_batch', features[3])\n",
    "plt13 = get_barplot(as.data.frame(metadata), 'seq_batch', features[4])\n",
    "plt14 = get_barplot(as.data.frame(metadata), 'seq_batch', features[5])\n",
    "\n",
    "cont_plots = (plt00|plt01|plt02|plt03|plt04)\n",
    "disc_plots = (plt10|plt11|plt12|plt13|plt14)\n",
    "\n",
    "coords$batch = ifelse(coords$batch=='JBM', 'batch_1', 'batch_2')\n",
    "coords$LoF = ifelse(coords$LoF=='LOF', 'LoF', 'Con')\n",
    "\n",
    "# plot cells\n",
    "size=6\n",
    "con = plot_coords_by_grp(coords[coords$LoF=='Con',], 'x', 'y', 'LoF', 'node_centrality', cols)+\n",
    "annotate(geom=\"text\", x=-2, y=13, label=\"Control\",color=cols['Con'], size=size, fontface='italic')\n",
    "\n",
    "lof = plot_coords_by_grp(coords[coords$LoF=='LoF',], 'x', 'y', 'LoF', 'node_centrality', cols)+\n",
    "annotate(geom=\"text\", x=-2, y=13, label=\"ABCA7 LoF\",color=cols['LoF'], size=size, fontface='italic')\n",
    "\n",
    "batch1 = plot_coords_by_grp(coords[coords$batch=='batch_1',], 'x', 'y', 'batch', 'node_centrality', cols_batch)+\n",
    "annotate(geom=\"text\", x=-2, y=13, label=\"Batch #1\",color=cols_batch['batch_1'], size=size, fontface='italic')\n",
    "\n",
    "batch2 = plot_coords_by_grp(coords[coords$batch=='batch_2',], 'x', 'y', 'batch', 'node_centrality', cols_batch)+\n",
    "annotate(geom=\"text\", x=-2, y=13, label=\"Batch #2\",color=cols_batch['batch_2'], size=size, fontface='italic')\n",
    "\n",
    "# plot deg correlations\n",
    "out = lapply(names(degs$degs_all), function(x) merge(degs$degs_all[[x]]$res, degs$degs_JBM[[x]]$res, by = 0))\n",
    "names(out) = names(degs$degs_all)        \n",
    "\n",
    "plt1 = plot_degs(out$Ex, out$Ex[order(out$Ex$P.Value.x, decreasing=F),'Row.names'][1:5], 'Excitatory neurons') # choose specific genes of interest to show\n",
    "plt2 = plot_degs(out$In,  out$In[order(out$In$P.Value.x, decreasing=F),'Row.names'][1:5], 'Inhibitory neurons')\n",
    "plt3 = plot_degs(out$Ast,  out$Ast[order(out$Ast$P.Value.x, decreasing=F),'Row.names'][1:5], 'Astrocytes')\n",
    "plt4 = plot_degs(out$Mic,  out$Mic[order(out$Mic$P.Value.x, decreasing=F),'Row.names'][1:5], 'Microglia')\n",
    "plt5 = plot_degs(out$Oli,  out$Oli[order(out$Oli$P.Value.x, decreasing=F),'Row.names'][1:5], 'Oligodendrocytes')\n",
    "plt6 = plot_degs(out$Opc,  out$Opc[order(out$Opc$P.Value.x, decreasing=F),'Row.names'][1:5], 'OPCs')\n",
    "\n",
    "df = as.data.frame(table(coords$celltype, coords$batch))\n",
    "props = ggplot(coords, aes(x=batch, fill=celltype), color='black') + geom_bar(position = \"fill\", color='black')+ scale_fill_manual(values = (colors[as.character(unique(coords$celltype))])) + theme_classic() + ylab('proportion') + xlab('')\n",
    "             \n",
    "options(repr.plot.width=13, repr.plot.height=15)\n",
    "\n",
    "extended2 = (cont_plots / disc_plots / (batch1 | batch2 | props) / (plt1 | plt2 | plt3) / (plt4 | plt5 | plt6)) + plot_layout(heights=c(2,2,4,4,4)) + plot_annotation(tag_levels = 'A', title='Extended Figure 1')\n",
    "\n",
    "extended2 \n",
    "pdf('./pdf_figures/Extended_Figure_2.pdf', width=13, height=15)\n",
    "extended2\n",
    "dev.off()\n",
    "             \n",
    "# check fisher test for the other variables\n",
    "# add fisher test for the batch variable?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24eeb06a-f711-4136-bda6-4475b53d1f11",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 4**</u>\n",
    "<a name=\"EF3\"></a>\n",
    "##### QC and celltype anno overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c811cc4-03d0-49fb-bfe2-e2eb24d5f55e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# per-cell marker expression\n",
    "marker_logcounts_melted = read.csv('./processed_data/for_plotting/marker_logcounts_melted.csv')\n",
    "\n",
    "# colors\n",
    "colors = readRDS('./processed_data/single_cell_data/Cell_group_colors.rds')\n",
    "\n",
    "# various QC/stats metrics\n",
    "df = read.csv('./processed_data/for_plotting/celltype_annos_qc.csv')\n",
    "data = readRDS('./processed_data/for_plotting/celltype_anno_counts.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21e6a629-5d2c-40ea-9666-cdb85ae27e9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot marker genes\n",
    "marker_genes = c('SYT1', 'NRGN', 'GAD1', 'AQP4', 'CSF1R', 'MBP', 'PLP1', 'VCAN',  'PDGFRB', 'FLT1')\n",
    "marker_logcounts_melted$celltype[marker_logcounts_melted$celltype=='Endo'] = 'Vascular'\n",
    "marker_logcounts_melted$Var1 = factor(marker_logcounts_melted$Var1, levels = marker_genes)\n",
    "\n",
    "plt0 = ggplot(marker_logcounts_melted, aes(x = celltype, y = value, fill = celltype)) + geom_boxplot(outlier.shape = NA)  + facet_wrap(Var1 ~ ., ncol = 5, scales = 'free')+ theme_classic() + theme( strip.background = element_blank() )  + scale_fill_manual(values = colors[marker_logcounts_melted$celltype])\n",
    "\n",
    "# plot fractions\n",
    "df$celltype[df$celltype=='Endo'] = 'Vascular'\n",
    "cells = unique(df$celltype)\n",
    "df$celltype = factor(df$celltype, levels = cells[order(cells)])\n",
    "colourCount = length(unique(df$projid))\n",
    "getPalette = colorRampPalette(brewer.pal(9, \"Set1\"))\n",
    "\n",
    "plt1 = ggplot(df) + aes(x = factor(projid), fill = factor(celltype)) + geom_bar(color = \"black\", position = \"fill\") + ggtitle('individual-level cell fractions') + xlab(\"LoF variants\") + ylab(\"fraction\") + theme_classic() + labs(x = \"\", fill = 'fraction') + scale_fill_manual(values = colors[as.character(df$celltype)]) + facet_grid(. ~ LOF, scales = \"free\", space = \"free\") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "plt2 = ggplot(df) + aes(fill = factor(projid), x = factor(celltype)) + geom_bar(color = \"black\", position = \"fill\") + ggtitle('cell-level individual fractions') + xlab(\"LoF variants\") + ylab(\"fraction\") + theme_classic() + labs(x = \"\", fill = 'fraction') + scale_fill_manual(values = getPalette(colourCount)) + theme(legend.position = \"none\")\n",
    "\n",
    "# plot correlation heatmap\n",
    "C = unname(colors[c('Ex', 'In', 'Ast', 'Mic', 'Oli', 'Opc')])\n",
    "\n",
    "d = data$ind_cor%>%melt(.)%>%mutate(X=Var1)%>%mutate(Y=Var2)%>%\n",
    "separate(., 'Var1', c('From_celltype', 'From_projid'), '[.]')%>%\n",
    "separate(., 'Var2', c('To_celltype', 'To_projid'), '[.]')\n",
    "d$From_celltype[d$From_celltype=='Per'] = 'Vascular'\n",
    "d$To_celltype[d$To_celltype=='Per'] = 'Vascular'\n",
    "\n",
    "plt3 = d%>%as_tibble(.)%>%\n",
    "tidyHeatmap::heatmap(column_title='',row_title='',.row = X,.column = Y,.value = value,show_heatmap_legend =TRUE,show_column_names=F, show_row_names=F, palette_value = circlize::colorRamp2(seq(0, 1, length.out = 11), rev(RColorBrewer::brewer.pal(11, \"RdBu\"))), , clustering_distance_rows='pearson', , clustering_distance_columns='pearson')%>%\n",
    "add_tile(From_celltype, palette=C, show_annotation_name=FALSE)%>%\n",
    "add_tile(To_celltype, palette=C, show_annotation_name=FALSE, show_legend=FALSE)%>%wrap_heatmap()\n",
    "\n",
    "# plot correlation as boxplot\n",
    "df_subset = data$cross_cors\n",
    "df_subset$celltype1 = as.character(df_subset$celltype1)\n",
    "df_subset$celltype2 = as.character(df_subset$celltype2)\n",
    "df_subset$celltype1[df_subset$celltype1=='Per'] = 'Vascular'\n",
    "df_subset$celltype2[df_subset$celltype2=='Per'] = 'Vascular'\n",
    "\n",
    "plt4 = ggplot(df_subset) + aes(fill = factor(celltype1), x = factor(celltype1), y = value) + geom_boxplot() + ylab(\"individual correlations\") + theme_classic() + scale_fill_manual(values = colors[as.character(df_subset$celltype1)]) + theme(legend.position = \"none\")\n",
    "\n",
    "# plot barplots\n",
    "df = data$median_cells\n",
    "df$celltype = as.character(df$celltype)\n",
    "df$celltype[df$celltype=='Endo'] = 'Vascular' \n",
    "plt5 = ggplot(df) + aes(fill = factor(celltype), x = factor(celltype), y = median) + geom_bar(color = \"black\", stat=\"identity\") + ylab(\"Median cells/ind.\") + theme_classic() + scale_fill_manual(values = colors[as.character(df$celltype)]) + theme(legend.position = \"none\")\n",
    "\n",
    "df = data$N_cells\n",
    "df$celltype = as.character(df$celltype)\n",
    "df$celltype[df$celltype=='Endo'] = 'Vascular' \n",
    "plt6 = ggplot(df) + aes(fill = factor(celltype), x = factor(celltype), y = N) + geom_bar(color = \"black\", stat=\"identity\") + ylab(\"N cells/ind.\") + theme_classic() + scale_fill_manual(values = colors[as.character(df$celltype)]) + theme(legend.position = \"none\") + \n",
    "  geom_text(label = df$N, vjust = 1.5, colour = \"white\")\n",
    "\n",
    "df = data$no_cells\n",
    "plt7 = ggplot(df) + aes(fill = factor(celltype), x = factor(celltype), y = N) + geom_bar(color = \"black\", stat=\"identity\") + ylab(\"No cells/ind.\") + theme_classic() + scale_fill_manual(values = colors[as.character(df$celltype)]) + theme(legend.position = \"none\") + \n",
    "  geom_text(label = df$N, vjust = 1.5, colour = \"white\")\n",
    "\n",
    "# add some plot formatting\n",
    "plt0 = plt0 + ylab('') + xlab('') +  theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())\n",
    "plt1 = plt1 + ggtitle('')+  theme(axis.title.x=element_blank(),axis.text.x=element_blank(), axis.ticks.x=element_blank())\n",
    "plt2 = plt2 + ggtitle('') + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "plt4 = plt4 + xlab('') +ggtitle('')+ theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "plt5 = plt5 + xlab('') + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "plt6 = plt6 + xlab('') + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "plt7 = plt7 + xlab('') + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))\n",
    "\n",
    "options(repr.plot.width=10, repr.plot.height=14)\n",
    "\n",
    "extended3 = (((((plt0 / plt1) + plot_layout(heights=c(2,1))) | plt2) + plot_layout(widths=c(8,1.5)))/((plt3|(plt4/plt5/plt6/plt7))+plot_layout(widths=c(8, 1.5))))+plot_layout(widths=c(5,3,8))  + plot_annotation(tag_levels = 'A')\n",
    "\n",
    "extended3 \n",
    "pdf('./pdf_figures/Extended_Figure_3.pdf', width=9, height=12.6)\n",
    "extended3\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43e404f6-ce4b-4ff7-ac53-37c54671dbcf",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 5**</u>\n",
    "<a name=\"EF5\"></a>\n",
    "##### DEGS and GO overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e92719d2-453c-403d-b789-289dc7df0940",
   "metadata": {},
   "outputs": [],
   "source": [
    "source('./functions/analysis_functions.r')\n",
    "degs = readRDS('./processed_data/differentially_expressed_genes_data/abca7lof_pseudobulk_degs_nov10.rds')$degs_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b895c2a3-265a-4b7a-b4ea-7c58ba3a19ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot degs as heatmap\n",
    "scores = get_deg_scores(degs)\n",
    "\n",
    "sele_scores = list()\n",
    "for(i in names(scores)){\n",
    "    curr = as.data.frame(scores[[i]]$scores[genes])\n",
    "    colnames(curr) = 'score'\n",
    "    curr$celltype = i\n",
    "    curr$gene = names(scores[[i]]$scores[genes])\n",
    "    sele_scores[[i]] = na.omit(curr)\n",
    "}\n",
    "\n",
    "df = do.call('rbind', sele_scores)\n",
    "\n",
    "df = df%>%pivot_wider(., names_from = celltype, values_from=score)\n",
    "df = as.data.frame(df)\n",
    "rownames(df) = df$gene\n",
    "g = df$gene\n",
    "df$gene = NULL\n",
    "df[is.na(df)] = 0\n",
    "df$gene = g\n",
    "\n",
    "df = melt(df, id='gene')\n",
    "head(df)\n",
    "\n",
    "genes_hmap = df %>% as_tibble(.) %>%\n",
    "tidyHeatmap::heatmap(column_title='',row_title='',.row = gene,.column = variable,.value = value, row_names_gp = gpar(fontsize = 11),\n",
    "                     palette_value = circlize::colorRamp2(seq(-5, 5, length.out = 11), rev(RColorBrewer::brewer.pal(11, \"RdBu\"))))%>%wrap_heatmap()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "084db7c6-ede8-4b01-b380-18f0197847fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "### UMAP projections overview and specific genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77b0b28c-d25f-4bc6-8f0d-4ff71d6f9f22",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 6**</u>\n",
    "<a name=\"EF5\"></a>\n",
    "##### KL clustering overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ac4cc58-64af-481f-a398-ad721e6ab040",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in image_ggplot(image_trim(image_read(\"./svgs/KL_cartoon.png\"))): could not find function \"image_ggplot\"\n",
     "output_type": "error",
     "traceback": [
      "Error in image_ggplot(image_trim(image_read(\"./svgs/KL_cartoon.png\"))): could not find function \"image_ggplot\"\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "cartoon = image_ggplot(image_trim(image_read('./svgs/KL_cartoon.png')))\n",
    "extended5 = cartoon  + plot_annotation(tag_levels = 'A', title='Extended Figure 4')\n",
    "\n",
    "options(repr.plot.width=11, repr.plot.height=14)\n",
    "\n",
    "extended5\n",
    "pdf('./pdf_figures/Extended_Figure_5.pdf', width=11, height=14)\n",
    "extended5\n",
    "dev.off()\n",
    "\n",
    "### and supplmentary clustering analysis for the other celltypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac6ca244-816a-4b03-a925-36d70f5b8cca",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 7**</u>\n",
    "<a name=\"EF7\"></a>\n",
    "##### Lipidomics overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98d77770-6f4d-47e5-8aa1-e9bce1cfda98",
   "metadata": {},
   "outputs": [],
   "source": [
    "# lipidomics data\n",
    "data = readRDS('./processed_data/for_plotting/lipidomics.rds')\n",
    "key = read.csv('./raw_data/ngn2_data/lipidkey.csv', sep = '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2786f89e-22d5-4a2f-95cf-461c7b7ef1ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "sce = data$ngn2_all\n",
    "sce = sce[,sce$treatment!='Choline']\n",
    "\n",
    "# plot correlations\n",
    "d = cor(counts(sce)) %>% melt(.)\n",
    "d$LoF = colData(sce)[d$Var1,'Genotype']\n",
    "d$LoF2 = colData(sce)[d$Var2,'Genotype']\n",
    "d$b = colData(sce)[d$Var1,'treatment']\n",
    "d$b2 = colData(sce)[d$Var2,'treatment']\n",
    "d$b = ifelse(d$b=='H2O', 'batch_1', 'batch_2')\n",
    "d$b2 = ifelse(d$b2=='H2O', 'batch_1', 'batch_2')\n",
    "\n",
    "p2 = d %>% as_tibble(.) %>%\n",
    "tidyHeatmap::heatmap(column_title='',row_title='',.row = Var1,.column = Var2,.value = value,show_heatmap_legend =TRUE, show_row_names=F, show_column_names=F, palette_value = circlize::colorRamp2(seq(0, 1, length.out = 11), rev(RColorBrewer::brewer.pal(11, \"RdBu\"))), clustering_distance_columns='pearson', clustering_distance_rows='pearson')%>%\n",
    "add_tile(LoF, palette=c('grey', 'red'), show_annotation_name=FALSE, show_legend=TRUE)%>%\n",
    "add_tile(LoF2, palette=c('grey', 'red'), show_annotation_name=FALSE, show_legend=FALSE)%>%\n",
    "add_tile(b, palette=c('purple', 'blue'), show_annotation_name=FALSE, show_legend=TRUE)%>%\n",
    "add_tile(b2, palette=c('purple', 'blue'), show_annotation_name=FALSE, show_legend=FALSE)%>%wrap_heatmap()\n",
    "\n",
    "# plot scatterplot between batches\n",
    "sample1 = sign(rowData(sce)$stats.log_ratio.s2.over.s1.)*rowData(sce)$stats.log_p_value.s2.over.s1.\n",
    "sample2 = sign(rowData(sce)$stats.log_ratio.s4.over.s3.)*rowData(sce)$stats.log_p_value.s4.over.s3.\n",
    "df = as.data.frame(cbind(sample1, sample2))\n",
    "df$Row.names = rowData(sce)$lipid.class\n",
    "df$Row.names = ifelse(abs(df$sample1)>1.3 & abs(df$sample2)>1.3, df$Row.names, '')\n",
    "gene = c('Cer','PC','TG')\n",
    "p3 = plot_lipid_diff(df, gene, 'con vs LoF batch comparison', 'Con vs LoF batch 2', 'Con vs LoF batch 1') #+ xlim(-2.5,5) + ylim(-2.5,5)\n",
    "\n",
    "# plot fractions\n",
    "key = key[!duplicated(key$Key),]\n",
    "rownames(key) = key$Key\n",
    "\n",
    "meta = rowData(sce)\n",
    "meta$class = key[meta$lipid.class,'Class']\n",
    "\n",
    "meta$change = ifelse(meta$pvals_both_batch<0.005 & meta$logfc_both_batch>0, 'up', \n",
    "       ifelse(meta$pvals_both_batch<0.005 & meta$logfc_both_batch<0, 'down', 'unchanged'))\n",
    "\n",
    "# have to figure out the NA values\n",
    "meta$change = factor(meta$change, levels = (c('up', 'down', 'unchanged')))\n",
    "meta = na.omit(as.data.frame(meta))\n",
    "x = colSums(table(meta$change, meta$class)[1:2,])\n",
    "\n",
    "meta$class = factor(meta$class, levels = (names(x[order(x)])))\n",
    "p4 = ggplot(meta, aes(x=class, fill=change), color='black') + scale_fill_manual(values = c('red', 'blue', 'white')) + geom_bar(position = \"fill\", color='black') + theme_classic() + ylab('proportion') + xlab('') + theme(axis.text.x=element_text(angle=45, hjust=1))\n",
    "\n",
    "meta = meta[meta$class%in%c('Neutral lipids', 'Phospholipids', 'Sphingolipids'),]\n",
    "x = colSums(table(meta$change, meta$lipid.class)[1:2,])\n",
    "meta$lipid.class = factor(meta$lipid.class, levels = (names(x[order(x)])))\n",
    "\n",
    "p5 = ggplot(meta, aes(x=lipid.class, fill=change), color='black') + scale_fill_manual(values = c('red', 'blue', 'white')) + geom_bar(position = \"fill\", color='black') + theme_classic() + ylab('proportion') + xlab('') +  facet_grid(. ~ class, scales=\"free\", space=\"free\")+ theme(axis.text.x=element_text(angle=45, hjust=1))+ theme(legend.position = \"none\")\n",
    "\n",
    "# plot table\n",
    "tbl = as.data.frame(table(meta$lipid.class, meta$change))\n",
    "tbl = tbl%>%pivot_wider(values_from=Freq, names_from=Var2)\n",
    "tbl$total = rowSums(cbind(tbl$up, tbl$down))\n",
    "tbl = tbl[order(tbl$total, decreasing=T),]\n",
    "tbl$percentage = round(tbl$total/(tbl$total+tbl$unchanged), 2)*100\n",
    "tbl = as.data.frame(tbl)\n",
    "rownames(tbl) = tbl$Var1\n",
    "tbl$Var1 = NULL\n",
    "tbl = tbl[, c('up', 'down', 'total', 'percentage')]\n",
    "colnames(tbl) = c('U', 'D', 'T', '%')\n",
    "tbl$Name = key[rownames(tbl),c('subclass')]\n",
    "p6 = gridExtra::tableGrob(tbl)\n",
    "\n",
    "\n",
    "options(repr.plot.width=11, repr.plot.height=11)\n",
    "\n",
    "extended7 = (((p2/p3)|p6)/p5) + plot_layout(heights=c(5,1))+ plot_annotation(tag_levels = 'A') \n",
    "extended7\n",
    "\n",
    "pdf('./pdf_figures/Extended_Figure_7.pdf', width=11, height=11)\n",
    "extended7\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb54b739-b9af-4907-8736-2fa8f7310625",
   "metadata": {},
   "source": [
    "<u>**Extended Figure 8**</u>\n",
    "<a name=\"EF8\"></a>\n",
    "##### CRISPRi data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r_env]",
   "language": "R",
   "name": "conda-env-.conda-r_env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
