## Software & Package Versions

For single-cell analysis,library demultiplexing was conducted using the BMC/BCC pipelines v1.8 (BioMicroCenter Software,https://openwetware.org/wiki/BioMicroCenter:Software), and Fast-q reads were aligned to the human genome using Cell Ranger version 6.1.2 (lOx Genomics). Batch effects were corrected using Harmony (https://github.com/immunogenomics/harmony). Molecular dynamics simulations employed PyMOL v2.0, CHARMM-GUI (https://charmm-gui.org/), GROMACS 2022.3, and VMD v1.94 software. LC-MS analysis was performed with Lipidsearch Â© software (version 4.2.27) and Compound Discoverer 3.2 (CD, Thermo Fisher Scientific). OCRs were computed in the XFe Assay Version (2.6.3.5) software. Electrophysiological recordings and data analysis were conducted using GraphPad Prism 10 software suites. Custom pipelines and analysis scripts were developed in Python version 3.8.13 and R version 4.2.3. The following Python packages (algorithms) were used for analysis : Scikit-learn v1.6.1 (GaussianMixture), Statsmodels v0.14.4 (mixedlm), Scanpy v1.10.3, Umap-learn v0.5.7, GSEAPY v1.1.5, Numpy v1.24.3 (Kernighan-Lin), NetworkX v3.2.1 (spring_layout), AICSImageIO v4.14.0. The following R packages (algorithms) were used for analysis: edgeR v4.4.0, Limma v3.62.1 (voom, lmFit, eBayes, topTable), fGSEA v1.32.2. Additional software packages used were METIS https://github.com/KarypisLab/METIS, Trim Galore v0.6.10, STAR aligner v2.5.2b, featureCounts v2.0.1.
