% \documentclass[12pt]{article}
% \input{../preamble.tex}
% \begin{document}
\begin{figure}[H]
    \begin{subfigure}[t]{\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{/Users/djuna/Documents/ABCA7lof2/editorial_paper/main_panels_svgs/fs2/batch_cont_var.pdf}        
    \end{subfigure}
    \begin{subfigure}[t]{\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{/Users/djuna/Documents/ABCA7lof2/editorial_paper/main_panels_svgs/fs2/batch_disc_var.pdf}        
    \end{subfigure}  
    \begin{subfigure}[t]{\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{/Users/djuna/Documents/ABCA7lof2/editorial_paper/main_panels_svgs/fs2/additional_projections.pdf}        
    \end{subfigure}   
    \begin{subfigure}[t]{\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{/Users/djuna/Documents/ABCA7lof2/editorial_paper/main_panels_svgs/fs2/score_correlations_by_batch.pdf}        
    \end{subfigure}   
\end{figure}
\paragraph*{Fig. S2.} Overview of snRNA-sequencing batch correction and data quality.
\phantomsection
\addcontentsline{toc}{paragraph}{Fig. S2.}
\textbf{a, b,} Distribution of continuous clinical variables (Supplementary Note 1) comparing sequencing batches (batch 1, N=10; batch 2, N=26 subjects). \textbf{c,} Distribution of discrete metadata variables across sequencing batches; Fisher’s exact tests (R stats::fisher.test, extended to r×c tables). \textbf{d,} 2D UMAP projection of snRNA-seq cells after quality control (N=102,710 cells), colored by selected metadata variables. \textbf{e,} Correlation of gene perturbation scores ($S = -\log_{10}(p)\times\text{sign}(\log_2(\text{fold-change}))$, unadjusted Limma-Voom p-values; N=24 controls, 12 LoF subjects), computed using all samples compared to excluding batch 2 (v2 chemistry), demonstrating robustness and minimal batch-specific effects. \textbf{a,b:} boxplots show median, IQR (box), and whiskers = 1.5×IQR.