\paragraph{Acknowledgments:} We thank the individuals who donated postmortem brain samples, and their families, for enabling this research; Y. Zhou, E. McNamara and T. Garvey for administrative support and animal care; U. Geigenmüller for reviewing and editing the manuscript and for helpful discussions; Rebecca Pinals for helpful discussions and input on the manuscript; J. Davila-Velderrain and R. Firenze for providing input on the manuscript; the MIT SuperCloud and Lincoln Laboratory Supercomputing Center for providing HPC and consultation resources that have contributed to the research results reported within this paper; the Harvard Center for Mass Spectrometry (HCMS) for lipidomic and metabolomic sample runs and Charles Vidoudez (HCMS) for consultation on the lipidomic data analysis; the MIT BioMicro Center for bulk and single-nucleus RNA-sequencing runs and Stuart Levine and his Team for consultations on data processing; ROSMAP resources can be requested at \url{https://www.radc.rush.edu}. Graphic illustrations were generated using BioRender under agreement \# X25SQ61W7.

\paragraph{Funding:} This work was supported in part by the Cure Alzheimer’s Fund, The JBP Foundation, the Carol and Gene Ludwig Family Foundation, James D. Cook and NIH grants R56-AG081376,RF1-AG062377, RF1-AG054321, RO1-AG054012 (L.-H.T.), andRO1-AG075901 (P.I. Ernest Fraenkel).

\paragraph{Author contributions:} DVM and L-HT designed the study, with L-HT supervising the overall project and acquiring funding alongside MK. Experimental work included snRNA-seq experiments performed by DVM, JMB, HR, and LL; Seahorse assays conducted by SEW; and differentiation and maintenance of iPSC lines, induced neurons, and cortical organoids by SEW, CS, P-CP, and OK. SEW and P-CP prepared samples for LC-MS analysis and performed amyloid ELISA experiments and neuronal marker staining. P-CP conducted TMRM and CellRox assays and SEW performed MitoHealth assays. P-CP, CS, and OK prepared bulk RNA samples. LL conducted electrophysiological recordings and analysis, and CJY generated the p.Tyr622* cell line. AS performed molecular dynamic simulations, analyses, and visualizations and DVM performed formal data analysis and visualization, with contributions from C-AB, GL, and AS. Experimental and technical support was provided by A-NS, ML, G-SM, GW, AG, NL, and GS. C-CC and DL helped with revision experiments. DVM, SEW, and L-HT wrote the first draft of the manuscript. DVM and L-HT wrote the revised draft of the manuscript.

% \begin{table}[ht]
%     \centering
%     \caption{Example CRediT Roles Contributions}
%     \begin{tabular}{@{}lcccc@{}}
%     \toprule
%     \textbf{CRediT Role}               & \textbf{DVM} & \textbf{SEW} & \textbf{MK} & \textbf{LHT} \\ \midrule
%     Conceptualization                  & X               &                &                 &  X               \\
%     Data Curation                      & X                & X               &                &                 \\
%     Formal Analysis                    & X               &                 &                &                 \\
%     Funding Acquisition                & x                &                & X                & x               \\
%     Investigation                      & X               & X               &                 &                \\
%     Methodology                        & X               & X                &                &                 \\
%     Project Administration             & X                &                 &                &                \\
%     Resources                          &                 &                & X                & X               \\
%     Software                           & X               &                 &                &                 \\
%     Supervision                        & X                &                 &                &  X               \\
%     Validation                         &                &                 &                 &                \\
%     Visualization                      & X               & X               &                 &                 \\
%     Writing - Original Draft           & X               & X                &                &                 \\
%     Writing - Review \& Editing        & X                &                &                & X               \\ \bottomrule
%     \end{tabular}
%     \label{tab:credit_roles}
% \end{table}



\paragraph{Competing interests:} Authors declare that they have no competing interests.


\paragraph{Ethics Statement:} The study protocol involving the use of human stem cells was approved by the Coriell Institutional Review Board (Coriell IRB) in compliance with DHHS regulations (45 CFR Part 46). The initial cell lines were obtained from the Coriell Institute, which ensured that informed consent was received from all donors. Donors were informed that their tissue donations would be used for the creation of cell lines intended for educational and research purposes, and that all biological materials would be anonymized. More information can be found here. For postmortem human brain samples, informed consent was obtained from each subject, and the Religious Orders Study and Rush Memory and Aging Project were approved by an Institutional Review Board (IRB) of Rush University Medical Center.


\paragraph{Data availability:} All postmortem human data can be accessed through the Synapse AD Knowledge Portal (syn53461705), which also includes associated ROSMAP metadata. These data are subject to controlled access in compliance with human privacy regulations. To obtain the data, a data use agreement (DUA) must be completed. This requirement ensures the anonymity of ROSMAP study participants. A DUA can be established with either the Rush University Medical Center (RUMC) or SAGE, the organization that manages Synapse. The necessary forms are available for download on their respective websites. All iPSC-related data are accessible through links provided in our code repositories. For a complete list of data availability and download links, please refer to the code repositories listed below. Additionally, relevant processed datasets are available in the supplementary files of this manuscript.


\paragraph{Code availability:} All code used in this study is available on GitHub (\url{https://github.com/djunamay/ABCA7lof2}).