\subsubsection{Single-nuclear transcriptomic profiling of human PFC from ABCA7 LoF-variant carriers} 
To investigate the cell type-specific impact of ABCA7 LoF variants in the human brain, we queried whole genome sequences of >1000 subjects from the Religious Order Study or the Rush Memory and Aging Project (collectively known as ROSMAP) for donors with Alzheimer’s disease diagnoses who are carriers of rare damaging variants in ABCA7 that result in a PTC. We identified 12 heterozygous carriers of ABCA7 LoF variants, including splice region variants (c.4416+2T>G and c.5570+5G>C), frameshift variants (p.Leu1403fs and p.Glu709fs), and nonsense ‘stop gained’ variants (p.Trp1245* and p.Trp1085*) (Figure~\ref{fig:main_atlas}A-C; Data~\ref{data:ptc_variants}). These variants have previously been associated with increased AD risk in genetic association studies (Table~\ref{tab:annotation_abca7}) \cite{Steinberg2015-mu,Holstege2022-vp} and are presumed to induce risk via ABCA7 haploinsufficiency \cite{Duchateau2024-rf}. Analysis of published proteomic data for a subset of the 12 ABCA7 PTC-variant carriers and controls \cite{Johnson2020-ip} (Table~\ref{tab:external_datasets}) confirmed that ABCA7 PTC-variant carriers indeed had lower ABCA7 protein levels in the human postmortem PFC compared to non-carriers (p=0.018; Figure~\ref{fig:main_atlas}D; Figure~\ref{fig:snRNA_cohort}A). 

We next selected 24 ABCA7 PTC non-carrier controls from the ROSMAP cohort that were matched to the ABCA7 LoF variant-carriers based on several potentially confounding variables, including Alzheimer's disease (AD) pathology, age at death, \textit{postmortem} intervals, sex, APOE genotype, and cognitive status (Figure~\ref{fig:main_atlas}C; Figure~\ref{fig:snRNA_cohort}B,C; Data~\ref{data:cohort_metadata}; Supplementary Text). We confirmed that none of the 36 selected subjects carried damaging variants in other known AD risk genes (\textit{TREM2}, \textit{SORL1}, \textit{ATP8B4}, \textit{ABCA1}, and \textit{ADAM10}) \cite{Holstege2022-vp} and verified \textit{ABCA7} genotypes in a subset of ABCA7 LoF carriers and matched controls using Sanger sequencing (Figure~\ref{fig:snRNA_cohort}D).

\newcommand{\quoteE}{\textcolor{blue}{For a subset of the selected samples, raw data (fastq files) for snRNAseq of the BA10 region of the prefrontal cortex (PFC) could be obtained from a previous study (10 non-carrier controls from \cite{Mathys2019-wb}). For the remaining samples, fresh-frozen tissue samples from PFC BA10 were obtained for analysis. SnRNAseq was performed using the 10x Genomics Chromium platform. Accurate genotype assignments were confirmed by matching each single-cell library to its corresponding whole genome sequencing data (Figure~\ref{fig:snRNA_cohort}E). Following extensive quality control measures—including detailed analysis and correction of batch effects (Figure~\ref{fig:snRNA_batch_quality}; Data~\ref{data:cellranger_metrics}; Methods)—our final dataset consisted of 102,710 high-quality cells (Figure~\ref{fig:main_atlas}E), out of an initial total of 150,456 cells. \label{quoteE-label}}}
\quoteE This dataset encompassed diverse populations of inhibitory neurons (In, \textit{SYT1} \& \textit{GAD1}+), excitatory neurons (Ex, \textit{SYT1} \& \textit{NRGN}+), astrocytes (Ast, \textit{AQP4}+), microglia (Mic, \textit{CSF1R}+), oligodendrocytes (Oli, \textit{MBP} \& \textit{PLP1}+), and oligodendrocyte precursor cells (OPCs, \textit{VCAN}+) (Figure~\ref{fig:main_atlas}E; Figure~\ref{fig:snRNA_quality_annotation}A-E). A small putative vascular cell cluster did not meet our quality thresholds and was excluded from further analysis. Post-quality control, cell types were robustly represented across subjects (Figure~\ref{fig:snRNA_quality_annotation}F,G), and gene expression profiles showed high consistency within cell types (mean correlation 0.95) (Figure~\ref{fig:snRNA_quality_annotation}H,I).

\subsubsection{Cell type-specific perturbations in the presence of ABCA7 LoF}
To investigate gene expression changes related to ABCA7 LoF across major cell types, we identified genes significantly perturbed (p<0.05, linear model; total genes = 2,389) in at least one of six major cell types (Ex, In, Ast, Mic, Oli, or OPC). We controlled for known and unknown covariates and considered only genes detected in >10\% of cells within each specific cell type (Methods; Data~\ref{data:degs}). Next, \newcommand{\quoteF}{\textcolor{blue}{we visualized these perturbed genes by projecting their high-dimensional perturbation scores ($\operatorname{score} = \operatorname{sign}(\log(\text{FC})) \times -\log_{10}(p\text{-value})$ for each cell type) onto two dimensions, as shown in Figure~\ref{fig:main_atlas}F. Genes exhibiting similar perturbation patterns across cell types are positioned closer together in this two-dimensional visualization.\label{quoteF-label}}}\quoteF

\newcommand{\quoteG}{\textcolor{blue}{The two-dimensional visualization effectively captured the transcriptional landscape of ABCA7 LoF gene changes across all major cell types (Figure~\ref{fig:main_atlas}F; Figure~\ref{fig:snRNAseq_gene_scores}A). To summarize this landscape in terms of biological pathways, we grouped genes into clusters based on their positions in the projection and analyzed each cluster for enrichment in biological pathways using the Gene Ontology Biological Process database (Figure~\ref{fig:main_atlas}G; Methods). This analysis identified several biological pathways correlated with ABCA7 LoF in the \textit{postmortem} human PFC, including pathways related to heat shock response, inflammation, cell dynamics and adhesion, synaptic function, DNA repair, and cellular metabolism (Figure~\ref{fig:main_atlas}G; Data~\ref{data:pathway_enrichments}).\label{quoteG-label}}}\quoteG

Decomposition of the ABCA7 LoF transcriptional signature identified shared and cell-specific gene perturbations across major PFC cell types (Figure~\ref{fig:main_atlas}G,H). Microglia showed significant downregulation of heat shock response genes (cluster 11), which was also observed to a lesser degree in neurons and OPCs (FDR-adjusted $p<0.01$, $|\operatorname{score}|>0.25$; Figure~\ref{fig:main_atlas}H). Astrocytes and microglia exhibited increased expression of transcriptional regulatory genes (clusters 10 and 9, respectively), including regulators of inflammatory gene transcription (e.g., \textit{NFKB1}). OPCs were disrupted in cell motility (clusters 6, 13) and inflammatory signaling (cluster 0; Figure~\ref{fig:main_atlas}H). Neurons had elevated DNA repair genes (cluster 12; \textit{e.g.}, \textit{FANCC}) and reduced synaptic transmission genes (cluster 1; \textit{e.g.}, \textit{NLGN1}, \textit{SHISA6}). Excitatory neurons uniquely showed enhanced expression of cellular respiration genes (cluster 7; \textit{e.g.}, \textit{NDUFV2}) and decreased expression of a cluster enriched for triglyceride biosynthesis (cluster 5; \textit{e.g.}, \textit{PPARD}; Figure~\ref{fig:main_atlas}H). Overlaps of specific differentially expressed genes across cell types are shown in Figure~\ref{fig:snRNAseq_gene_scores_2}A,B.

Together, these findings indicate that ABCA7 LoF variants may induce widespread, cell type-specific transcriptional changes in the human PFC. This single-cell atlas provides a rich resource for future studies aiming to elucidate the contributions of individual neural cell types to ABCA7 LoF-driven forms of AD risk. This resource will be made available for exploration via the UCSC Single Cell Browser and for further analysis via Synapse (accession ID: syn53461705).

\subsubsection{ABCA7 is expressed most highly in excitatory neurons}
Our snRNAseq data suggest that excitatory neurons expressed the highest levels of ABCA7, compared to other major cell types in the brain (Figure~\ref{fig:abca7_expression}A).  ABCA7 transcripts were detected (count>0) in ~ 30\% of excitatory neurons and ~ 15\% of inhibitory neurons, while the detection rate was considerably lower (<10\%) for microglia and astrocytes and an order of magnitude lower (<3\%) for oligodendrocytes and OPCs (Figure~\ref{fig:abca7_expression}A, B). We validated this expression pattern in an independent published dataset \cite{Welch2022-ef} (Table~\ref{tab:external_datasets}), where bulk RNA sequencing of NeuN- (glial) and NeuN+ (neuronal) cell populations derived from six human \textit{postmortem} temporal cortex samples showed significantly higher ABCA7 levels in the neuronal population versus the glial cell population (p=0.021; Figure~\ref{fig:abca7_expression}C). Several control genes, whose expression patterns in glial versus neuronal cells are well established (\textit{ABCA1}, \textit{APOE}, and \textit{NEUROD1}), had expected expression patterns that matched those in the snRNAseq data (Figure~\ref{fig:abca7_expression}B,C). These results indicate that neurons, particularly excitatory neurons, are the primary ABCA7-expressing cell type in the aged human PFC. Given the relatively higher expression of ABCA7 in excitatory neurons and the evidence of transcriptional perturbations by ABCA7 LoF in this cell type, we focused our subsequent analysis specifically on excitatory neurons.

\subsubsection{ABCA7 LoF perturbations in excitatory neurons}
As an alternative approach to the unsupervised clustering of gene perturbation scores among all cell types, we next used prior knowledge of biological pathway structure to perform an in-depth characterization of perturbed biological processes specifically in ABCA7 LoF excitatory neurons. To this end, we first estimated statistical overrepresentation of biological gene sets (WikiPathways, N pathways = 472) among up and down-regulated genes in ABCA7 LoF excitatory neurons vs controls (by GSEA; Methods). We observed a total of 34 pathways with evidence for transcriptional perturbation at p<0.05 in excitatory neurons (Data~\ref{data:exci_pathways}). Enrichments of these pathways were driven by 268 unique genes (“leading edge” genes \cite{Subramanian2005-pu}; Data~\ref{data:exci_pathways}).   

To extract unique information from leading-edge genes and limit pathway redundancy, we next separated these genes and their associated pathway annotations into non-overlapping groups, formalized as a graph partitioning problem (Figure~\ref{fig:main_neurons}A; Figure~\ref{fig:benchmarking_clustering}; Methods; Supplementary Text). Establishing gene-pathway groupings of approximately equal size revealed eight biologically interpretable “clusters” associated with ABCA7 LoF in excitatory neurons (Figure~\ref{fig:main_neurons}A,B; Data~\ref{data:kl_clusters}). Predominantly, these gene clusters centered around two themes: (1) energy metabolism and homeostasis (PM.0, PM.1) and (2) DNA damage (PM.2, PM.3), cell stress (PM.4, PM.5), and synaptic dysfunction (PM.7) (Figure~\ref{fig:main_neurons}B). \textcolor{blue}{Cortical layer-specific analysis indicated that these perturbation patterns remained largely consistent across cortical layers, from deeper to superficial regions (Figure~\ref{fig:ex_layers}).}

Clusters PM.0 and PM.1 were primarily defined by genes involved in cellular energetics, including genes related to lipid metabolism, mitochondrial function, and oxidative phosphorylation (OXPHOS) (Figure~\ref{fig:main_neurons}B). Cluster PM.0, characterized by transcriptional regulators of lipid homeostasis (\textit{e.g.,} \textit{NR1H3}, \textit{ACLY}, \textit{PPARD}), exhibited evidence for down-regulation in ABCA7 LoF and featured pathways related to "SREBP Proteins" and "Adipogenesis" (Figure~\ref{fig:main_neurons}B; Data~\ref{data:kl_clusters}). Cluster PM.1 comprised multiple mitochondrial complex genes (\textit{e.g.,} \textit{COX7A2}, \textit{NDUFV2}) responsible for ATP generation from carbohydrate and lipid catabolism and showed up-regulation in ABCA7 LoF (Figure~\ref{fig:main_neurons}B; Data~\ref{data:kl_clusters}). The remaining clusters PM.2-7 were characterized by DNA damage and proteasomal, inflammatory, and apoptotic mediators. Clusters PM.2, PM.3, and PM.6 were up-regulated in ABCA7 LoF excitatory neurons and characterized by pathway terms such as "DNA Damage Response" (PM.2 \& PM.3) and "DNA Replication" (PM.6) (Figure~\ref{fig:main_neurons}B; Data~\ref{data:kl_clusters}). They included up-regulated DNA damage/repair and proteasomal genes (\textit{e.g.,} \textit{RECQL}, \textit{TLK2}, \textit{BARD1}, \textit{RBL2}, \textit{MSH6}, \textit{PSMD5}). Genes in clusters PM.4, linked to “Proteasome Degradation” and "ciliogenesis"; PM.5, associated with “Apoptosis” and “TNFalpha Signaling Pathway”; and PM.7, linked to “GABA receptor Signaling,” “Gastric Cancer Network 1,” and “Prader-Willi and Angelman Syndrome”,  included both up- and down-regulated genes (Figure~\ref{fig:main_neurons}B; Data~\ref{data:kl_clusters}).

Together, these data suggest that ABCA7 LoF may disrupt energy metabolism in excitatory neurons and that these disruptions coincide with a state of increased cellular stress, characterized by genomic instability and neuronal dysfunction.

\subsubsection{ABCA7 LoF and common missense variants lead to overlapping neuronal perturbations}
\newcommand{\quoteL}{\textcolor{blue}{ABCA7 LoF variants substantially increase AD risk (Odds Ratio = 2.03) \cite{Steinberg2015-mu} but are rare and therefore only contribute to a small portion of AD cases \cite{Duchateau2024-rf}. To evaluate whether ABCA7 LoF transcriptomic effects in neurons generalize to more common, moderate-risk genetic variants in ABCA7, we examined the ROSMAP WGS cohort for carriers of the prevalent ABCA7 missense variant p.Ala1527Gly (rs3752246: Minor Allele Frequency ≈ 0.18; \% carriers ≥1 allele ≈30\%; Figure~\ref{fig:main_neurons}C). Although Gly1527 is listed as the reference allele, it represents the less common variant associated with increased AD risk (Odds Ratio = 1.15 [1.11-1.18]) \cite{Kunkle2019-yo,Holstege2022-vp,Naj2011-bs}. \label{quoteL-label}}} 
\quoteL We identified 133 individuals carrying at least one copy of the p.Ala1527Gly risk variant and 227 non-carriers (Figure~\ref{fig:main_neurons}D), all with available snRNAseq data from \textit{postmortem} PFC \cite{Mathys2023-rs}. We ensured that none of these 360 individuals were part of our earlier ABCA7 LoF snRNAseq cohort or carried ABCA7 LoF variants. Using this cohort, we investigated whether excitatory neurons from p.Ala1527Gly carriers exhibited evidence of transcriptomic perturbations in the ABCA7 LoF-associated clusters PM.0-7.

Remarkably, all clusters displayed directional trends in p.Ala1527Gly neurons consistent with the directionality observed in ABCA7 LoF neurons (Figure~\ref{fig:main_neurons}B,E,F), while controlling for pathology, age, sex, and other covariates (Methods). Notably, 4 out of 8 clusters exhibited substantial evidence of perturbation in p.Ala1527Gly variant carriers, with perturbation directions aligning with predictions for ABCA7 LoF (Figure~\ref{fig:main_neurons}E,F). Specifically, we observed an up-regulation in the DNA damage cluster PM.3 and the proteasomal cluster PM.4 in p.Ala1527Gly carriers compared to controls, suggesting a similar cell stress and genomic instability signature to ABCA7 LoF carriers (Figure~\ref{fig:main_neurons}E,F), and a borderline significant up-regulation of the mitochondrial cluster PM.1, again consistent with ABCA7 LoF (Figure~\ref{fig:main_neurons}E,F). Finally, we observed significant perturbation to the lipid cluster PM.0, which was down-regulated (Figure~\ref{fig:main_neurons}E,F) similar to our observations in ABCA7 LoF carriers. 

\newcommand{\quoteP}{\textcolor{blue}{Because missense variants often influence protein dynamics—and glycine substitutions typically introduce greater local flexibility than alanine—we next examined whether the convergent transcriptional signature associated with ABCA7 variants could be explained by structural changes in the protein. To directly investigate local structural consequences of the p.Ala1527Gly variant, we performed molecular dynamics simulations using newly available cryo-EM structures of ABCA7 in both the ATP-bound closed (Figure~\ref{fig:main_neurons}G,H) and ATP-unbound open (Figure~\ref{fig:md_simulations}A,B) conformations \cite{LeThiMy2022-dp,Jumper2021-na}. Specifically, simulations were conducted on a 239-residue region of ABCA7 embedded within a lipid bilayer, comparing the Ala1527 and Gly1527 variants over a 300-ns timescale (Figure~\ref{fig:main_neurons}I; Figure~\ref{fig:md_simulations}C; Figure~\ref{fig:md_simulations_2}; Methods; Supplementary Text).\label{quoteL-labelP}}}\quoteP

Our simulations revealed that the AD risk-associated Gly1527 variant increased local structural flexibility in the ATP-bound closed conformation, indicated by pronounced conformational fluctuations over time (Figure~\ref{fig:main_neurons}J). In contrast, the Ala1527 variant exhibited limited conformational fluctuations, suggesting minimal local structural flexibility in this conformation (Figure~\ref{fig:main_neurons}J). Both variants demonstrated stable conformational behavior in the ATP-unbound open state (Figure~\ref{fig:md_simulations}C-E). These results are further supported by analyses of $\phi$/$\psi$ dihedral angle distributions and secondary structure persistence, as detailed in the Supplementary Text (Figure~\ref{fig:md_simulations_2}).

Together, these data suggest that the Gly1527 variant may introduce increased local flexibility, potentially disrupting the stability of secondary structural elements specifically within the ATP-bound closed conformation. Given that this conformation is proposed to mediate lipid presentation to apolipoproteins \cite{LeThiMy2022-dp,Fang2025}, the p.Ala1527Gly substitution may impact the efficiency of lipid extrusion, consistent with recent experimental findings from \cite{Fang2025}. Combined with our transcriptomics analyses, these structural insights suggest that both rare, high-effect ABCA7 LoF variants and common, mild-effect variants may influence AD risk through similar ABCA7-dependent mechanisms, indicating that our in-depth studies of rare variants may generalize to broader at-risk populations.

\subsubsection{Deriving human neurons with ABCA7 LoF variants}
To complement the correlative analyses in ABCA7 LoF human tissue, we next used CRISPR-Cas9 genome editing to generate two isogenic iPSC lines, each homozygous for a different ABCA7 LoF variant, from a parental line without ABCA7 variants (WT). The first LoF variant, ABCA7 p.Glu50fs*3, was generated by a single base-pair insertion in ABCA7 exon 3, resulting in a PTC early in the ABCA7 gene (Figure~\ref{fig:main_mitochondrial}A; Figure~\ref{fig:ipsc_lines}A-C). The second LoF variant, ABCA7 p.Tyr622*, was generated by a single base-pair mutation in ABCA7 exon 15 (Figure~\ref{fig:main_mitochondrial}A; Figure~\ref{fig:ipsc_lines}A-C). This PTC re-creates a variant previously observed in patients as associated with AD \cite{Steinberg2015-mu} and thus provides clinical context to ABCA7 dysfunction. Both variants are expected to generate severely truncated ABCA7 proteins or, due to nonsense-mediated mRNA decay, no ABCA7 protein at all. However, transcript rescue from nonsense-mediated decay and possible generation of mutated forms of ABCA7 through mechanisms such as exon skipping, which have previously been reported for multiple ABCA7 LoF variants \cite{De_Roeck2017-hv}, cannot be excluded.

We differentiated isogenic iPSCs into neurons (iNs) via lentiviral delivery of a doxycycline-inducible NGN2 expression cassette as previously described \cite{Ho2016-kz} (Figure~\ref{fig:iN_markers}A). At 2 and 4 weeks post-NGN2 induction, cells expressed neuronal markers TUJ1 and MAP2 and exhibited robust neuronal processes as demonstrated by pan-axonal staining (Figure~\ref{fig:iN_markers}B,C). Both WT and ABCA7 LoF lines were capable of firing action potentials upon current injections (Figure~\ref{fig:differentiating_iPSC_neurons}A–D). Although the ABCA7 genotype did not alter resting membrane potential (Figure~\ref{fig:differentiating_iPSC_neurons}E), ABCA7 LoF iNs fired action potentials more readily and at lower current injection thresholds compared to WT iNs (Figure~\ref{fig:differentiating_iPSC_neurons}F,G), indicating a hyperexcitability phenotype. Collectively, these data confirm successful neuronal differentiation from iPSCs, robust electrophysiological activity, and recapitulation of Alzheimer's disease-associated neuronal hyperexcitability.

% Previous studies suggest that ABCA7 dysfunction affects amyloid processing \cite{Satoh2015-yu,Sakae2016-uy,Bamji-Mirza2018-xt,Chan2008-qu,De_Roeck2018-fw}. To verify these findings in our ABCA7 LoF iNs, we measured Aβ42 and Aβ40 levels in media from iNs after four weeks of culture. Using an enzyme-linked immunosorbent assay (ELISA), we found that ABCA7 LoF iNs exhibited a significant increase in both secreted Aβ42 and Aβ40 compared to wild-type cells (Figure~\ref{fig:differentiating_iPSC_neurons}H). Together, these data (1) confirm successful neuronal differentiation from iPSCs, and (2) show that ABCA7 LoF iNs recapitulate key AD-associated phenotypes of hyperexcitability and amyloid pathology.

\subsubsection{ABCA7 LoF iNs Recapitulate Excitatory Neuronal Transcriptional Signatures}
To investigate whether transcriptional changes associated with ABCA7 LoF observed in \emph{postmortem} human neurons are recapitulated in iNs, we performed bulk mRNA sequencing on ABCA7 WT, p.Glu50fs*3, and p.Tyr622* iNs (N=2, N=5, and N=5, respectively) after four weeks in culture (Data~\ref{data:ngn2_rnaseq}). Gene perturbation scores (defined as $\operatorname{score} = \operatorname{sign}(\log(\text{FC})) \times -\log_{10}(p\text{-value})$) showed a strong correlation between p.Glu50fs*3 vs. WT and p.Tyr622* vs. WT comparisons (Pearson correlation coefficient = 0.84; Figure~\ref{fig:main_mitochondrial}B), indicating consistency in the transcriptional impact of ABCA7 variants.

We next conducted gene set enrichment analysis (GSEA) on the differentially expressed genes from these comparisons, identifying 15 significantly perturbed pathways in each comparison, p.Glu50fs*3 vs. WT and Y in p.Tyr622* vs. WT (FDR-adjusted p < 0.05; WikiPathways). 
These pathways were driven by 356 and 334 unique "leading edge" genes, respectively \cite{Subramanian2005-pu}. K/L partitioning of these leading edge genes identified 9 clusters for p.Tyr622* (Figure~\ref{fig:main_mitochondrial}C;) and 10 clusters for p.Glu50fs*3 (Figure~\ref{fig:bulk_RNAseq_supplement}A). Eight of nine p.Tyr622* T clusters and eight of ten p.Glu50fs*3 G clusters showed significant overlap (FDR-adjusted p < 0.05) (Figure~\ref{fig:bulk_RNAseq_supplement}B), indicating substantial concordance between the two ABCA7 variant lines.

We also observed that transcriptional signatures in ABCA7 LoF iNs closely aligned with those identified in \textit{postmortem} excitatory neurons. Specifically, we found significant overlap in 5 out of 9 p.Tyr622*-associated clusters (Figure~\ref{fig:main_mitochondrial}D) and in 7 out of 10 p.Glu50fs*3-associated clusters (Figure~\ref{fig:bulk_RNAseq_supplement}C) with the clusters identified in \textit{postmortem} excitatory neurons, with the majority (4 out of 5 and 6 out of 7, respectively) showing concordant directional changes.

Due to the transcriptional similarity between the two LoF lines, our primary analysis focuses on the patient variant p.Tyr622*, with results for the p.Glu50fs*3 variant provided in supplementary materials (Figure~\ref{fig:bulk_RNAseq_supplement}). Consistent with findings from \textit{postmortem} data, p.Tyr622* iNs exhibited downregulated clusters associated with lipid metabolism (T.9 and T.13) and upregulated clusters related to cell cycle regulation and proteasomal activity (T.8 and T.14) compared to WT iNs (Figure~\ref{fig:main_mitochondrial}E). Notably, a mitochondrial cluster (T.10) demonstrated the most robust overlap with \textit{postmortem} data (PM.1) and was consistently up-regulated in both the p.Tyr622* and p.Glu50fs*3 lines, mirroring the findings in \textit{postmortem} neurons (Figure~\ref{fig:main_mitochondrial}D; Figure~\ref{fig:bulk_RNAseq_supplement}C). The probability of observing this degree of overlap by chance alone is very low ($p<5x10^{-5}$ in both cases, binomial test). Together, these data support a causal relationship between ABCA7 LoF variants and multiple transcriptional signatures observed in \textit{postmortem} excitatory neurons, including mitochondrial, proteasomal, cell cycle, and lipid metabolism components.

\subsubsection{ABCA7 LoF impairs mitochondrial uncoupling in neurons}
To further characterize mitochondrial alterations in ABCA7 LoF iNs, extending beyond the gene sets used for K/L cluster analysis, we examined the expression of 1,136 mitochondrial genes curated from the MitoCarta database in our bulk RNAseq data. Among the most significantly upregulated genes in p.Tyr622* versus WT iNs were genes encoding components of mitochondrial apoptosis pathways (e.g., \textit{CASP3}, \textit{BID}) and OXPHOS subunits (previously captured in clusters PM.1 and T.10) (Figure~\ref{fig:main_mitochondrial}F; Table~\ref{tab:y622_mito_genes}). Conversely, downregulated genes were significantly enriched (padj < 0.05) for key metabolic processes, including $\beta$-oxidation (\textit{ACAD} and \textit{CPT} genes), mitochondrial metabolite transport (\textit{SLC25} genes), and oxidative stress detoxification (\textit{CAT}) (Figure~\ref{fig:main_mitochondrial}F; Table~\ref{tab:y622_mito_genes}). These MitoCarta mitochondrial gene expression profiles were highly correlated between p.Tyr622* and p.Glu50fs*3 relative to WT iNs (Figure~\ref{fig:bulk_RNAseq_supplement}E).

To directly assess mitochondrial function in ABCA7 LoF neurons, we measured the oxygen consumption rate (OCR) of WT and ABCA7 LoF iNs over time using the Seahorse metabolic flux assay (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}A,B). The OCR-driven movement of protons across the inner mitochondrial membrane during OXPHOS builds and maintains the mitochondrial membrane potential (ΔѰm)(Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}C), and  \newcommand{\quoteC}{\textcolor{blue}{measuring OCR in the presence of mitochondrial inhibitors provides several functional readouts. Because OCR can be influenced by cell viability and mitochondrial abundance \cite{Divakaruni2014-eq,Gu2021-ms}, we only report internally normalized OCR ratios rather than absolute values \cite{Divakaruni2022-bp} for WT, ABCA7 p.Glu50fs30, and ABCA7 p.Tyr622 iNs. To assess the spare respiratory capacity, we normalized the OCR measured following pharmacological collapse of the proton gradient to the basal OCR, with higher values indicating more spare respiratory capacity \cite{Divakaruni2022-bp} (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}D). We then quantified the proportion of basal oxygen consumption that can be attributed to rebuilding the membrane potential lost due to proton leakage through the membrane (i.e., uncoupled mitochondrial OCR) rather than due to ATP synthesis \cite{Divakaruni2022-bp} (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}E).\label{quoteC-label}}}
\quoteC

While spare respiratory capacity was comparable between WT and ABCA7 LoF iNs (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}F), ABCA7 LoF iNs showed significantly reduced uncoupled mitochondrial respiration (Figure~\ref{fig:main_mitochondrial}G), a finding replicated in two independent experiments (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}G,H). Uncoupled mitochondrial oxygen consumption rates in WT iNs ($\approx 20\%$; Figure\ref{fig:main_mitochondrial}G) align with previously reported values for neurons and other cell types \cite{Divakaruni2011-uj,Jekabsons2004-fn,Jain2024-br}, indicating that ABCA7 LoF iNs exhibit abnormally low mitochondrial uncoupling. Consistent with this observation, expression levels of UCP2 - a member of the mitochondrial uncoupling protein family expressed in the brain \cite{Kumar2022-bb} - were reduced in ABCA7 LoF iNs (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}I). 

Because decreased mitochondrial uncoupling often correlates with elevated mitochondrial membrane potential (ΔѰm) \cite{Demine2019-yq,Zorov2021-sj}, we next assessed ΔѰm in NeuN-positive soma using the fixable MitoHealth dye, which accumulates in mitochondria proportionally to membrane potential. We observed significantly increased MitoHealth fluorescence in both p.Tyr622* and p.Glu50fs* iNs compared to WT per NeuN surface (Figure~\ref{fig:main_mitochondrial}H). To further confirm these findings, we measured ΔѰm in soma and neuronal processes using the fluorescent cation tetramethylrhodamine methyl ester (TMRM) in non-quenching mode. TMRM accumulation was higher in p.Tyr622* iNs relative to WT (Figure~\ref{fig:main_mitochondrial}I), and the specificity of this TMRM signal was validated by showing drastically reduced TMRM signal intensity after depolarization of the ΔѰm with the uncoupler FCCP (Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}J). Together, these results indicate that ABCA7 LoF iNs exhibit elevated ΔѰm.

Regulated mitochondrial uncoupling serves as a mechanism to control mitochondrial membrane potential and mitigate reactive oxygen species (ROS) generation \cite{Monteiro2021-ei,Demine2019-yq}. To assess whether ABCA7 LoF iNs exhibited elevated ROS levels, we incubated p.Tyr622* iNs with Cellrox dye, a fluorescent indicator of oxidative stress. We observed significantly increased Cellrox fluorescence in p.Tyr622* iNs compared to WT iNs (Figure~\ref{fig:main_mitochondrial}J), indicating elevated ROS accumulation in ABCA7 LoF iNs. Together, these data suggest that ABCA7 LoF variants decrease mitochondrial uncoupling and increase oxidative stress in neurons.

\subsubsection{ABCA7 LoF induces phosphatidylcholine imbalance in neurons}
\newcommand{\quoteA}{\textcolor{blue}{Since ABCA7 functions as a lipid transporter, we examined the lipidome of WT and ABCA7 LoF iNs using LC-MS (Data~\ref{data:ngn2_lipidome}). Comparing lipidomic profiles between WT and p.Glu50fs*3 iNs revealed significant alterations across multiple lipid classes, including neutral lipids, phospholipids, sphingolipids, and steroids (Figure~\ref{fig:main_lipids}A,B). Among these, triglycerides (TGs)—particularly species enriched in long-chain, predominantly polyunsaturated fatty acids—were frequently altered, showing significant upregulation in p.Glu50fs*3 iNs (Figure~\ref{fig:main_lipids}B,C).\label{quoteA-label}}}
\quoteA

\newcommand{\quoteB}{\textcolor{blue}{In line with ABCA7's established preference for phospholipids \cite{Tomioka2017-sq,Picataggi2022-hf,Fang2025}, several phospholipid species also exhibited notable differences (Figure~\ref{fig:main_lipids}B). Phosphatidylcholines (PCs), which are essential structural components of biological membranes and potential ABCA7 substrates \cite{LeThiMy2022-dp,Fang2025}, were most prominently affected; the majority ($\approx$64\% of perturbed PC species) showed increased abundance in p.Glu50fs3 iNs (Figure~\ref{fig:main_lipids}B). Further analysis based on fatty acid saturation—an important factor influencing membrane fluidity—revealed significant enrichment of saturated and monounsaturated (MUFA) PCs among the upregulated species (hypergeometric p=0.0027) (Figure~\ref{fig:main_lipids}D). In contrast, polyunsaturated fatty acid-containing (PUFA) PCs were primarily downregulated (p=0.0148) (Figure~\ref{fig:main_lipids}E,F).}}
\quoteB

To determine whether neutral lipid and PC imbalances were conserved in p.Tyr622* iNs, we performed targeted lipidomic analysis in positive ionization mode. Consistent with p.Glu50fs*3 iNs, upregulated lipids in p.Tyr622* iNs were significantly enriched for saturated PCs (hypergeometric p=0.0436) (Figure~\ref{fig:main_mitochondrial}K; Figure~\ref{fig:main_lipids}G,H). However, PUFA PCs and long-chain triglycerides were not reliably detected in this LC-MS run (Figure~\ref{fig:main_lipids}I,J), leaving it unclear whether p.Tyr622* iNs exhibit the same extent of PUFA PC downregulation or long-chain triglycerides upregulation as p.Glu50fs*3 iNs.

\textit{De novo} PC synthesis occurs via the Kennedy pathway, and subsequent remodeling of the fatty acyl chains is catalyzed by LPCAT enzymes through the Lands cycle, with LPCAT3 specifically introducing PUFA chains into PCs \cite{Boumann2003-ew,Wang2019-om,Zhao2008-pq}. LPCAT3 expression was reduced in p.Tyr622* and p.Glu50fs*3 iNs compared to WT (Figure~\ref{fig:main_lipids}K,L), aligning with increased levels of saturated PCs in these cells. \newcommand{\quoteH}{\textcolor{blue}{Overall, our data indicate accumulation of neutral lipids in ABCA7 LoF iNs, including long-chain polyunsaturated triglycerides and sterol lipids (zymosteryl), and reveal imbalances in PC composition, with higher saturated and monounsaturated species.\label{quoteH-label}}}
\quoteH

\subsubsection{Treatment with CDP-choline reverses impacts of ABCA7 LoF in neurons}
Previous work indicated that exogenous choline supplementation normalized lipid saturation levels in yeast and ameliorated APOE4-related lipid phenotypes \cite{Boumann2006-nz,Sienski2021-zt}. We therefore next examined whether CDP-choline treatment could similarly mitigate ABCA7 LoF-induced phenotypes in iNs.

\newcommand{\quoteD}{\textcolor{blue}{Targeted LC-MS analysis confirmed that CDP-choline treatment increased its concentration in the media from undetectable to detectable levels (Figure~\ref{fig:choline_treatment}A). Additionally, both CDP and choline specifically accumulated in media conditioned by p.Tyr622* cells after treatment (Figure~\ref{fig:choline_treatment}A), indicating extracellular hydrolysis of CDP-choline. While intracellular CDP and CDP-choline could  not reliably be detected in this experiment, intracellular choline levels significantly increased after treatment (Figure~\ref{fig:choline_treatment}B) and expression levels of choline transporters were significantly upregulated (Figure~\ref{fig:choline_treatment}C). This suggests that choline was successfully taken up by p.Tyr622* iNs upon CDP-choline treatment.\label{quoteD-label}}} 
\quoteD

We anticipated that higher intracellular choline availability would lead to increased phospholipid synthesis. Lipidomic analysis indeed revealed elevated levels of phospholipids, particularly choline-containing phospholipids (PC and lysophosphatidylcholines (LPC)) and sphingolipids (sphingomyelins (SM)), alongside a reduction in a single TG species, with other neutral lipid species showing a similar downward trend (Figure~\ref{fig:main_choline}A). Consistent with these changes, we observed increased expression of \textit{PCYT1B}, the enzyme responsible for the rate-limiting step in PC synthesis through the Kennedy pathway (Figure~\ref{fig:choline_treatment}C). Additionally, expression of several LPCAT genes, including LPCAT3, was elevated after treatment (Figure~\ref{fig:choline_treatment}D), coinciding with observed increases in both saturated and unsaturated PC species (Figure~\ref{fig:main_choline}A; Figure~\ref{fig:choline_treatment}E). These findings suggest that CDP-choline treatment promotes synthesis and remodeling of choline-containing lipids.

Next, we characterized changes induced by CDP-choline treatment using LC-MS-based metabolomics and bulk RNAseq. While most of the metabolites increasing or decreasing after treatment could not be annotated, a principal component analysis of the overall metabolite changes indicated that CDP-choline treatment reversed the separation of WT and pTyr622* iN along the axis of the first principal component (PC1; Figure~\ref{fig:choline_treatment}F). Transcriptionally, CDP-choline treatment also induced significant changes, clearly distinguishing treated from untreated samples (Figure~\ref{fig:choline_treatment}G). The transcriptional signature of CDP-choline treatment negatively correlated with that of p.Tyr622* (Figure~\ref{fig:main_choline}B), suggesting partial restoration toward the WT state. Performing K/L cluster analysis on the p.Tyr622* vs CDP-choline treated p.Tyr622* samples (Figure~\ref{fig:main_choline}C), we observed significant overlap in 7 of the 9 clusters identified in the p.Tyr622* vs WT comparison (Figure~\ref{fig:main_choline}D), with 5 of these clusters showing reversed directional changes following treatment (Figure~\ref{fig:main_choline}E). 

Specifically, clusters related to proteasomal and ribosomal functions (T+C.25 and T+C.31)—previously upregulated in p.Tyr622* (see T.14 and T.12)—were downregulated following CDP-choline treatment (Figure~\ref{fig:main_choline}D). Most notably, mitochondrial cluster T+C.26, exhibiting the strongest overlap with the T.10 p.Tyr622*and PM.1 \textit{postmortem} mitochondrial clusters, was also reversed following treatment (Figure~\ref{fig:main_choline}E). Further analysis using the MitoCarta database confirmed a significant reversal in expression of genes encoding mitochondrial proteins, including reduced expression of apoptosis-related genes (\textit{BID}, \textit{CASP3}; Figure~\ref{fig:main_choline}F), restoration (upregulation) of the mitochondrial metabolic signature (Table~\ref{tab:choline_mito_genes}), and increased expression of regulators of mitochondrial fusion (\textit{MFN2}, \textit{OPA1}), a process which enables high metabolic capacity, dissipation of mitochondrial membrane potential, and mitochondrial biogenesis \cite{Westermann2010-au}. Overall, ABCA7 LoF-related changes to expression of MitoCarta genes were significantly reversed following CDP-choline treatment (Figure~\ref{fig:choline_treatment}H). 

To determine whether CDP-choline treatment could restore mitochondrial uncoupling to WT levels, we repeated the Seahorse assay on p.Tyr622* iNs with and without CDP-choline treatment (Figure~\ref{fig:choline_treatment}I,J). CDP-choline treatment significantly increased uncoupled respiration in p.Tyr622* iNs, restoring it to WT levels (Figure~\ref{fig:main_choline}G), with no significant change in spare respiratory capacity (Figure~\ref{fig:choline_treatment}K). Consistent with this finding, both TMRM staining (Figure~\ref{fig:main_choline}H) and MitoHealth fluorescence per NeuN-positive surface (Figure~\ref{fig:choline_treatment}L) confirmed a decrease in the mitochondrial membrane potential (ΔѰm) in treated cells. Additionally, CDP-choline treatment significantly decreased Cellrox fluorescence (Figure~\ref{fig:main_choline}I), indicating a reduction in oxidative stress. 

\subsubsection{CDP-Choline Ameliorates AD-Associated Phenotypes in Cortical Organoids}
Next, we tested whether CDP-choline treatment could improve key AD-associated phenotypes, since previous studies have linked ABCA7 dysfunction to altered amyloid-beta (Aβ) processing \cite{Satoh2015-yu,Sakae2016-uy,Bamji-Mirza2018-xt,Chan2008-qu,De_Roeck2018-fw}. Indeed, p.Tyr622* iNs secreted higher levels of Aβ40 and Aβ42 into the media, as measured by enzyme-linked immunosorbent assay (ELISA), although absolute levels remained relatively low (Figure~\ref{fig:neurospheroid_figure}A). To study CDP-choline's effects in a model with stronger pathology, we differentiated p.Tyr622* and WT lines into cortical organoids matured for $\approx 6$ months (Figure~\ref{fig:neurospheroid_figure}B), a stage at which we observed robust Aβ secretion (approximately two- to four-fold higher levels of Aβ40 and Aβ42 compared to iNs)(Figure~\ref{fig:neurospheroid_figure}C). Treatment with 1 mM CDP-choline for four weeks reduced Aβ40 and Aβ42 secretion from p.Tyr622* organoids to WT levels (Figure~\ref{fig:main_choline}K). This effect was not observed at lower concentrations or shorter treatment durations (Figure~\ref{fig:neurospheroid_figure}C). Additionally, treatment of dissociated cortical organoids with 100 $\mu$M CDP-choline for two weeks significantly reduced neuronal hyperexcitability in p.Tyr622* organoids, as shown by electrophysiology (Figure~\ref{fig:main_choline}L).

% \subsubsection{CDP-Choline Ameliorates AD-Associated Phenotypes in Cortical Organoids}
% Next, we sought to determine whether CDP-choline treatment could ameliorate key hallmarks ofAD pathology, given prior evidence linking ABCA7 dysfunction to altered amyloid-beta (Aβ) processing \cite{Satoh2015-yu,Sakae2016-uy,Bamji-Mirza2018-xt,Chan2008-qu,De_Roeck2018-fw}. Consistent with these findings, p.Tyr622* iNs showed increased secretion of Aβ40 and Aβ42 into the media, although absolute levels remained relatively low (Figure~\ref{fig:neurospheroid_figure}B). To evaluate CDP-choline effects within a model exhibiting more robust pathology, we differentiated p.Tyr622* and WT lines into cortical organoids matured for 6 months, a developmental stage characterized by strong Aβ secretion and confirmed neuronal maturity (Figure~\ref{fig:neurospheroid_figure}A,C). Four weeks of treatment with 1 µM CDP-choline reduced Aβ40 and Aβ42 secretion from p.Tyr622* organoids to WT levels (Figure~\ref{fig:main_choline}K), an effect not observed at lower concentrations or with shorter treatment durations (Figure~\ref{fig:neurospheroid_figure}C). Additionally, 1 µM CDP-choline significantly reversed neuronal hyperexcitability in p.Tyr622* organoids, as measured by cell-attached electrophysiological recordings (Figure~\ref{fig:main_choline}L).