\begin{figure}[ht]
    \begin{subfigure}[t]{0.3\textwidth}
        \begin{subfigure}[t]{\textwidth}
            \caption{}
            \includegraphics[width=\textwidth]{./extended_plots/glu50fs3_cartoon.png}        
        \end{subfigure}    
        \begin{subfigure}[t]{\textwidth}
            \caption{}
            \includegraphics[width=\textwidth]{./extended_plots/tyr622_cartoon.png}        
        \end{subfigure}  
    \end{subfigure}  
    \begin{subfigure}[t]{0.2\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/karyotypes.png}        
    \end{subfigure}  
    \begin{subfigure}[t]{0.4\textwidth}
        \begin{subfigure}[t]{\textwidth}
            \caption{}
            \includegraphics[width=\textwidth]{./extended_plots/iN_induction_cartoon.png}        
        \end{subfigure}    
        \begin{subfigure}[t]{\textwidth}
            \caption{}
            \includegraphics[width=\textwidth]{./extended_plots/iN_markers.png}        
        \end{subfigure}  
    \end{subfigure}    
    \begin{subfigure}[t]{0.33\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/in_out_current_trace.png}        
    \end{subfigure}  
    \begin{subfigure}[t]{0.33\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/out_current_barplot.png}        
    \end{subfigure}  
    \begin{subfigure}[t]{0.25\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/iN_resting_membrane_potential.png}        
    \end{subfigure}  
    \begin{subfigure}[t]{0.25\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/iN_rheobase.png}        
    \end{subfigure} 
    \hspace{1cm}
    \begin{subfigure}[t]{0.3\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/iN_AP_frequency.png}        
    \end{subfigure}  
    \hspace{1cm}
    \begin{subfigure}[t]{0.35\textwidth}
        \caption{}
        \includegraphics[width=\textwidth]{./extended_plots/iN_abeta_elisa.png}        
    \end{subfigure}  
    \caption{
        \textbf{Differentiating and Profiling iPSC-Derived Neurons Harboring ABCA7 PTC Variants.}\\[1ex]
        (A) Sanger sequencing chromatogram confirming single nucleotide insertion in ABCA7 exon 3 to introduce a premature termination codon into the isogenic iPSC line ABCA7 p.Glu50fs*3 using CRISPR-Cas9 gene editing. 
        (B) Sanger sequencing chromatogram confirming patient single nucleotide polymorphism in ABCA7 exon 15 to introduce a premature termination codon into the isogenic iPSC line ABCA7 p.Tyr622* using CRISPR-Cas9 gene editing. 
        (C) Normal karyotypes were observed for control, ABCA7 p.Glu50fs*3, and ABCA7 p.Tyr622* isogenic iPSC lines. 
        (D) iPSCs were plated at low density for NGN2 viral transduction. Expression of NGN2 was driven by doxycycline (DOX) induction with puromycin (PURO) selection, then re-plated to match neuronal densities. Neurons were maintained for 4 weeks (DIV 28) before experimentation (Created with BioRender.com). 
        (E) Neuronal marker gene expression in 2 and 4-week matured iNs. 
        (F) Representative sweeps of whole-cell current flow of inward (upper panel) and outward (lower panel) current recordings from WT 4-week-old neurons. 
        (G) Quantification of (F). 
        (H) Resting membrane potential (mV) of 4-week-old WT, ABCA7 p.Tyr622*, and ABCA7 p.Glufs*3 neurons. 
        (I) Rheobase (pA) of 4-week-old WT, ABCA7 p.Tyr622*, and ABCA7 p.Glufs*3 neurons. 
        (J) Action potential frequency of 4-week-old WT, ABCA7 p.Tyr622*, and ABCA7 p.Glufs*3 neurons with indicated current injections. For panels F-J: WT: $n=24$; Y622: $n=13$; G2: $n=23$. For all panels: $P*<0.05$, $P***<0.001$. Graphs are mean Â± SEM.
    }
    \label{fig:differentiating_iPSC_neurons}
\end{figure}