{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c1ca78aa-d1bc-4d33-b9e7-713f55044410",
   "metadata": {},
   "outputs": [],
   "source": [
    "# load data\n",
    "data = readRDS('../../ABCA7lof//processed_data/for_plotting/lipidomics.rds')\n",
    "key = read.csv('../../ABCA7lof//raw_data/ngn2_data/lipidkey.csv', sep = '\\t')\n",
    "key = key[!duplicated(key$Key),]\n",
    "rownames(key) = key$Key\n",
    "proteomics = read.csv('../processed_data/for_plotting/ABCA7_proteomics.csv')\n",
    "wgs = read.csv('../raw_data/metadata/WGS_sample_QC_info.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a2887285-3577-4485-8abf-767bc47acfed",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>projid</th><th scope=col>WGS_id</th><th scope=col>Source.Tissue.Type</th><th scope=col>Gender_match</th><th scope=col>Concordant_with_array</th><th scope=col>Concordant_with_RNAseq</th><th scope=col>Duplicate</th><th scope=col>QC</th><th scope=col>GQN</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>68539908</td><td>SM-CTEGU</td><td>Brain-Cerebellum</td><td>FALSE</td><td>   NA</td><td>FALSE</td><td>identical to SM-CTELW</td><td>Fail (discordant with RNA-seq;gender mismatch)  </td><td>8.8</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>20882194</td><td>SM-CJEI2</td><td>Whole Blood     </td><td> TRUE</td><td>   NA</td><td>FALSE</td><td>identical to SM-CTEFA</td><td>Fail (discordant with RNA-seq)                  </td><td>8.2</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>50101523</td><td>SM-CTEDY</td><td>Whole Blood     </td><td>FALSE</td><td>FALSE</td><td>   NA</td><td>NA                   </td><td>Fail (discordant with SNP array;gender mismatch)</td><td>7.3</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>50410283</td><td>SM-CTEGE</td><td>Brain-Cerebellum</td><td>FALSE</td><td>FALSE</td><td>   NA</td><td>identical to SM-CTEN4</td><td>Fail (discordant with SNP array;gender mismatch)</td><td>9.3</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>75175129</td><td>SM-CJFKP</td><td>Brain-DLPFC     </td><td>FALSE</td><td>FALSE</td><td>   NA</td><td>identical to SM-CTDVH</td><td>Fail (discordant with SNP array;gender mismatch)</td><td>4.4</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>10100736</td><td>SM-CTDVR</td><td>Brain-Cerebellum</td><td> TRUE</td><td>FALSE</td><td>   NA</td><td>identical to SM-CJFO9</td><td>Fail (discordant with SNP array)                </td><td>8.9</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & projid & WGS\\_id & Source.Tissue.Type & Gender\\_match & Concordant\\_with\\_array & Concordant\\_with\\_RNAseq & Duplicate & QC & GQN\\\\\n",
       "  & <int> & <chr> & <chr> & <lgl> & <lgl> & <lgl> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & 68539908 & SM-CTEGU & Brain-Cerebellum & FALSE &    NA & FALSE & identical to SM-CTELW & Fail (discordant with RNA-seq;gender mismatch)   & 8.8\\\\\n",
       "\t2 & 20882194 & SM-CJEI2 & Whole Blood      &  TRUE &    NA & FALSE & identical to SM-CTEFA & Fail (discordant with RNA-seq)                   & 8.2\\\\\n",
       "\t3 & 50101523 & SM-CTEDY & Whole Blood      & FALSE & FALSE &    NA & NA                    & Fail (discordant with SNP array;gender mismatch) & 7.3\\\\\n",
       "\t4 & 50410283 & SM-CTEGE & Brain-Cerebellum & FALSE & FALSE &    NA & identical to SM-CTEN4 & Fail (discordant with SNP array;gender mismatch) & 9.3\\\\\n",
       "\t5 & 75175129 & SM-CJFKP & Brain-DLPFC      & FALSE & FALSE &    NA & identical to SM-CTDVH & Fail (discordant with SNP array;gender mismatch) & 4.4\\\\\n",
       "\t6 & 10100736 & SM-CTDVR & Brain-Cerebellum &  TRUE & FALSE &    NA & identical to SM-CJFO9 & Fail (discordant with SNP array)                 & 8.9\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | projid &lt;int&gt; | WGS_id &lt;chr&gt; | Source.Tissue.Type &lt;chr&gt; | Gender_match &lt;lgl&gt; | Concordant_with_array &lt;lgl&gt; | Concordant_with_RNAseq &lt;lgl&gt; | Duplicate &lt;chr&gt; | QC &lt;chr&gt; | GQN &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | 68539908 | SM-CTEGU | Brain-Cerebellum | FALSE |    NA | FALSE | identical to SM-CTELW | Fail (discordant with RNA-seq;gender mismatch)   | 8.8 |\n",
       "| 2 | 20882194 | SM-CJEI2 | Whole Blood      |  TRUE |    NA | FALSE | identical to SM-CTEFA | Fail (discordant with RNA-seq)                   | 8.2 |\n",
       "| 3 | 50101523 | SM-CTEDY | Whole Blood      | FALSE | FALSE |    NA | NA                    | Fail (discordant with SNP array;gender mismatch) | 7.3 |\n",
       "| 4 | 50410283 | SM-CTEGE | Brain-Cerebellum | FALSE | FALSE |    NA | identical to SM-CTEN4 | Fail (discordant with SNP array;gender mismatch) | 9.3 |\n",
       "| 5 | 75175129 | SM-CJFKP | Brain-DLPFC      | FALSE | FALSE |    NA | identical to SM-CTDVH | Fail (discordant with SNP array;gender mismatch) | 4.4 |\n",
       "| 6 | 10100736 | SM-CTDVR | Brain-Cerebellum |  TRUE | FALSE |    NA | identical to SM-CJFO9 | Fail (discordant with SNP array)                 | 8.9 |\n",
       "\n"
      ],
      "text/plain": [
       "  projid   WGS_id   Source.Tissue.Type Gender_match Concordant_with_array\n",
       "1 68539908 SM-CTEGU Brain-Cerebellum   FALSE           NA                \n",
       "2 20882194 SM-CJEI2 Whole Blood         TRUE           NA                \n",
       "3 50101523 SM-CTEDY Whole Blood        FALSE        FALSE                \n",
       "4 50410283 SM-CTEGE Brain-Cerebellum   FALSE        FALSE                \n",
       "5 75175129 SM-CJFKP Brain-DLPFC        FALSE        FALSE                \n",
       "6 10100736 SM-CTDVR Brain-Cerebellum    TRUE        FALSE                \n",
       "  Concordant_with_RNAseq Duplicate            \n",
       "1 FALSE                  identical to SM-CTELW\n",
       "2 FALSE                  identical to SM-CTEFA\n",
       "3    NA                  NA                   \n",
       "4    NA                  identical to SM-CTEN4\n",
       "5    NA                  identical to SM-CTDVH\n",
       "6    NA                  identical to SM-CJFO9\n",
       "  QC                                               GQN\n",
       "1 Fail (discordant with RNA-seq;gender mismatch)   8.8\n",
       "2 Fail (discordant with RNA-seq)                   8.2\n",
       "3 Fail (discordant with SNP array;gender mismatch) 7.3\n",
       "4 Fail (discordant with SNP array;gender mismatch) 9.3\n",
       "5 Fail (discordant with SNP array;gender mismatch) 4.4\n",
       "6 Fail (discordant with SNP array)                 8.9"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(wgs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1cfd0590-99e5-4afa-936b-ab490ea7ba62",
   "metadata": {},
   "outputs": [],
   "source": [
    "data = merge(proteomics[proteomics$LOF==0,], wgs[,c('projid', 'WGS_id')], by='projid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7c97c28e-5a60-47ee-b105-8a5079c571c7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "Warning message in FUN(newX[, i], ...):\n",
      "“NAs introduced by coercion”\n",
      "ERROR while rich displaying an object: \u001b[1m\u001b[33mError\u001b[39m in `geom_bar()`:\u001b[22m\n",
      "\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Problem while computing aesthetics.\n",
      "\u001b[36mℹ\u001b[39m Error occurred in the 1st layer.\n",
      "\u001b[1mCaused by error in `[.data.frame`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m undefined columns selected\n",
      "\n",
      "Traceback:\n",
      "1. tryCatch(withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler), error = outer_handler)\n",
      "2. tryCatchList(expr, classes, parentenv, handlers)\n",
      "3. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "4. doTryCatch(return(expr), name, parentenv, handler)\n",
      "5. withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler)\n",
      "6. repr::mime2repr[[mime]](obj)\n",
      "7. repr_text.default(obj)\n",
      "8. paste(capture.output(print(obj)), collapse = \"\\n\")\n",
      "9. capture.output(print(obj))\n",
      "10. withVisible(...elt(i))\n",
      "11. print(obj)\n",
      "12. print.ggplot(obj)\n",
      "13. ggplot_build(x)\n",
      "14. ggplot_build.ggplot(x)\n",
      "15. by_layer(function(l, d) l$compute_aesthetics(d, plot), layers, \n",
      "  .     data, \"computing aesthetics\")\n",
      "16. try_fetch(for (i in seq_along(data)) {\n",
      "  .     out[[i]] <- f(l = layers[[i]], d = data[[i]])\n",
      "  . }, error = function(cnd) {\n",
      "  .     cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .         call = layers[[i]]$constructor, parent = cnd)\n",
      "  . })\n",
      "17. tryCatch(withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }), stackOverflowError = handlers[[1L]])\n",
      "18. tryCatchList(expr, classes, parentenv, handlers)\n",
      "19. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "20. doTryCatch(return(expr), name, parentenv, handler)\n",
      "21. withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . })\n",
      "22. f(l = layers[[i]], d = data[[i]])\n",
      "23. l$compute_aesthetics(d, plot)\n",
      "24. compute_aesthetics(..., self = self)\n",
      "25. lapply(aesthetics, eval_tidy, data = data, env = env)\n",
      "26. FUN(X[[i]], ...)\n",
      "27. factor(variant_categories[, \"Lof\"])\n",
      "28. variant_categories[, \"Lof\"]\n",
      "29. `[.data.frame`(variant_categories, , \"Lof\")\n",
      "30. stop(\"undefined columns selected\")\n",
      "31. .handleSimpleError(function (cnd) \n",
      "  . {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }, \"undefined columns selected\", base::quote(`[.data.frame`(variant_categories, \n",
      "  .     , \"Lof\")))\n",
      "32. h(simpleError(msg, call))\n",
      "33. handlers[[1L]](cnd)\n",
      "34. cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .     call = layers[[i]]$constructor, parent = cnd)\n",
      "35. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n",
      "  .     .frame = .frame)\n",
      "36. signal_abort(cnd, .file)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR while rich displaying an object: \u001b[1m\u001b[33mError\u001b[39m in `geom_bar()`:\u001b[22m\n",
      "\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Problem while computing aesthetics.\n",
      "\u001b[36mℹ\u001b[39m Error occurred in the 1st layer.\n",
      "\u001b[1mCaused by error in `[.data.frame`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m undefined columns selected\n",
      "\n",
      "Traceback:\n",
      "1. tryCatch(withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler), error = outer_handler)\n",
      "2. tryCatchList(expr, classes, parentenv, handlers)\n",
      "3. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "4. doTryCatch(return(expr), name, parentenv, handler)\n",
      "5. withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler)\n",
      "6. repr::mime2repr[[mime]](obj)\n",
      "7. repr_text.default(obj)\n",
      "8. paste(capture.output(print(obj)), collapse = \"\\n\")\n",
      "9. capture.output(print(obj))\n",
      "10. withVisible(...elt(i))\n",
      "11. print(obj)\n",
      "12. print.ggplot(obj)\n",
      "13. ggplot_build(x)\n",
      "14. ggplot_build.ggplot(x)\n",
      "15. by_layer(function(l, d) l$compute_aesthetics(d, plot), layers, \n",
      "  .     data, \"computing aesthetics\")\n",
      "16. try_fetch(for (i in seq_along(data)) {\n",
      "  .     out[[i]] <- f(l = layers[[i]], d = data[[i]])\n",
      "  . }, error = function(cnd) {\n",
      "  .     cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .         call = layers[[i]]$constructor, parent = cnd)\n",
      "  . })\n",
      "17. tryCatch(withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }), stackOverflowError = handlers[[1L]])\n",
      "18. tryCatchList(expr, classes, parentenv, handlers)\n",
      "19. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "20. doTryCatch(return(expr), name, parentenv, handler)\n",
      "21. withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . })\n",
      "22. f(l = layers[[i]], d = data[[i]])\n",
      "23. l$compute_aesthetics(d, plot)\n",
      "24. compute_aesthetics(..., self = self)\n",
      "25. lapply(aesthetics, eval_tidy, data = data, env = env)\n",
      "26. FUN(X[[i]], ...)\n",
      "27. factor(variant_categories[, \"ABCA7 p.1527\"])\n",
      "28. variant_categories[, \"ABCA7 p.1527\"]\n",
      "29. `[.data.frame`(variant_categories, , \"ABCA7 p.1527\")\n",
      "30. stop(\"undefined columns selected\")\n",
      "31. .handleSimpleError(function (cnd) \n",
      "  . {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }, \"undefined columns selected\", base::quote(`[.data.frame`(variant_categories, \n",
      "  .     , \"ABCA7 p.1527\")))\n",
      "32. h(simpleError(msg, call))\n",
      "33. handlers[[1L]](cnd)\n",
      "34. cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .     call = layers[[i]]$constructor, parent = cnd)\n",
      "35. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n",
      "  .     .frame = .frame)\n",
      "36. signal_abort(cnd, .file)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# load Henne Holstege et al data\n",
    "df = read.csv('../processed_data/holstege_et_al//41588_2022_1208_MOESM4_ESM.csv', skip = 2, header = TRUE)\n",
    "\n",
    "# merge with ABCA7 variant info\n",
    "variant_info = read.csv('../raw_data/ROSMAP_WGS/HIGHandMED_coding_annotations_syn11724057_subset.csv', check.names=FALSE)\n",
    "variant_info$Variant = paste0(variant_info$CHROM,':',variant_info$POS,':',variant_info$REF_x, '>', variant_info$ALT_0)\n",
    "df = df[,c('Variant', 'REVEL', 'LOF')]\n",
    "var_info = merge(variant_info, df, by = 'Variant')\n",
    "var_info = var_info[var_info$GENE=='ABCA7',]\n",
    "\n",
    "genotypes = var_info[,unlist(lapply(names(var_info), function(x) sum(startsWith(x, c('ROS', 'MAP', 'SM')))))>0]\n",
    "genotypes[genotypes=='0/0']=0\n",
    "genotypes[genotypes=='0/1']=1\n",
    "genotypes[genotypes=='1/0']=1\n",
    "genotypes[genotypes=='1/1']=2\n",
    "df = apply(genotypes,2, function(x){as.numeric(x)})\n",
    "\n",
    "variant_categories = as.data.frame(colSums(df[var_info$LOF.y==1,]))\n",
    "temp = as.numeric(unlist(genotypes[var_info$ID=='rs3752246',]))                                  \n",
    "temp = -1*(temp-2)\n",
    "variant_categories$rs3752246 = temp\n",
    "\n",
    "# Create a pie chart\n",
    "percentage_col1_1 <- mean(variant_categories$Lof == 1) * 100\n",
    "\n",
    "pdf('../pdf_figures/percentage_lof.pdf')\n",
    "ggplot(variant_categories, aes(x = \"\",fill = factor(variant_categories[,'Lof']), color=factor(variant_categories[,'Lof']))) +\n",
    "  geom_bar(width = 1, stat = \"count\") +\n",
    "  coord_polar(theta = \"y\") +   scale_color_manual(values = c(\"white\", \"white\")) +  # Set black outline\n",
    "\n",
    "  scale_fill_manual(values = c(\"grey\", \"red\"), name='Lof') +   geom_text(x = 1, y = 0, label = paste0(round(percentage_col1_1, 1), \"%\"), color = \"white\", size = 8) + theme_classic() +   theme_void() +   theme(legend.text=element_text(size=15)) \n",
    "\n",
    "dev.off()\n",
    "\n",
    "# Create a pie chart\n",
    "\n",
    "pdf('../pdf_figures/percentage_common.pdf')\n",
    "percentage_col1_1 <- mean(variant_categories$common == 1) * 100\n",
    "\n",
    "ggplot(variant_categories, aes(x = \"\",fill = factor(variant_categories[,'ABCA7 p.1527']), color=factor(variant_categories[,'ABCA7 p.1527']))) +\n",
    "  geom_bar(width = 1, stat = \"count\") +\n",
    "  coord_polar(theta = \"y\") +   scale_color_manual(values = c(\"white\", \"white\")) +  # Set black outline\n",
    "\n",
    "  scale_fill_manual(values = c(\"grey\", \"salmon\"), name='ABCA7 p.1527') +   geom_text(x = 1, y = 0, label = paste0(round(percentage_col1_1, 1), \"%\"), color = \"white\", size = 8) + theme_classic() +   theme_void() +   theme(legend.text=element_text(size=15)) \n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "49696694-b052-42cf-92a2-61a321971655",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1196 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>colSums(df[var_info$LOF.y == 1, ])</th><th scope=col>rs3752246</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>MAP15387421</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP22868024</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP26637867</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP29629849</th><td>1</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP33332646</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP34726040</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP46246604</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP46251007</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50103679</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP50104134</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP50104846</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP50106442</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50106992</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50108462</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP50301099</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50302680</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50402431</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP50409406</th><td>1</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP61344957</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP85980779</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP87264456</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP89164957</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>MAP93787649</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP95330358</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>MAP95453354</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>ROS10524640</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ROS11430815</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>ROS11697592</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>ROS15114174</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>ROS15738428</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEML</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMM</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMN</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMO</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMP</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMQ</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMR</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMS</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMT</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMU</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMW</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMX</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMY</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEMZ</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN1</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN2</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN3</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN4</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN5</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN6</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN7</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN8</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEN9</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENA</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENC</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTEND</th><td>0</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENE</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENF</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENG</th><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>SM-CTENH</th><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1196 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & colSums(df{[}var\\_info\\$LOF.y == 1, {]}) & rs3752246\\\\\n",
       "  & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tMAP15387421 & 0 & 0\\\\\n",
       "\tMAP22868024 & 0 & 1\\\\\n",
       "\tMAP26637867 & 0 & 0\\\\\n",
       "\tMAP29629849 & 1 & 0\\\\\n",
       "\tMAP33332646 & 0 & 0\\\\\n",
       "\tMAP34726040 & 0 & 1\\\\\n",
       "\tMAP46246604 & 0 & 0\\\\\n",
       "\tMAP46251007 & 0 & 0\\\\\n",
       "\tMAP50103679 & 0 & 1\\\\\n",
       "\tMAP50104134 & 0 & 1\\\\\n",
       "\tMAP50104846 & 0 & 1\\\\\n",
       "\tMAP50106442 & 0 & 0\\\\\n",
       "\tMAP50106992 & 0 & 0\\\\\n",
       "\tMAP50108462 & 0 & 1\\\\\n",
       "\tMAP50301099 & 0 & 0\\\\\n",
       "\tMAP50302680 & 0 & 0\\\\\n",
       "\tMAP50402431 & 0 & 0\\\\\n",
       "\tMAP50409406 & 1 & 0\\\\\n",
       "\tMAP61344957 & 0 & 1\\\\\n",
       "\tMAP85980779 & 0 & 0\\\\\n",
       "\tMAP87264456 & 0 & 0\\\\\n",
       "\tMAP89164957 & 0 & 1\\\\\n",
       "\tMAP93787649 & 0 & 0\\\\\n",
       "\tMAP95330358 & 0 & 0\\\\\n",
       "\tMAP95453354 & 0 & 0\\\\\n",
       "\tROS10524640 & 0 & 1\\\\\n",
       "\tROS11430815 & 0 & 0\\\\\n",
       "\tROS11697592 & 0 & 0\\\\\n",
       "\tROS15114174 & 0 & 1\\\\\n",
       "\tROS15738428 & 0 & 0\\\\\n",
       "\t⋮ & ⋮ & ⋮\\\\\n",
       "\tSM-CTEML & 0 & 0\\\\\n",
       "\tSM-CTEMM & 0 & 1\\\\\n",
       "\tSM-CTEMN & 0 & 0\\\\\n",
       "\tSM-CTEMO & 0 & 0\\\\\n",
       "\tSM-CTEMP & 0 & 0\\\\\n",
       "\tSM-CTEMQ & 0 & 1\\\\\n",
       "\tSM-CTEMR & 0 & 1\\\\\n",
       "\tSM-CTEMS & 0 & 0\\\\\n",
       "\tSM-CTEMT & 0 & 0\\\\\n",
       "\tSM-CTEMU & 0 & 0\\\\\n",
       "\tSM-CTEMW & 0 & 0\\\\\n",
       "\tSM-CTEMX & 0 & 1\\\\\n",
       "\tSM-CTEMY & 0 & 0\\\\\n",
       "\tSM-CTEMZ & 0 & 1\\\\\n",
       "\tSM-CTEN1 & 0 & 1\\\\\n",
       "\tSM-CTEN2 & 0 & 0\\\\\n",
       "\tSM-CTEN3 & 0 & 0\\\\\n",
       "\tSM-CTEN4 & 0 & 0\\\\\n",
       "\tSM-CTEN5 & 0 & 1\\\\\n",
       "\tSM-CTEN6 & 0 & 0\\\\\n",
       "\tSM-CTEN7 & 0 & 0\\\\\n",
       "\tSM-CTEN8 & 0 & 0\\\\\n",
       "\tSM-CTEN9 & 0 & 0\\\\\n",
       "\tSM-CTENA & 0 & 1\\\\\n",
       "\tSM-CTENC & 0 & 0\\\\\n",
       "\tSM-CTEND & 0 & 1\\\\\n",
       "\tSM-CTENE & 0 & 0\\\\\n",
       "\tSM-CTENF & 0 & 0\\\\\n",
       "\tSM-CTENG & 0 & 0\\\\\n",
       "\tSM-CTENH & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1196 × 2\n",
       "\n",
       "| <!--/--> | colSums(df[var_info$LOF.y == 1, ]) &lt;dbl&gt; | rs3752246 &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| MAP15387421 | 0 | 0 |\n",
       "| MAP22868024 | 0 | 1 |\n",
       "| MAP26637867 | 0 | 0 |\n",
       "| MAP29629849 | 1 | 0 |\n",
       "| MAP33332646 | 0 | 0 |\n",
       "| MAP34726040 | 0 | 1 |\n",
       "| MAP46246604 | 0 | 0 |\n",
       "| MAP46251007 | 0 | 0 |\n",
       "| MAP50103679 | 0 | 1 |\n",
       "| MAP50104134 | 0 | 1 |\n",
       "| MAP50104846 | 0 | 1 |\n",
       "| MAP50106442 | 0 | 0 |\n",
       "| MAP50106992 | 0 | 0 |\n",
       "| MAP50108462 | 0 | 1 |\n",
       "| MAP50301099 | 0 | 0 |\n",
       "| MAP50302680 | 0 | 0 |\n",
       "| MAP50402431 | 0 | 0 |\n",
       "| MAP50409406 | 1 | 0 |\n",
       "| MAP61344957 | 0 | 1 |\n",
       "| MAP85980779 | 0 | 0 |\n",
       "| MAP87264456 | 0 | 0 |\n",
       "| MAP89164957 | 0 | 1 |\n",
       "| MAP93787649 | 0 | 0 |\n",
       "| MAP95330358 | 0 | 0 |\n",
       "| MAP95453354 | 0 | 0 |\n",
       "| ROS10524640 | 0 | 1 |\n",
       "| ROS11430815 | 0 | 0 |\n",
       "| ROS11697592 | 0 | 0 |\n",
       "| ROS15114174 | 0 | 1 |\n",
       "| ROS15738428 | 0 | 0 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| SM-CTEML | 0 | 0 |\n",
       "| SM-CTEMM | 0 | 1 |\n",
       "| SM-CTEMN | 0 | 0 |\n",
       "| SM-CTEMO | 0 | 0 |\n",
       "| SM-CTEMP | 0 | 0 |\n",
       "| SM-CTEMQ | 0 | 1 |\n",
       "| SM-CTEMR | 0 | 1 |\n",
       "| SM-CTEMS | 0 | 0 |\n",
       "| SM-CTEMT | 0 | 0 |\n",
       "| SM-CTEMU | 0 | 0 |\n",
       "| SM-CTEMW | 0 | 0 |\n",
       "| SM-CTEMX | 0 | 1 |\n",
       "| SM-CTEMY | 0 | 0 |\n",
       "| SM-CTEMZ | 0 | 1 |\n",
       "| SM-CTEN1 | 0 | 1 |\n",
       "| SM-CTEN2 | 0 | 0 |\n",
       "| SM-CTEN3 | 0 | 0 |\n",
       "| SM-CTEN4 | 0 | 0 |\n",
       "| SM-CTEN5 | 0 | 1 |\n",
       "| SM-CTEN6 | 0 | 0 |\n",
       "| SM-CTEN7 | 0 | 0 |\n",
       "| SM-CTEN8 | 0 | 0 |\n",
       "| SM-CTEN9 | 0 | 0 |\n",
       "| SM-CTENA | 0 | 1 |\n",
       "| SM-CTENC | 0 | 0 |\n",
       "| SM-CTEND | 0 | 1 |\n",
       "| SM-CTENE | 0 | 0 |\n",
       "| SM-CTENF | 0 | 0 |\n",
       "| SM-CTENG | 0 | 0 |\n",
       "| SM-CTENH | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "            colSums(df[var_info$LOF.y == 1, ]) rs3752246\n",
       "MAP15387421 0                                  0        \n",
       "MAP22868024 0                                  1        \n",
       "MAP26637867 0                                  0        \n",
       "MAP29629849 1                                  0        \n",
       "MAP33332646 0                                  0        \n",
       "MAP34726040 0                                  1        \n",
       "MAP46246604 0                                  0        \n",
       "MAP46251007 0                                  0        \n",
       "MAP50103679 0                                  1        \n",
       "MAP50104134 0                                  1        \n",
       "MAP50104846 0                                  1        \n",
       "MAP50106442 0                                  0        \n",
       "MAP50106992 0                                  0        \n",
       "MAP50108462 0                                  1        \n",
       "MAP50301099 0                                  0        \n",
       "MAP50302680 0                                  0        \n",
       "MAP50402431 0                                  0        \n",
       "MAP50409406 1                                  0        \n",
       "MAP61344957 0                                  1        \n",
       "MAP85980779 0                                  0        \n",
       "MAP87264456 0                                  0        \n",
       "MAP89164957 0                                  1        \n",
       "MAP93787649 0                                  0        \n",
       "MAP95330358 0                                  0        \n",
       "MAP95453354 0                                  0        \n",
       "ROS10524640 0                                  1        \n",
       "ROS11430815 0                                  0        \n",
       "ROS11697592 0                                  0        \n",
       "ROS15114174 0                                  1        \n",
       "ROS15738428 0                                  0        \n",
       "⋮           ⋮                                  ⋮        \n",
       "SM-CTEML    0                                  0        \n",
       "SM-CTEMM    0                                  1        \n",
       "SM-CTEMN    0                                  0        \n",
       "SM-CTEMO    0                                  0        \n",
       "SM-CTEMP    0                                  0        \n",
       "SM-CTEMQ    0                                  1        \n",
       "SM-CTEMR    0                                  1        \n",
       "SM-CTEMS    0                                  0        \n",
       "SM-CTEMT    0                                  0        \n",
       "SM-CTEMU    0                                  0        \n",
       "SM-CTEMW    0                                  0        \n",
       "SM-CTEMX    0                                  1        \n",
       "SM-CTEMY    0                                  0        \n",
       "SM-CTEMZ    0                                  1        \n",
       "SM-CTEN1    0                                  1        \n",
       "SM-CTEN2    0                                  0        \n",
       "SM-CTEN3    0                                  0        \n",
       "SM-CTEN4    0                                  0        \n",
       "SM-CTEN5    0                                  1        \n",
       "SM-CTEN6    0                                  0        \n",
       "SM-CTEN7    0                                  0        \n",
       "SM-CTEN8    0                                  0        \n",
       "SM-CTEN9    0                                  0        \n",
       "SM-CTENA    0                                  1        \n",
       "SM-CTENC    0                                  0        \n",
       "SM-CTEND    0                                  1        \n",
       "SM-CTENE    0                                  0        \n",
       "SM-CTENF    0                                  0        \n",
       "SM-CTENG    0                                  0        \n",
       "SM-CTENH    0                                  1        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "variant_categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a5d3ae67-06a7-4f29-a934-72974ca4a878",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR while rich displaying an object: \u001b[1m\u001b[33mError\u001b[39m in `geom_bar()`:\u001b[22m\n",
      "\u001b[1m\u001b[22m\u001b[33m!\u001b[39m Problem while computing aesthetics.\n",
      "\u001b[36mℹ\u001b[39m Error occurred in the 1st layer.\n",
      "\u001b[1mCaused by error in `[.data.frame`:\u001b[22m\n",
      "\u001b[33m!\u001b[39m undefined columns selected\n",
      "\n",
      "Traceback:\n",
      "1. tryCatch(withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler), error = outer_handler)\n",
      "2. tryCatchList(expr, classes, parentenv, handlers)\n",
      "3. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "4. doTryCatch(return(expr), name, parentenv, handler)\n",
      "5. withCallingHandlers({\n",
      " .     if (!mime %in% names(repr::mime2repr)) \n",
      " .         stop(\"No repr_* for mimetype \", mime, \" in repr::mime2repr\")\n",
      " .     rpr <- repr::mime2repr[[mime]](obj)\n",
      " .     if (is.null(rpr)) \n",
      " .         return(NULL)\n",
      " .     prepare_content(is.raw(rpr), rpr)\n",
      " . }, error = error_handler)\n",
      "6. repr::mime2repr[[mime]](obj)\n",
      "7. repr_text.default(obj)\n",
      "8. paste(capture.output(print(obj)), collapse = \"\\n\")\n",
      "9. capture.output(print(obj))\n",
      "10. withVisible(...elt(i))\n",
      "11. print(obj)\n",
      "12. print.ggplot(obj)\n",
      "13. ggplot_build(x)\n",
      "14. ggplot_build.ggplot(x)\n",
      "15. by_layer(function(l, d) l$compute_aesthetics(d, plot), layers, \n",
      "  .     data, \"computing aesthetics\")\n",
      "16. try_fetch(for (i in seq_along(data)) {\n",
      "  .     out[[i]] <- f(l = layers[[i]], d = data[[i]])\n",
      "  . }, error = function(cnd) {\n",
      "  .     cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .         call = layers[[i]]$constructor, parent = cnd)\n",
      "  . })\n",
      "17. tryCatch(withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }), stackOverflowError = handlers[[1L]])\n",
      "18. tryCatchList(expr, classes, parentenv, handlers)\n",
      "19. tryCatchOne(expr, names, parentenv, handlers[[1L]])\n",
      "20. doTryCatch(return(expr), name, parentenv, handler)\n",
      "21. withCallingHandlers(expr, condition = function(cnd) {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . })\n",
      "22. f(l = layers[[i]], d = data[[i]])\n",
      "23. l$compute_aesthetics(d, plot)\n",
      "24. compute_aesthetics(..., self = self)\n",
      "25. lapply(aesthetics, eval_tidy, data = data, env = env)\n",
      "26. FUN(X[[i]], ...)\n",
      "27. factor(variant_categories[, \"ABCA7 p.1527\"])\n",
      "28. variant_categories[, \"ABCA7 p.1527\"]\n",
      "29. `[.data.frame`(variant_categories, , \"ABCA7 p.1527\")\n",
      "30. stop(\"undefined columns selected\")\n",
      "31. .handleSimpleError(function (cnd) \n",
      "  . {\n",
      "  .     {\n",
      "  .         .__handler_frame__. <- TRUE\n",
      "  .         .__setup_frame__. <- frame\n",
      "  .         if (inherits(cnd, \"message\")) {\n",
      "  .             except <- c(\"warning\", \"error\")\n",
      "  .         }\n",
      "  .         else if (inherits(cnd, \"warning\")) {\n",
      "  .             except <- \"error\"\n",
      "  .         }\n",
      "  .         else {\n",
      "  .             except <- \"\"\n",
      "  .         }\n",
      "  .     }\n",
      "  .     while (!is_null(cnd)) {\n",
      "  .         if (inherits(cnd, \"error\")) {\n",
      "  .             out <- handlers[[1L]](cnd)\n",
      "  .             if (!inherits(out, \"rlang_zap\")) \n",
      "  .                 throw(out)\n",
      "  .         }\n",
      "  .         inherit <- .subset2(.subset2(cnd, \"rlang\"), \"inherit\")\n",
      "  .         if (is_false(inherit)) {\n",
      "  .             return()\n",
      "  .         }\n",
      "  .         cnd <- .subset2(cnd, \"parent\")\n",
      "  .     }\n",
      "  . }, \"undefined columns selected\", base::quote(`[.data.frame`(variant_categories, \n",
      "  .     , \"ABCA7 p.1527\")))\n",
      "32. h(simpleError(msg, call))\n",
      "33. handlers[[1L]](cnd)\n",
      "34. cli::cli_abort(c(\"Problem while {step}.\", i = \"Error occurred in the {ordinal(i)} layer.\"), \n",
      "  .     call = layers[[i]]$constructor, parent = cnd)\n",
      "35. rlang::abort(message, ..., call = call, use_cli_format = TRUE, \n",
      "  .     .frame = .frame)\n",
      "36. signal_abort(cnd, .file)\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in file(con, \"rb\"): cannot open the connection\n",
     "output_type": "error",
     "traceback": [
      "Error in file(con, \"rb\"): cannot open the connection\nTraceback:\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "percentage_col1_1 <- mean(variant_categories$common == 1) * 100\n",
    "\n",
    "ggplot(variant_categories, aes(x = \"\",fill = factor(variant_categories[,'ABCA7 p.1527']), color=factor(variant_categories[,'ABCA7 p.1527']))) +\n",
    "  geom_bar(width = 1, stat = \"count\") +\n",
    "  coord_polar(theta = \"y\") +   scale_color_manual(values = c(\"white\", \"white\")) +  # Set black outline\n",
    "\n",
    "  scale_fill_manual(values = c(\"grey\", \"salmon\"), name='ABCA7 p.1527') +   geom_text(x = 1, y = 0, label = paste0(round(percentage_col1_1, 1), \"%\"), color = \"white\", size = 8) + theme_classic() +   theme_void() +   theme(legend.text=element_text(size=15)) \n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r_env]",
   "language": "R",
   "name": "conda-env-.conda-r_env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
