\textbf{Extended Data table legends}\newline\newline
\textbf{Supplementary Table 1.} Annotation of ABCA7 loss of function variants used in this study.\newline
\textbf{Supplementary Table 2.} ABCA7 PTC variants identified in the snRNA-seq cohort\newline
\textbf{Supplementary Table 3.} External datasets used. \newline
\textbf{Supplementary Table 4.} Metadata for selected individuals from the ROSMAP cohort used in this study \newline
\textbf{Supplementary Table 5.} snRNA-seq quality control metrics \newline
\textbf{Supplementary Table 6.} Differential gene expression statistics by cell type \newline
\textbf{Supplementary Table 7.} Gene Ontology (GO) pathway enrichment results \newline
\textbf{Supplementary Table 8.} WikiPathways enrichment results in excitatory neurons with K/L gene cluster assignments \newline
\textbf{Supplementary Table 9.} RNA-seq differential expression statistics in NGN2-induced neurons \newline
\textbf{Supplementary Table 10.} WikiPathways enrichment results in iNs with K/L gene cluster assignments \newline
\textbf{Supplementary Table 11.} Enrichments of MitoCarta mitochondrial pathways in WT vs p.Tyr622* mRNA \newline
\textbf{Supplementary Table 12.} Lipidomics differential abundance statistics in NGN2-induced neurons \newline
\textbf{Supplementary Table 13.} Targeted metabolomics on cells and media, with and without CDP-choline treatment \newline
\textbf{Supplementary Table 14.} Enrichments of MitoCarta mitochondrial pathways in p.Tyr622* + H20 vs p.Tyr622* + CDP-choline mRNA \newline
\textbf{Supplementary Table 15.} Experimentally-determined 3D ABCA7 structures used in molecular dynamics simulations \newline
\textbf{Supplementary Table 16.} PCR/Sanger sequencing (SS) primers \newline
\textbf{Supplementary Table 17.} Antibodies used \newline

