# Data availability
<table>
  <tr>
    <th colspan="2">Human postmortem – snRNAseq processing</th>
  </tr>
  <tr>
    <th>Description</th>
    <th>Details</th>
  </tr>
  <tr>
    <td>
      Raw sequencing reads, per-cell count matrices (raw and QC’ed), per-cell and per-gene metadata, and DEG input statistics
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/snRNAseq_processing">snRNAseq_processing</a><br>
      Available @: <a href="https://www.synapse.org/#!Synapse:syn53461705">raw and unredacted data</a> | 
      <a href="https://singlecell.broadinstitute.org/single_cell/study/SCP3182/a-single-cell-atlas-of-abca7-loss-of-function-in-human-brain#study-download">redacted data</a> | 
      <a href="https://osf.io/xegjk">input statistics</a>
    </td>
  </tr>
  <tr>
    <td>
      Reference cell-type marker definitions for adult brain cell types and excitatory layer annotations
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/snRNAseq_processing">snRNAseq_processing</a><br> | <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/snRNAseq_reviewer_analysis">snRNAseq_ex_layers</a><br>
      Available @: <a href="https://osf.io/5k8v2/files/osfstorage">cell-type markers</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Human postmortem – snRNAseq statistics</th>
  </tr>
  <tr>
    <td>
      Differential expression tables (per cell type) and pathway databases
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/snRNAseq_stats">snRNAseq_stats</a><br>
      Available @: <a href="https://osf.io/v6y3d/files/osfstorage">pathways</a> | 
      <a href="https://osf.io/5cnfy/wiki/home/">DEGs</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Human postmortem – gene–pathway partitioning</th>
  </tr>
  <tr>
    <td>
      Processed data supporting analyses in Figures 1–2
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/snRNAseq_score_partitioning">snRNAseq_score_partitioning</a><br>
      Available @: <a href="https://osf.io/n9a35/files/osfstorage">processed data and intermediate results</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Induced neurons – LCMS</th>
  </tr>
  <tr>
    <td>
      Lipidomic and metabolomic quantifications (WT, p.Glu50fs*3, p.Tyr622*, ±CDP-choline)
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/iN_LCMS">iN_LCMS</a><br>
      Available @: <a href="https://osf.io/mnysb/files/osfstorage">processed data</a>
    </td>
  </tr>
  <tr>
    <td>
      Raw lipidomic and metabolomic LCMS files
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/iN_LCMS">iN_LCMS</a><br>
      Available @: <a href="https://osf.io/u68k3/">raw data</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Induced neurons – bulk RNA-seq</th>
  </tr>
  <tr>
    <td>
      Raw reads, gene count matrices, and DEG tables (WT, p.Glu50fs*3, p.Tyr622*, ±CDP-choline)
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/tree/main/analyses/bulkRNAseq">bulkRNAseq</a><br>
      Available @: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE299277">raw data</a> | 
      <a href="https://osf.io/5cnfy/wiki/home/">DEGs</a> |
      <a href="https://osf.io/mnysb/files/osfstorage">other</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Induced neurons – O₂ consumption rates</th>
  </tr>
  <tr>
    <td>
      Raw Seahorse analyzer exports and processed quantifications
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/iN_O2_consumption">iN_O2_consumption</a><br>
      Available @: <a href="https://osf.io/kptbw/files/osfstorage">raw data</a> | 
      <a href="https://osf.io/mnysb/files/osfstorage">processed data</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Induced neurons / cortical organoids – Imaging</th>
  </tr>
  <tr>
    <td>
      Raw images and image quantifications
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/imaging">membrane_potential</a><br> | <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/imaging">cellrox</a><br> | <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/imaging">iN_markers</a><br> | <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/imaging">organoid_markers</a><br>

      Available @: <a href="https://osf.io/htb23/">raw images</a> | 
      <a href="https://osf.io/mnysb/files/osfstorage">quantifications</a>
    </td>
  </tr>

  <tr>
    <th colspan="2">Induced neurons / cortical organoids – Amyloid & Electrophysiology</th>
  </tr>
  <tr>
    <td>
      Amyloid ELISA data, iN ephys datasets, and cortical organoid ephys datasets
    </td>
    <td>
      Related code: <a href="https://github.com/djunamay/ABCA7lof2/blob/main/analyses/amyloid_ephys">amyloid_ephys</a><br>
      Available @: <a href="https://osf.io/szuvt/">ELISA</a> | 
      <a href="https://osf.io/jy3fe/">iN ephys</a> | <a href="https://osf.io/em92c/">cortical organoid ephys</a>
    </td>
  </tr>
</table>
