\subsubsection{Experimental Methods using human iPSCs} 

\paragraph{Culture and generation of human isogenic iPSCs.}
A control parental line was derived from a 75-year-old female (AG09173) with an APOE3/3 genotype by the Picower Institute for Learning and Memory iPSC Facility as first described \cite{Lin2018-zu}. Two ABCA7 LoF isogenic lines were derived from parental AG09173. ABCA7 p.Glu50fs*3, generated by Synthego (www.synthego.com), contains a premature termination codon in exon 3 (Figure~\ref{fig:snRNA_cohort}1A), which to our knowledge has not been discovered in patients, but is functionally analogous to patient loss-of-function mutations.

ABCA7 p.Tyr622* contains a patient-derived mutation (Y622*) \cite{De_Roeck2019-ee} and was generated in house by CRISPR-Cas9 genome editing. The CRISPR/Cas9-ABCA7-Y622* sgRNA plasmid was prepared followed by the published protocol \cite{Ran2013-lx}. In brief, a sgRNA sequence within 10 nucleotides from the target site was designed using the CRISPR/Cas9 Design Tool (http://crispr.mit.edu). The oligomer pairs (forward: 5’-CACCGCCCCTACAGCCACCCGGGCG-3’ and reverse: 5’-AAACCGCCCGGGTGGCTGTAGGGGC-3’) were annealed and cloned into pSpCas9-2A-GFP (PX458) plasmid (Addgene #48138). Plasmid DNA was submitted for Sanger sequencing to confirm correct ABCA7 sgRNA sequence (Figure~\ref{fig:snRNA_cohort}1B).

AG09173 iPSCs were dissociated with Accutase (Thermo Fisher Scientific) supplemented with 10 μM ROCK inhibitor (Tocris) for electroporation using Amaxa and Human Stem Cell Nucleofector Kit I (Lonza). ~5x106 cells were resuspended in 100 μl of reaction buffer supplemented with 7.5 μg of CRISPR/Cas9-ABCA7 sgRNA plasmid and 15 μg of single-strand oligodeoxynucleotide (ssODN) template (5’-GGTGCGCGCCCCCAGGCCAATCCAGGAGCTGCACCCTAAGCTCCCGTTGCCTCTCACAGCTGGGAGACATCCTCCCCTAGAGCCACCCGGGCGTCGTCTTCCTGTTCTTGGCAGCCTTCGCGGTGGCCACGGTGACCCAGAGCTTCCTGCTCAGCGCCTTCTTCTCCCGCGCCAACCTGG-3’). This reaction mixture was nucleofected with program A-23, resuspended with media supplemented with 10 μM ROCK inhibitor and seeded on MEF plates. Two days after electroporation, cells were dissociated and filtered through Falcon polystyrene test tubes (Corning #352235), transferred to Falcon polypropylene test tubes (Corning #352063) and sorted by BD FACS Aria IIU in FACS Facility at the Whitehead Institute and seeded as single cells in media supplemented with 1X Penicillin-Streptomycin (P/S, Gemini Bio-products) and 10 μM ROCK inhibitor. After sufficient colony growth, each colony was transferred in part to a 12-well plate while the remainder was collected and used to extract genomic DNA (Qiagen DNeasy Blood & Tissue Kit, Cat. No. 69504) and screen for the Y622* mutation by sanger sequencing.

All lines used were confirmed to have normal karyotypes before use and periodically reviewed (Cell Line Genetics) (Figure~\ref{fig:snRNA_cohort}0C). All human iPSCs were maintained at 37°C and 5\% CO2, in feeder-free conditions in mTeSR-1 medium (Cat #85850; STEMCELL Technologies) on Matrigel-coated plates (Cat # 354277; Corning; hESC-Qualified Matrix). iPSCs were passaged at 60–80\% confluence using ReLeSR (Cat# 05872; STEMCELL Technologies) and reseeded between 1:6 and 1:24 (depending on desired density) onto Matrigel-coated plates.

\paragraph{rTTA and NGN2 Virus production.}
HEK293T cells (ATCC, Cat#CRL-3216) were maintained in DMEM/F-12, GlutaMAX (ThermoFisher, Cat#10565018), 10\% fetal bovine serum (GeminiBio, SKU#100-106), 1\% MEM Non-essential amino acids (Sigma, Cat#M7145), 1\% sodium pyruvate (ThermoFisher, Cat#11360070), and 1\% Penicillin-Streptomycin (GeminiBio, SKU#400-109). Cells were passaged for maintenance with TrypLE (ThermoFisher, Cat#12605010) at 70-80\% confluence and reseeded 1:10 in 10 cm tissue culture plates.

For transfection, HEK293T cells were seeded at 5x106 cells per 10 cm plate. Transfection mixtures containing the components required for 3rd generation lentiviral production (per 10 cm dish: 10 µg EF1a-rtTA-Hygro (Addgene #66810) or pLV-TetO-hNGN2-eGFP-Puro (Addgene #79823), 5 µg pMDL g/pRRE, 2.5 µg pRSV-Rev, 2.5 µg MD2.G, and 48 µL polyethyleneimine (1 mg/mL) in 600 uL OptiMEM (Fisher, Cat#51-985-034)). Mixtures were inverted 10X and incubated at RT for 20 min, then added dropwise to the dish. Transfection media was removed 16h later and replaced with 10 mL fresh media. Three days after transfection, media was collected and centrifuged at 3000 xg for 5 min at 4°C to pellet any contaminating cells. Supernatant was transferred to sterile Millex glass ultracentrifuge tubes and centrifuged at 25,000 rpm for 2 hours using a SW32Ti rotor in a Beckman Optima L-90K Ultracentrifuge. The pellets were resuspended in 1 mL PBS per 10 cm plate, and stored at -80°C until use.

\paragraph{Lentivirus-mediated NGN2 induction in iPSCs and drug treatments.}

iPSCs were dissociated into single cell suspension with Cell Dissociation Buffer (Life Technologies, Cat#13151-014), centrifuged at 300 xg for 5 min, and resuspended in mTeSR1 media with Rock inhibitor (Rockout; Abcam, ab285418). Single-cell suspension was plated in a 6-well plate coated with Matrigel for an optimized seeding density of 50-60\% confluence 24 hours after plating. One day after plating, cells were co-transduced with 80 µL pLV-TetO-hNGN2-eGFP-Puro and 80 µL EF1a-rtTA-Hygro added in 1 mL fresh media per well and incubated overnight at 37°C. NGN2 expression was then induced 24 with addition of 2 mL fresh media supplemented with doxycycline (DOX, 1 µg/mL, final concentration) and Rock inhibitor. Puromycin (1 µg/mL) selection of non-NGN2 expressing cells was performed with media change 24 hours after induction, with continued DOX supplementation. After 24 hours of puromycin selection, immature neurons were re-plated onto PDL/laminin coated plates at 1x106 cells/well on 6-well plates or 5x104 cells/well on 96-well plates. Neurons were maintained in BrainPhys Neuronal Media (STEMCELL Technologies, Cat#05793) with Neurocult SM1 Neuronal Supplement (STEMCELL Technologies, Cat#05711), N2-supplement-A (STEMCELL Technologies, Cat#07152), and DOX (1 µg/mL) with half media changes every 3-4 days. Neuronal cultures were maintained for 28 days before experimentation.

iPSC-derived neurons were treated with cytidine 5’-diphosphocholine (CDP-choline; Millipore Sigma 30290) or DGAT1/2 inhibitors (PF-04620110; PF-06424439) to final concentrations of 100 µM and 1 µM, respectively, beginning at day 14 and repeated with each media change until 28 days matured.

\paragraph{Lipidomics and Metabolomics of iPSC-derived neurons.}
iPSC-derived neurons were washed once with cold PBS (Fisher; Cat#MT21040CM) and lifted off plate with a cell scraper in 1 mL cold PBS . Cells were centrifuged at 2000 xg for 5 min. PBS was removed, and cells were resuspended in 2 mL cold methanol for biphasic extraction as described above for post-mortem brain tissue. In addition to transferring the chloroform phase for lipidomics, the aqueous phase was also collected for metabolomics. Lipidomics and Metabolomics was performed in collaboration with the Harvard Center for Mass Spectrometry (HCMS).

Lipidomics was performed as described above for post-mortem samples. For metabolomics, samples were dried under nitrogen flow and resuspended in acetonitrile 50\% in water. Resuspension volume was scaled to biomass, with the lowest biomass resuspended in 25 µL. 15 µL of each sample was transferred to glass microinserts for analysis. The remaining of the sample volumes were combined to create a pool sample used for MS2/MS3 data acquisition.

Samples were analyzed by LC-MS on a Vanquish LC coupled to an ID-X MS (Thermofisher Scientific). Five µL of sample was injected on a ZIC-pHILIC peek-coated column (150 mm x 2.1 mm, 5 micron particles, maintained at 40 °C, SigmaAldrich). Buffer A was 20 mM Ammonium Carbonate, 0.1\% Ammonium hydroxide in water and Buffer B was Acetonitrile 97\% in water. The LC program was as follows: starting at 93\% B, to 40\% B in 19 min, then to 0\% B in 9 min, maintained at 0\% B for 5 min, then back to 93\% B in 3 min and re-equilibrated at 93\% B for 9 min. The flow rate was maintained at 0.15 mL min-1, except for the first 30 seconds where the flow rate was uniformly ramped from 0.05 to 0.15 mL min-1. Data was acquired on the ID-X in switching polarities at 120000 resolution, with an AGC target of 1e5, and a m/z range of 65 to 1000. MS1 data is acquired in switching polarities for all samples. MS2 and MS3 data was acquired on the pool sample using the AquirX DeepScan function, with 5 reinjections, separately in positive and negative ion mode. 


\paragraph{Electrophysiological recordings.}
Cells were placed in a recording chamber and perfused with oxygenated artificial cerebrospinal fluid (ACSF) contains (in mM) 125 NaCl, 2.5 KCl, 1.2 NaH2PO4•H2O, 2.4 CaCl2•2H2O, 1.2 MgCl2•6H2O, 26 NaHCO3 and 11 D-Glucose at a constant rate of 2 mL/min at ~32°C. Cells were visualized using infrared differential interference contrast (IR-DIC) imaging on an Olympus BX-50WI microscope.

Recordings were performed using Axon Multiclamp 700B and Clampex 11.2 (Molecular Devices). Action potentials were generated by injecting various steps of currents using current clamp configuration.  Whole-cell currents were recorded from a holding potential of −80 mV by stepping to various voltages using voltage clamp configuration. Signals were filtered at 1 kHz using the amplifier’s four-pole, low-pass Bessel filter, digitized at 10 kHz with a Digidata 1550B interface (Molecular Devices). Pipette solution contained (in mM) 120 K gluconate, 5 KCl, 2 MgCl2•6H2O, 10 HEPES, 4 ATP, 0.2 GTP. pClamp 11.2 (Molecular Devices) and GraphPad Prism 10 software suites were used for data acquisition and analysis. Data are presented as means ± standard errors of means (SEM).

\paragraph{Aꞵ Enzyme linked immunosorbent assays.}
Media was collected from 4 week old iNs and flash frozen. ELISAs were performed on thawed media according to manufacturer’s instructions to measure Aꞵ40 (ThermoFisher Scientific, KHB3481) and Aꞵ42 (ThermoFisher Scientific, KHB3441) respectively. Amyloid levels were normalized to total protein content in media calculated using the Pierce BCA Protein assay (ThermoFisher Scientific, 23225) according to manufacturer’s instructions.

\paragraph{Immunocytochemistry, LipidSpot stain, and Mitochondrial Health live cell stain.}
Neurons were fixed in 4\% paraformaldehyde/4\% sucrose in PBS at 4°C for 15 min at room temperature, washed 3X with PBS, then permeabilized with 0.1\% Triton-X in PBS for 5 min at room temperature. Cells were blocked in 2\% Bovine Serum Albumin (BSA, Fisher Bioreagents, BP9703) in PBS for 20 min at room temperature, then incubated in primary antibodies diluted in blocking solution (1:500) overnight at 4°C. Cells were washed 3X for 5 min with PBS, then incubated in secondary antibodies diluted in blocking solution (1:1000) for 2 hours at room temperature. When used, LipidSpot (1:1000; Biotium 70069) was added along with secondary antibodies. Cells were washed 3X for 5 min with PBS, then incubated for 10 min with 1:2000 Hoechst 33342 (Invitrogen, H3570). Cells were washed 1X with PBS, and wells flooded with PBS for imaging.

HCS Mitochondrial Health kit (ThermoFisher, Cat#H10295) were used on live cells according to manufacturer’s protocols. In brief, CellROX dye was added directly to cell media for a final concentration of 5µM, while 50 uL of media containing 1.5 µL MitoHealth dye was added to each well. Both stains were incubated on live cells for 30 min at 37°C. Cells were fixed in 4\% paraformaldehyde in PBS, then co-stained according to immunocytochemistry procedures described above.

\paragraph{Seahorse Metabolic Assays.}
iPSCs were differentiated as described above directly on 96-well Agilent Seahorse XFe96/XF Pro cell culture microplates and matured for 28 days before assaying on a Seahorse XFe96 Analyzer. Seahorse XF Cell Mito Stress Test and Oxidation Stress Tests were performed according to manufacturer protocol with the following final drug concentration: Oligomycin, 2.5 µM; FCCP, 1 µM; Rotenone/Antimycin, 0.5 µM. 

\paragraph{Confocal Image acquisition.}
All confocal images were acquired on a Zeiss LSM900. Acquisition settings were kept constant within each imaging batch (where conditions of interest were uniformly distributed across plates). The minimum and maximum z-plane was manually determined for each culture well, to accommodate differences in culture thickness. Cultures were imaged at 1 µm intervals along the z-axis.