\renewcommand{\thefigure}{S\arabic{figure}}
\setcounter{figure}{0}

\clearpage

% Cohort metadata
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/snRNA_cohort.tex}
\clearpage
% Batch quality
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/snRNA_batch_quality.tex}
\clearpage
% Annotation
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/snRNA_annotation.tex}
\clearpage
% Gene scores
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/snRNAseq_gene_scores.tex}
\clearpage
% ABCA7 expression
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/abca7_expression.tex}
\clearpage
% Benchmarking clustering
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/benchmarking_clustering.tex}
\clearpage
% MD simulations
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/md_simulations.tex}
\clearpage
% Differentiating iPSC neurons
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/differentiating_iPSC_neurons.tex}
\clearpage
% bulk RNAseq supplement
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/bulk_RNAseq_supplement.tex}
\clearpage
% LCMS supplement
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/LCMS_supplement.tex}
\clearpage
% mitochondrial o2 consumption
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/mitochondrial_o2_consumption.tex}
\clearpage
% mitochondrial mmp
% \input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/mitochondrial_mmp.tex}
% \clearpage
% choline treatment
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/choline_treatment.tex}
\clearpage
% neurospheroid figure
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/neurospheroid_figure.tex}
\clearpage
% updated model
\input{/Users/djuna/Documents/ABCA7lof2/paper/appendix/extended_figures/model.tex}
\clearpage




% \begin{figure}[ht]
%    % \centerline{\includegraphics[width=\textwidth]{./extended_plots/quantification_Abeta42_iPSC_neurons.pdf}}
%     \caption{
%         \textbf{Quantification of Aβ42 in iPSC-Derived Neurons Harboring ABCA7 PTC Variants.}\\[1ex]
%         (A) Quantification of neuronal Aβ42 fluorescence intensity. P-values were computed by a linear mixed-effects model on per-NeuN+ volume averages, including well-of-origin as a random effect. $N = 16$ (WT; 2261 cells), $N=8$ (p.Tyr622*; 1466 cells), $N=6$ wells (p.Glu50fs*3; 999 cells) from 4-week-old iNs. Boxes indicate per-condition dataset quartiles, and whiskers extend to the most extreme data points not considered outliers (i.e., within 1.5 times the interquartile range from the first or third quartile). Individual data points represent per-well averages of cell-level intensities. 
%         (B) Representative images per condition showing mean-intensity projections of the entire image (NeuN+) and projections within NeuN+ volumes considered for quantification (Aβ42). Representative images for the Aβ42 channel were processed with condition-wide percentile-based background subtraction and thresholding. Representative images of cell soma underwent per-image percentile-based background subtraction and thresholding, reflecting the segmentation methodology.
%     }
%     \label{fig:quantification_Abeta42_iPSC_neurons}
% \end{figure}


