# Analysis of snRNAseq data from p.Ala1527Gly carriers

## Data availability
- Annotated snRNAseq counts were downloaded [here](https://personal.broadinstitute.org/cboix/ad427_data/Data/Processed_data/). Pseudo-bulk count measurements were generated by summing across neurons from Excitatory "Sets 1 - 3". The corresponding raw snRNAseq datasets are available for download from [Synapse](https://www.synapse.org/\#!Synapse:syn52293417).
- Follow instructions [here](https://github.com/djunamay/ROSMAPwgs) for information on accessing and processing the WGS data.
- We use data from [Holstege et al](https://www.nature.com/articles/s41588-022-01208-7) (`41588_2022_1208_MOESM4_ESM.csv `) to exclude samples with potentially damaging ABCA7 LoF variants from this analysis.
- Other data used here are outputs from [`analyses/snRNAseq_stats/compute_stats.ipynb`](https://github.com/djunamay/ABCA7lof2/blob/main/analyses/snRNAseq_stats/compute_stats.ipynb) and [`analyses/snRNAseq_score_partitioning/KL_clusters.ipynb`](https://github.com/djunamay/ABCA7lof2/blob/main/analyses/snRNAseq_score_partitioning/KL_clusters.ipynb)
  
## Code overview
- Run `./common_variant_degs.ipynb` to compute p.Ala1527Gly-associated DEGs across all excitatory neurons.
- Run `./common_variant_fgsa.ipynb` to compute FGSEA gene cluster enrichments for these DEGs.
- Run `./common_variant_plotting.ipynb` for plotting.
