\subsubsection*{Referee #4 (Remarks to the Author)}

The study by Maydell et al investigates the impact of ABCA7 LoF mutations on gene expression in patient brains with follow-up functional in vitro experiments in human neurons. The authors first performed snRNA-seq analysis to show that ABCA7 LoF cells in the brain have perturbed lipid metabolism and mitochondrial function, which might contribute to impaired neurotransmission. The authors showed that principal neurons in the affected cortex have the largest impairments at transcriptomic level, which they also cross-validated by multiple other published high-throughput datasets. The authors then moved to culture experiments to reveal the molecular mechanisms of lipid metabolism and mitochondrial dysfunction and how these processes can lead to AD pathology. Overall, this manuscript is one of the rare examples when single-cell transcriptomics data is followed by multiple complementary analysis, including functional experiments. Thus, the authors did not stop at hypothesis generating by single-cell analysis and performed extensive functional validation of their hypotheses. The level of the findings and novelty are of high importance and the manuscript can be accepted in Nature upon addressing several major and minor issues.

\textbf{Major issues:}

Although the authors tried to address batch-related effects in their snRNA-seq data, the analysis is rather superficial. Below are several batch-associated issues that should be addressed. As we understand, based on Methods section, there are 3 batches, not 2: 2 batches in batch #1 (this study), 1 batch in batch #2 (2019 study). Thus, the batch effects should be re-analyzed and shown for 3 batches, not 2.

To understand how many samples are in this study and how many are from 2019 study, the reader needs to go through Extended Table 2. The authors should make it easier to find and include in Fig 1C which samples are new which are from 2019 study.

Extended Table 2 is rather short and better description of metadata is necessary, which will help for future re-use of the data by other groups. The authors should include multiple technical parameters, such as version of 10x kit, type of Illumina kit used for sequencing, date/batch of sequencing (we believe that multiple Nextseq kits were used to sequence for 2019 study), total nucleus number (loaded/recovered), etc.

For DE analysis, the authors should perform a PCA or similar analysis for overall transcriptomic similarity between the samples and color by sequencing batch, sex, age and few other major covariates.

The difference between v2 and v3.1 10x Genomics chemistry should be extensively compared – number of cells per sample, number of genes per cell, number of UMIs. In addition, was it only v3.1 or also v3? v3.1 is different from v3 and these should be grouped separately for batch effect analysis. Finally, please compare v2 vs v3 data for its impact on cell composition.

The authors showed overall effects on principal and GABAergic neurons. However, it is a pity to miss any potential neuron type-specific effects since the data is there. If the data does not allow to subdivide and analyze fine subtypes of neurons, at least DE analysis should be performed for major families of principal (CUX2, RORB, FEZF2, THEMIS) and GABAergic (SST, PVALB, VIP, ID2) neurons. In particular, potential association of pathways in Fig 2C,D with specific types of principal neurons is interesting.

Analysis of cell type abundance changes is also missing – there multiple computational tools available.

The protocol for iNs is rather short and neurons are still immature. Although the changes in electrophysiology in Fig. S8 are robust, the authors should confirm whether these changes are due to maturational delay of ABCA7 LoF neurons or are stable and also persist in more mature neurons.

There are several issues that should be addressed to improve data and presentation in Fig. 5 experiments:
\begin{enumerate}
	\item NeuN/GFP signal in Fig5A is higher upon CDP-choline treatment – is it a consistent phenomenon or just happened for this particular image? If it is consistent, it can be that CDP-choline treatment for 2 weeks accelerates neuronal maturation and this should be measured.
	\item Quantifications in Fig. 5C show very high variability for vehicle treated cells. This experiment must be re-done to convince the reader that the mitochondrial membrane potential is indeed decreased upon CDP-choline treatment.
	\item The experiment in Fig. 5D is important to link changes in lipid metabolism and mitochondrial function to amyloid production. However, the change in Ab42 is rather minor. One of the reasons can be that vehicle-treated cells have too low levels of Ab42. Longer culture might help to increase Ab42 accumulation. In fact, same holds true for Fig. S9 experiments and longer culture might help the authors to have stronger support for Ab42-associated accumulation in ABCA7 LoF lines.
	\item In general, GFP signal is overexposed and better representative images with moderate exposition should be provided.
	\item It is unclear what was used as “n” in different Fig. 5 plots, is it always 1 well? N is quite variable in different quantification panels in Fig. 5 and some clarifications should be provided.
\end{enumerate}

The title is somewhat misguiding the readers – it is not single-cell atlas but functional experiments that reveal impaired neuronal respiration via choline-dependent lipid imbalances. Single-cell data helped the authors to generate the hypothesis, which they confirmed functionally. Thus, the title should be amended. Furthermore, the major strength of this manuscript is not in providing interactive single-cell atlas (which is already great), but rather in discovery of novel and potentially clinically relevant mechanisms of pathology in AD.

\textbf{Minor issues:}
Please explain why BA10 region was selected.

Fig S1H – why astrocytic marker expression overlap with glutamatergic?
	
Fig 1F has too many gene names, please either remove some and focus on most important or provide additional larger panels in extended data.
	
Data S6 does not have information on “34 pathways with evidence for transcriptional perturbation at p<0.05 in excitatory neurons”, at least we could not find it.

For “trend related to relatively increased triglyceride and decreased phosphatidylcholine abundance in the postmortem human PFC of ABCA7 LoF carriers (Fig. S9C )”This is not 9C, but 10C. In addition, p values are rather high; this data cannot be used to support the hypothesis and comments for trends should be removed. Otherwise, the experiment can be expanded to more samples.

The authors use MitoHealth dye as a proxy for mitochondrial membrane potential. However, when presenting immunofluorescent images, they label red fluorescence signal as delta psi m. This is not correct – the authors measured MitoHealth dye signal intensity and while in the text they can refer to MitoHealth dye as a proxy for mitochondrial membrane potential the labeling on the figure panels in Fig. 4 and 5 should be corrected.
	
Fig S11B, S12D are difficult to read, since color code is not provided. In addition, even with color code, there are too many lines – it will help for the reader to color by condition to visualize the variability within/between conditions.
		
Panels in Fig 4, Fig 5 and S11, S12: p values >10 to -4 should be presented in a regular form, without multiplication by 10.


