\subsubsection*{Referee \#4 (Remarks to the Author)}

The study by Maydell et al investigates the impact of ABCA7 LoF mutations on gene expression in patient brains with follow-up functional in vitro experiments in human neurons. The authors first performed snRNA-seq analysis to show that ABCA7 LoF cells in the brain have perturbed lipid metabolism and mitochondrial function, which might contribute to impaired neurotransmission. The authors showed that principal neurons in the affected cortex have the largest impairments at transcriptomic level, which they also cross-validated by multiple other published high-throughput datasets. The authors then moved to culture experiments to reveal the molecular mechanisms of lipid metabolism and mitochondrial dysfunction and how these processes can lead to AD pathology. Overall, this manuscript is one of the rare examples when single-cell transcriptomics data is followed by multiple complementary analysis, including functional experiments. Thus, the authors did not stop at hypothesis generating by single-cell analysis and performed extensive functional validation of their hypotheses. The level of the findings and novelty are of high importance and the manuscript can be accepted in Nature upon addressing several major and minor issues.

\textcolor{blue}{We thank the reviewer for their positive assessment and constructive suggestions!}

\textbf{Major issues:}

Although the authors tried to address batch-related effects in their snRNA-seq data, the analysis is rather superficial. Below are several batch-associated issues that should be addressed. As we understand, based on Methods section, there are 3 batches, not 2: 2 batches in batch \#1 (this study), 1 batch in batch \#2 (2019 study). Thus, the batch effects should be re-analyzed and shown for 3 batches, not 2.\\
\textcolor{blue}{We apologize for the confusion caused by our previous wording, which inadvertently implied that Batch \#1 consisted of two separate batches. In reality, Batch \#1 is a single, consistent batch where all samples underwent identical processing and library preparation. The samples were distributed across two lanes and sequenced twice on separate flow cells solely to enhance sequencing depth and overall data quality. We have now revised the methods section to clearly reflect this [page~\pageref{quoteJ-label}]}

\begin{quote}
	\textcolor{blue}{"\quoteJ\\ \\
	\quoteK\\ \\
	\quoteZ"}
\end{quote}

To understand how many samples are in this study and how many are from 2019 study, the reader needs to go through Extended Table 2. The authors should make it easier to find and include in Fig 1C which samples are new which are from 2019 study.\\
\textcolor{blue}{We have now clarified this distinction in Figure~\ref{fig:main_atlas}B. We have also updated the results section to make this clear [page~\pageref{quoteE-label}]:}

\begin{quote}
	\quoteE
\end{quote}

Extended Table 2 is rather short and better description of metadata is necessary, which will help for future re-use of the data by other groups. The authors should include multiple technical parameters, such as version of 10x kit, type of Illumina kit used for sequencing, date/batch of sequencing (we believe that multiple Nextseq kits were used to sequence for 2019 study), total nucleus number (loaded/recovered), etc.\\
\textcolor{blue}{We now provide an additional table (\ref{data:cellranger_metrics}) summarizing key technical parameters extracted from the Cell Ranger analysis, including: Sample ID, Chemistry, Include Introns, Transcriptome, Pipeline Version, Estimated Number of Cells, Mean Reads per Cell, Median Genes per Cell, Number of Reads, Valid Barcodes, Sequencing Saturation, Q30 Bases in Barcode, Q30 Bases in RNA Read, Q30 Bases in UMI, Reads Mapped to Genome, Reads Mapped Confidently to Genome, Reads Mapped Confidently to Intergenic Regions, Reads Mapped Confidently to Intronic Regions, Reads Mapped Confidently to Exonic Regions, Reads Mapped Confidently to Transcriptome, Reads Mapped Antisense to Gene, Fraction Reads in Cells, Total Genes Detected, and Median UMI Counts per Cell, and study/sequencing batch.}

\textcolor{blue}{We confirm that in the Nature 2019 study \cite{Mathys2019-dl}, sequencing was exclusively performed using NextSeq 500/550 High Output v2 kits (150 cycles). We have updated the Methods section to clearly specify the sequencing approach for each batch, including the 10x chemistry version and Illumina sequencing platforms and kits. [page~\pageref{quoteJ-label}] (see excerpt above).}\\

For DE analysis, the authors should perform a PCA or similar analysis for overall transcriptomic similarity between the samples and color by sequencing batch, sex, age and few other major covariates.\\
\textcolor{blue}{We thank the reviewer for this suggestion. We have now performed and included additional dimensionality-reduction analyses (UMAP) colored by sequencing batch, sex, age, and individual ID (Figure~\ref{fig:snRNA_batch_quality}C) to illustrate transcriptomic similarity following batch correction. Additionally, we conducted PCA on the residuals after linear model batch correction, with individuals colored according to the requested covariates, as shown below.}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{./extended_plots/PCA_of_Residuals.png}        
\end{figure}

The difference between v2 and v3.1 10x Genomics chemistry should be extensively compared – number of cells per sample, number of genes per cell, number of UMIs. In addition, was it only v3.1 or also v3? v3.1 is different from v3 and these should be grouped separately for batch effect analysis. Finally, please compare v2 vs v3 data for its impact on cell composition.

\textcolor{blue}{We can confirm that only V3 chemistry was used for samples in Batch 1 and only V2 chemistry was used for samples in Batch 2.}

\textcolor{blue}{Below, we now provide the requested comparisons of the v2 versus v3 chemistry datasets. These analyses demonstrate minor differences in cell type composition (Panel A) and more pronounced differences in the number of high-quality cells recovered (Panel B). These findings align with the documented differences between chemistries, with the v2 chemistry typically yielding fewer cells (\href{https://kb.10xgenomics.com/hc/en-us/articles/360026501692-Do-we-see-a-difference-in-the-expression-profile-of-3-Single-Cell-v3-chemistry-compared-to-v2-chemistry}{see \underline{this analysis} by 10x Genomics}).}

\begin{figure}[H]
	\begin{subfigure}[t]{0.33\textwidth}
		\caption{Sequencing metrics by chemistry}
		\includegraphics[width=\textwidth]{./extended_plots/composition_comparison.png}        
	\end{subfigure} 
	\begin{subfigure}[t]{0.65\textwidth}
		\caption{Cell type composition by chemistry}
		\includegraphics[width=\textwidth]{./extended_plots/number_of_cells_per_sample.png}        
	\end{subfigure} 
\end{figure}

\textcolor{blue}{To ensure these chemistry-related batch effects do not influence our conclusions, our analyses explicitly control for batch as a covariate. Additionally, we performed stratified analyses excluding the v2 chemistry batch and confirmed that differentially expressed gene scores remain robust and consistent with the full dataset (Figure~\ref{fig:snRNA_batch_quality}D), and shown below.}

\begin{figure}[H]
	\begin{subfigure}[t]{\textwidth}
	\includegraphics[width=\textwidth]{./extended_plots/score_correlations_by_batch.png}        
	\end{subfigure}   
\end{figure}

The authors showed overall effects on principal and GABAergic neurons. However, it is a pity to miss any potential neuron type-specific effects since the data is there. If the data does not allow to subdivide and analyze fine subtypes of neurons, at least DE analysis should be performed for major families of principal (CUX2, RORB, FEZF2, THEMIS) and GABAergic (SST, PVALB, VIP, ID2) neurons. In particular, potential association of pathways in Fig 2C,D with specific types of principal neurons is interesting.\\
\textcolor{blue}{We thank the reviewer for this insightful suggestion. While we agree that examining neuron type-specific effects is interesting, we believe this detailed analysis of principal and GABAergic neuron subtypes is beyond the primary scope of the current manuscript. However, we have made the data fully available to enable future studies exploring these finer-grained subtype-specific differences.}

Analysis of cell type abundance changes is also missing – there multiple computational tools available.\\
\textcolor{blue}{We appreciate the reviewer’s suggestion regarding cell type abundance changes. Although valuable, we consider analyses of cell type proportions outside the main scope of our current study, and better suited for dedicated follow-up work.}

The protocol for iNs is rather short and neurons are still immature. Although the changes in electrophysiology in Fig. S8 are robust, the authors should confirm whether these changes are due to maturational delay of ABCA7 LoF neurons or are stable and also persist in more mature neurons.\\
\textcolor{blue}{We thank the reviewer for this helpful suggestion. To address this point, we generated cortical organoids from p.Tyr622 and WT iNs and measured spontaneous action potential frequency using electrophysiology, as well as spontaneous calcium events using GCaMP6s calcium imaging. These experiments confirmed increased hyperexcitatbility of ABCA7 LoF neurons, demonstrating that this phenotype is robust, stable, and not due to delayed neuronal maturation. These new data are presented in Extended Data Figure~\ref{fig:main_choline}K (and Panel A directly below) and Figure~\ref{fig:neurospheroid_figure}C (and Panel B directly below), and discussed in the main text.}

\begin{figure}[H]
	\centering
	\begin{subfigure}[t]{.4\textwidth}
	\caption{Spontaneous action potentials measured in p.Tyr622* iNs with or without CDP-choline in cortical organoids.}
	\includegraphics[width=\textwidth]{./main_plots/cortical_organoids_ephys.png}        
	\end{subfigure} 
	\begin{subfigure}[t]{.4\textwidth}
	\caption{Spontaneous calcium events measured in p.Tyr622* iNs with or without CDP-choline in dissociated cortical organoids transduced with AAV1-GCaMP6s.}
	\includegraphics[width=\textwidth]{./extended_plots/calcium_data.png}        
	\end{subfigure} 
\end{figure}

There are several issues that should be addressed to improve data and presentation in Fig. 5 experiments:
\begin{enumerate}
	\item NeuN/GFP signal in Fig5A is higher upon CDP-choline treatment – is it a consistent phenomenon or just happened for this particular image? If it is consistent, it can be that CDP-choline treatment for 2 weeks accelerates neuronal maturation and this should be measured.\\
	\textcolor{blue}{We thank the reviewer for highlighting this observation. The apparent increase in NeuN/GFP signal in Figure 5A is specific to the selected image. We have now updated the panel with better representative images (\ref{fig:choline_treatment}K).}

	\item Quantifications in Fig. 5C show very high variability for vehicle treated cells. This experiment must be re-done to convince the reader that the mitochondrial membrane potential is indeed decreased upon CDP-choline treatment.\\
	\textcolor{blue}{We thank the reviewer for this helpful suggestion. We have now repeated the mitochondrial membrane potential measurements using tetramethylrhodamine methyl ester (TMRM), a fluorescent dye that accumulates within mitochondria proportionally to their membrane potential. These new TMRM experiments in p.Tyr622* iNs showed reduced variability, confirmed significantly increased mitochondrial membrane potential in ABCA7 LoF neurons compared to WT, and demonstrated rescue upon CDP-choline treatment. We have added these new data in Figure~\ref{fig:main_choline}I (and directly below), while retaining the original MitoHealth results in Figure~\ref{fig:main_mitochondrial}H and Figure~\ref{fig:choline_treatment}K for completeness.}

	\begin{figure}[H]
		\centering
		\begin{subfigure}[t]{.6\textwidth}
			\vspace{-0.15cm}
			\includegraphics[width=\textwidth]{./main_plots/tmrm_choline.png}        
		\end{subfigure} 
	\end{figure}

	\item The experiment in Fig. 5D is important to link changes in lipid metabolism and mitochondrial function to amyloid production. However, the change in Ab42 is rather minor. One of the reasons can be that vehicle-treated cells have too low levels of Ab42. Longer culture might help to increase Ab42 accumulation. In fact, same holds true for Fig. S9 experiments and longer culture might help the authors to have stronger support for Ab42-associated accumulation in ABCA7 LoF lines.\\
	\textcolor{blue}{We have validated increases in soluble Aβ40 and Aβ42 by performing ELISA on media collected from ABCA7 LoF (p.Tyr622* and p.Glu50fs3) iNs compared to WT controls, shown in Figure~\ref{fig:differentiating_iPSC_neurons}H. Due to relatively low secreted Aβ levels at this early developmental stage, we further differentiated these lines into dorsal cortical organoids and repeated ELISA assays at a mature timepoint (six months). These analyses similarly revealed elevated soluble Aβ40 and Aβ42 levels in media from p.Tyr622* cortical organoids compared to WT, as presented in Figure~\ref{fig:main_choline}L and Figure~\ref{fig:neurospheroid_figure}B, and shown below.}


	\begin{figure}[H] 
		\centering
		\begin{subfigure}[t]{.6\textwidth}
			\caption{Abeta ELISA on the media from cortical organoids (aged to six months, with 4 weeks of CDP-choline treatment (1mM)).}
			\includegraphics[width=\textwidth]{./main_plots/abeta_elisa.png}        
		\end{subfigure}  
	\end{figure}

	\item In general, GFP signal is overexposed and better representative images with moderate exposition should be provided.\\
	\textcolor{blue}{We have reduced the GFP saturation for all images, which are now presented in Figure~\ref{fig:main_mitochondrial}H and Extended Data Figure~\ref{fig:choline_treatment}K.} 

	\item It is unclear what was used as “n” in different Fig. 5 plots, is it always 1 well? N is quite variable in different quantification panels in Fig. 5 and some clarifications should be provided.\\
	\textcolor{blue}{We thank the reviewer for pointing this out. We have clarified what each "n" represents in all figure panels, including the updated choline rescue figure (Figure~\ref{fig:main_choline}, previously Figure 5), and in all associated quantification plots.}

\end{enumerate}

The title is somewhat misguiding the readers – it is not single-cell atlas but functional experiments that reveal impaired neuronal respiration via choline-dependent lipid imbalances. Single-cell data helped the authors to generate the hypothesis, which they confirmed functionally. Thus, the title should be amended. Furthermore, the major strength of this manuscript is not in providing interactive single-cell atlas (which is already great), but rather in discovery of novel and potentially clinically relevant mechanisms of pathology in AD.\\
\textcolor{blue}{We thank the reviewer for this helpful suggestion and have now updated the title to reflect the focus of the manuscript: "ABCA7 Loss-of-Function Variants Impact Phosphatidylcholine Metabolism and Mitochondrial Function in Neurons"}

\textbf{Minor issues:}
Please explain why BA10 region was selected.\\
\textcolor{blue}{We selected the BA10 region to enable direct comparison and integration with data from previous studies. Additionally, BA10 is functionally relevant to Alzheimer's disease due to its established roles in executive function and memory recall.}

Fig S1H – why astrocytic marker expression overlap with glutamatergic?\\	
\textcolor{blue}{We appreciate the reviewer noting this overlap. Upon careful inspection, we found that the overlap primarily results from the gene \textit{SLC1A2}, which is highly expressed in both astrocytes and neurons. This gene appears in both the Astrocytic and Glutamatergic annotation gene sets from PanglaoDB. Because the Glutamatergic set contains relatively few genes (10 genes) compared to the Astrocytic set (63 genes), shared high expression of \textit{SLC1A2} significantly increases the perceived overlap.}

Fig 1F has too many gene names, please either remove some and focus on most important or provide additional larger panels in extended data.\\
\textcolor{blue}{We thank the reviewer for this suggestion. We have improved the visualization by reducing the number of labeled genes in Figure~\ref{fig:main_atlas}C, clearly highlighting fewer top genes with enlarged labels. Additionally, an expanded view with more comprehensive gene annotations is provided in Figure~\ref{fig:snRNAseq_gene_scores}A.}

Data S6 does not have information on “34 pathways with evidence for transcriptional perturbation at p<0.05 in excitatory neurons”, at least we could not find it.\\
\textcolor{blue}{We have now fixed this and these data are included in Extended Data Table~\ref{data:pathway_enrichments}.}

For “trend related to relatively increased triglyceride and decreased phosphatidylcholine abundance in the postmortem human PFC of ABCA7 LoF carriers (Fig. S9C )”This is not 9C, but 10C. In addition, p values are rather high; this data cannot be used to support the hypothesis and comments for trends should be removed. Otherwise, the experiment can be expanded to more samples.\\
\textcolor{blue}{We have now removed these data.}

The authors use MitoHealth dye as a proxy for mitochondrial membrane potential. However, when presenting immunofluorescent images, they label red fluorescence signal as delta psi m. This is not correct – the authors measured MitoHealth dye signal intensity and while in the text they can refer to MitoHealth dye as a proxy for mitochondrial membrane potential the labeling on the figure panels in Fig. 4 and 5 should be corrected.\\
\textcolor{blue}{We thank the reviewer for pointing this out and have updated the text and figure panels to consistently refer to the MitoHealth dye explicitly}.

Fig S11B, S12D are difficult to read, since color code is not provided. In addition, even with color code, there are too many lines – it will help for the reader to color by condition to visualize the variability within/between conditions.\\
\textcolor{blue}{We apologize for the confusion. These plots were originally intended to display individual Seahorse assay traces without emphasizing experimental groups. We have now clarified this by adding descriptive titles to Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}B and Figure~\ref{fig:choline_treatment}I and removed the confusing color coding. To clearly visualize variability within and between conditions, traces grouped and colored by condition are provided in the preceding panels, Figure~\ref{fig:oxygen_consumption_rates_iPSC_neurons}A and Figure~\ref{fig:choline_treatment}H.}

Panels in Fig 4, Fig 5 and S11, S12: p values >10 to -4 should be presented in a regular form, without multiplication by 10.\\
\textcolor{blue}{We appreciate this suggestion from the reviewer and have now updated the p-values as suggested.}
